<!DOCTYPE html><html lang="zh-CN" data-theme="light"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width,initial-scale=1"><title>转录组分析流程｜数据处理与De novo组装（一） | 生信技术</title><meta name="keywords" content="教程,转录组组装,软件,Trinity,Rcorrector,Trimmomatic"><meta name="author" content="李星泽,lixingzee@gmail.com"><meta name="copyright" content="李星泽"><meta name="format-detection" content="telephone=no"><meta name="theme-color" content="#ffffff"><meta http-equiv="Cache-Control" content="no-transform"><meta http-equiv="Cache-Control" content="no-siteapp"><meta name="description" content="定义转录组（transcriptome）广义上指某一生理条件下，细胞内所有转录产物的集合，包括信使RNA、核糖体RNA、转运RNA及非编码RNA；狭义上指所有mRNA的集合。RNA-Seq (RNA-sequencing)也称为转录组测序，是最新发展起来的利用新一代测序技术进行转录组分析的技术，可以全面快速地获得特定细胞或组织在某一状态下几乎所有转录本的序列信息和表达信息，包括编码蛋白质的mRNA">
<meta property="og:type" content="article">
<meta property="og:title" content="转录组分析流程｜数据处理与De novo组装（一）">
<meta property="og:url" content="http://lxz9.com/2021/01/18/Trinity/index.html">
<meta property="og:site_name" content="生信技术">
<meta property="og:description" content="定义转录组（transcriptome）广义上指某一生理条件下，细胞内所有转录产物的集合，包括信使RNA、核糖体RNA、转运RNA及非编码RNA；狭义上指所有mRNA的集合。RNA-Seq (RNA-sequencing)也称为转录组测序，是最新发展起来的利用新一代测序技术进行转录组分析的技术，可以全面快速地获得特定细胞或组织在某一状态下几乎所有转录本的序列信息和表达信息，包括编码蛋白质的mRNA">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="https://z3.ax1x.com/2021/01/31/yE3IzV.jpg">
<meta property="article:published_time" content="2021-01-18T10:00:00.000Z">
<meta property="article:modified_time" content="2021-03-24T04:30:57.684Z">
<meta property="article:author" content="李星泽">
<meta property="article:tag" content="教程">
<meta property="article:tag" content="转录组组装">
<meta property="article:tag" content="软件">
<meta property="article:tag" content="Trinity">
<meta property="article:tag" content="Rcorrector">
<meta property="article:tag" content="Trimmomatic">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://z3.ax1x.com/2021/01/31/yE3IzV.jpg"><link rel="shortcut icon" href="https://z3.ax1x.com/2021/02/09/yawIzV.png"><link rel="canonical" href="http://lxz9.com/2021/01/18/Trinity/"><link rel="preconnect" href="//cdn.jsdelivr.net"/><link rel="preconnect" href="//www.google-analytics.com" crossorigin=""/><link rel="preconnect" href="//hm.baidu.com"/><link rel="preconnect" href="//busuanzi.ibruce.info"/><link rel="preconnect" href="//zz.bdstatic.com"/><link rel="stylesheet" href="/css/index.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free/css/all.min.css" media="print" onload="this.media='all'"><script>var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "https://hm.baidu.com/hm.js?72a556f842c8990127f397d4e0c54f2c";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script><script async="async" src="https://www.googletagmanager.com/gtag/js?id=G-QRG4NL4207"></script><script>window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-QRG4NL4207');
</script><script>var GLOBAL_CONFIG = { 
  root: '/',
  algolia: undefined,
  localSearch: {"path":"search.xml","languages":{"hits_empty":"找不到您查询的内容：${query}"}},
  translate: undefined,
  noticeOutdate: undefined,
  highlight: {"plugin":"highlighjs","highlightCopy":true,"highlightLang":true},
  copy: {
    success: '复制成功',
    error: '复制错误',
    noSupport: '浏览器不支持'
  },
  relativeDate: {
    homepage: false,
    post: false
  },
  runtime: '',
  date_suffix: {
    just: '刚刚',
    min: '分钟前',
    hour: '小时前',
    day: '天前',
    month: '个月前'
  },
  copyright: undefined,
  lightbox: 'mediumZoom',
  Snackbar: undefined,
  source: {
    jQuery: 'https://cdn.jsdelivr.net/npm/jquery@latest/dist/jquery.min.js',
    justifiedGallery: {
      js: 'https://cdn.jsdelivr.net/npm/justifiedGallery/dist/js/jquery.justifiedGallery.min.js',
      css: 'https://cdn.jsdelivr.net/npm/justifiedGallery/dist/css/justifiedGallery.min.css'
    },
    fancybox: {
      js: 'https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.js',
      css: 'https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.css'
    }
  },
  isPhotoFigcaption: false,
  islazyload: false,
  isanchor: false
};

var saveToLocal = {
  set: function setWithExpiry(key, value, ttl) {
    const now = new Date()
    const expiryDay = ttl * 86400000
    const item = {
      value: value,
      expiry: now.getTime() + expiryDay,
    }
    localStorage.setItem(key, JSON.stringify(item))
  },

  get: function getWithExpiry(key) {
    const itemStr = localStorage.getItem(key)

    if (!itemStr) {
      return undefined
    }
    const item = JSON.parse(itemStr)
    const now = new Date()

    if (now.getTime() > item.expiry) {
      localStorage.removeItem(key)
      return undefined
    }
    return item.value
  }
}

// https://stackoverflow.com/questions/16839698/jquery-getscript-alternative-in-native-javascript
const getScript = url => new Promise((resolve, reject) => {
  const script = document.createElement('script')
  script.src = url
  script.async = true
  script.onerror = reject
  script.onload = script.onreadystatechange = function() {
    const loadState = this.readyState
    if (loadState && loadState !== 'loaded' && loadState !== 'complete') return
    script.onload = script.onreadystatechange = null
    resolve()
  }
  document.head.appendChild(script)
})</script><script id="config_change">var GLOBAL_CONFIG_SITE = { 
  isPost: true,
  isHome: false,
  isHighlightShrink: false,
  isToc: true,
  postUpdate: '2021-03-24 12:30:57'
}</script><noscript><style type="text/css">
  #nav {
    opacity: 1
  }
  .justified-gallery img {
    opacity: 1
  }

  #recent-posts time,
  #post-meta time {
    display: inline !important
  }
</style></noscript><script>(function () {  window.activateDarkMode = function () {
    document.documentElement.setAttribute('data-theme', 'dark')
    if (document.querySelector('meta[name="theme-color"]') !== null) {
      document.querySelector('meta[name="theme-color"]').setAttribute('content', '#0d0d0d')
    }
  }
  window.activateLightMode = function () {
    document.documentElement.setAttribute('data-theme', 'light')
   if (document.querySelector('meta[name="theme-color"]') !== null) {
      document.querySelector('meta[name="theme-color"]').setAttribute('content', '#ffffff')
    }
  }
  const autoChangeMode = 'false'
  const t = saveToLocal.get('theme')
  if (autoChangeMode === '1') {
    const isDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches
    const isLightMode = window.matchMedia('(prefers-color-scheme: light)').matches
    const isNotSpecified = window.matchMedia('(prefers-color-scheme: no-preference)').matches
    const hasNoSupport = !isDarkMode && !isLightMode && !isNotSpecified
    if (t === undefined) {
      if (isLightMode) activateLightMode()
      else if (isDarkMode) activateDarkMode()
      else if (isNotSpecified || hasNoSupport) {
        const now = new Date()
        const hour = now.getHours()
        const isNight = hour <= 6 || hour >= 18
        isNight ? activateDarkMode() : activateLightMode()
      }
      window.matchMedia('(prefers-color-scheme: dark)').addListener(function (e) {
        if (saveToLocal.get('theme') === undefined) {
          e.matches ? activateDarkMode() : activateLightMode()
        }
      })
    } else if (t === 'light') activateLightMode()
    else activateDarkMode()
  } else if (autoChangeMode === '2') {
    const now = new Date()
    const hour = now.getHours()
    const isNight = hour <= 6 || hour >= 18
    if (t === undefined) isNight ? activateDarkMode() : activateLightMode()
    else if (t === 'light') activateLightMode()
    else activateDarkMode()
  } else {
    if (t === 'dark') activateDarkMode()
    else if (t === 'light') activateLightMode()
  }const asideStatus = saveToLocal.get('aside-status')
if (asideStatus !== undefined) {
   if (asideStatus === 'hide') {
     document.documentElement.classList.add('hide-aside')
   } else {
     document.documentElement.classList.remove('hide-aside')
   }
}})()</script><style>#article-container.post-content h1:before, h2:before, h3:before, h4:before, h5:before, h6:before { -webkit-animation: avatar_turn_around 1s linear infinite; -moz-animation: avatar_turn_around 1s linear infinite; -o-animation: avatar_turn_around 1s linear infinite; -ms-animation: avatar_turn_around 1s linear infinite; animation: avatar_turn_around 1s linear infinite; }</style><link rel="stylesheet" href="/css/iconfont.css"><link rel="stylesheet" href="/css/background.css"><link rel="stylesheet" href="/css/lxz9.css"><link rel="stylesheet" href="/css/icon.css" media="defer" onload="this.media='all'"><meta name="generator" content="Hexo 5.3.0"><link rel="alternate" href="/atom.xml" title="生信技术" type="application/atom+xml">
</head><body><div id="sidebar"><div id="menu-mask"></div><div id="sidebar-menus"><div class="author-avatar"><img class="avatar-img" src="https://z3.ax1x.com/2021/03/10/6JSyoq.png" onerror="onerror=null;src='/img/friend_404.gif'" alt="avatar"/></div><div class="site-data"><div class="data-item is-center"><div class="data-item-link"><a href="/archives/"><div class="headline">文章</div><div class="length-num">20</div></a></div></div><div class="data-item is-center"><div class="data-item-link"><a href="/tags/"><div class="headline">标签</div><div class="length-num">38</div></a></div></div><div class="data-item is-center"><div class="data-item-link"><a href="/categories/"><div class="headline">分类</div><div class="length-num">10</div></a></div></div></div><hr/><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fa fa-home"></i><span> 首页</span></a></div><div class="menus_item"><a class="site-page" href="/archives/"><i class="fa-fw fa fa-archive"></i><span> 时间轴</span></a></div><div class="menus_item"><a class="site-page" href="/tags/"><i class="fa-fw fa fa-tags"></i><span> 标签</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fa fa-folder-open"></i><span> 分类</span></a></div><div class="menus_item"><a class="site-page" href="javascript:void(0);"><i class="fa-fw fas fa-list"></i><span> 清单</span><i class="fas fa-chevron-down expand"></i></a><ul class="menus_item_child"><li><a class="site-page" href="/music/"><i class="fa-fw fas fa-music"></i><span> 音乐</span></a></li><li><a class="site-page" href="/Gallery/"><i class="fa-fw fas fa-images"></i><span> 照片</span></a></li><li><a class="site-page" href="/movies/"><i class="fa-fw fas fa-video"></i><span> 电影</span></a></li></ul></div><div class="menus_item"><a class="site-page" href="/link/"><i class="fa-fw fas fa-link"></i><span> 友链</span></a></div><div class="menus_item"><a class="site-page" href="/map/"><i class="fa-fw fas fa-heart"></i><span> 地图</span></a></div></div></div></div><div class="post" id="body-wrap"><header class="post-bg" id="page-header" style="background-image: url(https://z3.ax1x.com/2021/01/31/yE3IzV.jpg)"><nav id="nav"><span id="blog_name"><a id="site-name" href="/">生信技术</a></span><div id="menus"><div id="search-button"><a class="site-page social-icon search"><i class="fas fa-search fa-fw"></i><span> 搜索</span></a></div><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fa fa-home"></i><span> 首页</span></a></div><div class="menus_item"><a class="site-page" href="/archives/"><i class="fa-fw fa fa-archive"></i><span> 时间轴</span></a></div><div class="menus_item"><a class="site-page" href="/tags/"><i class="fa-fw fa fa-tags"></i><span> 标签</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fa fa-folder-open"></i><span> 分类</span></a></div><div class="menus_item"><a class="site-page" href="javascript:void(0);"><i class="fa-fw fas fa-list"></i><span> 清单</span><i class="fas fa-chevron-down expand"></i></a><ul class="menus_item_child"><li><a class="site-page" href="/music/"><i class="fa-fw fas fa-music"></i><span> 音乐</span></a></li><li><a class="site-page" href="/Gallery/"><i class="fa-fw fas fa-images"></i><span> 照片</span></a></li><li><a class="site-page" href="/movies/"><i class="fa-fw fas fa-video"></i><span> 电影</span></a></li></ul></div><div class="menus_item"><a class="site-page" href="/link/"><i class="fa-fw fas fa-link"></i><span> 友链</span></a></div><div class="menus_item"><a class="site-page" href="/map/"><i class="fa-fw fas fa-heart"></i><span> 地图</span></a></div></div><div id="toggle-menu"><a class="site-page"><i class="fas fa-bars fa-fw"></i></a></div></div></nav><div id="post-info"><h1 class="post-title">转录组分析流程｜数据处理与De novo组装（一）</h1><div id="post-meta"><div class="meta-firstline"><span class="post-meta-date"><i class="far fa-calendar-alt fa-fw post-meta-icon"></i><span class="post-meta-label">发表于</span><time class="post-meta-date-created" datetime="2021-01-18T10:00:00.000Z" title="发表于 2021-01-18 18:00:00">2021-01-18</time><span class="post-meta-separator">|</span><i class="fas fa-history fa-fw post-meta-icon"></i><span class="post-meta-label">更新于</span><time class="post-meta-date-updated" datetime="2021-03-24T04:30:57.684Z" title="更新于 2021-03-24 12:30:57">2021-03-24</time></span><span class="post-meta-categories"><span class="post-meta-separator">|</span><i class="fas fa-inbox fa-fw post-meta-icon"></i><a class="post-meta-categories" href="/categories/%E8%BD%AC%E5%BD%95%E7%BB%84%E5%88%86%E6%9E%90/">转录组分析</a></span></div><div class="meta-secondline"><span class="post-meta-separator">|</span><span class="post-meta-wordcount"><i class="far fa-file-word fa-fw post-meta-icon"></i><span class="post-meta-label">字数总计:</span><span class="word-count">2.6k</span><span class="post-meta-separator">|</span><i class="far fa-clock fa-fw post-meta-icon"></i><span class="post-meta-label">阅读时长:</span><span>12分钟</span></span><span class="post-meta-separator">|</span><span class="post-meta-pv-cv"><i class="far fa-eye fa-fw post-meta-icon"></i><span class="post-meta-label">阅读量:</span><span id="busuanzi_value_page_pv"></span></span></div></div></div></header><main class="layout" id="content-inner"><div id="post"><article class="post-content" id="article-container"><h1 id="定义"><a href="#定义" class="headerlink" title="定义"></a>定义</h1><p><a target="_blank" rel="noopener" href="https://baike.baidu.com/item/%E8%BD%AC%E5%BD%95%E7%BB%84/9998410?fr=aladdin">转录组（transcriptome）</a>广义上指某一生理条件下，细胞内所有转录产物的集合，包括信使RNA、核糖体RNA、转运RNA及非编码RNA；狭义上指所有mRNA的集合。<br><a target="_blank" rel="noopener" href="https://baike.baidu.com/item/%E8%BD%AC%E5%BD%95%E7%BB%84%E6%B5%8B%E5%BA%8F/486410">RNA-Seq (RNA-sequencing)</a>也称为转录组测序，是最新发展起来的利用新一代测序技术进行转录组分析的技术，可以全面快速地获得特定细胞或组织在某一状态下几乎所有转录本的序列信息和表达信息，包括编码蛋白质的mRNA和各种非编码RNA，基因选择性剪接产生的不同转录本的表达丰度等。在分析转录本的结构和表达水平的同时，还发现未知转录本和稀有转录本，从而准确地分析基因表达差异、基因结构变异、筛选分子标记等生命科学的重要问题。</p>
<h1 id="组装软件及用法"><a href="#组装软件及用法" class="headerlink" title="组装软件及用法"></a>组装软件及用法</h1><h2 id="数据矫正"><a href="#数据矫正" class="headerlink" title="数据矫正"></a>数据矫正</h2><p>使用<code>rcorrector</code>软件对数据进行矫正,输入<code>run_rcorrector.pl</code>弹出使用说明</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line">$ run_rcorrector.pl</span><br><span class="line">Usage: perl ./run_rcorrector.pl [OPTIONS]</span><br><span class="line">OPTIONS:</span><br><span class="line">Required parameters:</span><br><span class="line">	-s seq_files: comma separated files <span class="keyword">for</span> single-end data sets</span><br><span class="line">	-1 seq_files_left: comma separated files <span class="keyword">for</span> the first mate <span class="keyword">in</span> the paried-end data sets</span><br><span class="line">	-2 seq_files_right: comma separated files <span class="keyword">for</span> the second mate <span class="keyword">in</span> the paired-end data sets</span><br><span class="line">	-i seq_files_interleaved: comma sperated files <span class="keyword">for</span> interleaved paired-end data sets</span><br><span class="line">Other parameters:</span><br><span class="line">	-k kmer_length (&lt;=32, default: 23)</span><br><span class="line">	-od output_file_directory (default: ./)</span><br><span class="line">	-t number_of_threads (default: 1)</span><br><span class="line">	-maxcorK INT: the maximum number of correction within k-bp window (default: 4)</span><br><span class="line">	-wk FLOAT: the proportion of kmers that are used to estimate weak kmer count threshold, lower <span class="keyword">for</span> more divergent genome (default: 0.95)</span><br><span class="line">	-ek expected_number_of_kmers: does not affect the correctness of program but affect the memory usage (default: 100000000)</span><br><span class="line">	-stdout: output the corrected reads to stdout (default: not used)</span><br><span class="line">	-verbose: output some correction information to stdout (default: not used)</span><br><span class="line">	-stage INT: start from <span class="built_in">which</span> stage (default: 0)</span><br><span class="line">		0-start from begining(storing kmers <span class="keyword">in</span> bloom filter);</span><br><span class="line">		1-start from count kmers showed up <span class="keyword">in</span> bloom filter;</span><br><span class="line">		2-start from dumping kmer counts into a jf_dump file;</span><br><span class="line">		3-start from error correction.</span><br></pre></td></tr></table></figure>
<p>上面说的很详细，下面在介绍几个常用到的命令</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">-s seq_files:单端测序的文件选择-s命令</span><br><span class="line">-1 seq_files_left -2 seq_files_right:：双端测序的选择此命令输入左右两端测序的两个文件，目前大多为PE data</span><br><span class="line">-k 设置kmer值，也可以直接不选择按照默认参数</span><br><span class="line">-od 输出文件名称</span><br><span class="line">-t 所用线程数根据自己情况进行选择</span><br></pre></td></tr></table></figure>
<p>当然如果你只有一对数据的话可以单独进行操作，但是当你的转录组数据有很多时，这里推荐进行批量处理，具体的操作如下：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">cat 文件名称 | while read f; do run_rcorrector.pl -t 12 -p $&#123;f&#125;&quot;_1.fq.gz&quot; $&#123;f&#125;&quot;_2.fq.gz&quot;; done</span><br></pre></td></tr></table></figure>
<p>cat+自己创建的文件名，如果你的序列为</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">ABC_1.fq.gz,ABC_2.fq.gz;</span><br><span class="line">BCD_1.fq.gz,BCD_2.fq.gz;</span><br><span class="line">CDE_1.fq.gz,CDE_2.fq.gz;</span><br></pre></td></tr></table></figure>
<p>其中文件的格式应该为</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">ABC</span><br><span class="line">BCD</span><br><span class="line">CDE</span><br></pre></td></tr></table></figure>
<p>这里如果你的数据是_1.fq.gz以及_2.fq.gz则不需要修改，但是如果后缀和这里不一致，就修改成你自己数据的名称后缀<br>（注意，处理的所有数据的后缀名应保持一致）<br>跑完之后会在原有的<code>*.fq.gz</code>生成<code> *.cor.fq.gz</code></p>
<h2 id="去接头和低质量序列"><a href="#去接头和低质量序列" class="headerlink" title="去接头和低质量序列"></a>去接头和低质量序列</h2><p>去接头一般是需要去除掉测序时多余的接头，这一步的话如果测序数据下来之后已经做过了就可以选择可做可不做，说一下具体操作步骤<br>这里用到的软件为<code>Trimmomatic</code><br><strong>软件安装方法：</strong><br>1、直接进入<a target="_blank" rel="noopener" href="http://www.usadellab.org/cms/index.php?page=trimmomatic">Trimmomatic</a>官方下载<br>2、<code>conda install -c bioconda trimmomatic</code></p>
<p><strong>下载Illumina双端接头序列</strong></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">curl -O -L http://dib-training.ucdavis.edu.s3.amazonaws.com/mRNAseq-semi-2015-03-04/TruSeq2-PE.fa</span><br></pre></td></tr></table></figure>
<p>或者从官网下载的话自带有一般在adapter文件夹里 <code>~/Trimmomatic-0.39/adapters</code></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">$ trimmomatic</span><br><span class="line">Usage: </span><br><span class="line">       PE [-version] [-threads &lt;threads&gt;] [-phred33|-phred64] [-trimlog &lt;trimLogFile&gt;] [-summary &lt;statsSummaryFile&gt;] [-quiet] [-validatePairs] [-basein &lt;inputBase&gt; | &lt;inputFile1&gt; &lt;inputFile2&gt;] [-baseout &lt;outputBase&gt; | &lt;outputFile1P&gt; &lt;outputFile1U&gt; &lt;outputFile2P&gt; &lt;outputFile2U&gt;] &lt;trimmer1&gt;...</span><br><span class="line">   or: </span><br><span class="line">       SE [-version] [-threads &lt;threads&gt;] [-phred33|-phred64] [-trimlog &lt;trimLogFile&gt;] [-summary &lt;statsSummaryFile&gt;] [-quiet] &lt;inputFile&gt; &lt;outputFile&gt; &lt;trimmer1&gt;...</span><br><span class="line">   or: </span><br><span class="line">       -version</span><br></pre></td></tr></table></figure>
<p>单端测序<br><code>trimmomatic SE -phred33 input.fq.gz output.fq.gz ILLUMINACLIP:TruSeq3-SE:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36</code><br>双端测序<br><code>trimmomatic PE -phred33 input_forward.fq.gz input_reverse.fq.gz output_forward_paired.fq.gz output_forward_unpaired.fq.gz output_reverse_paired.fq.gz output_reverse_unpaired.fq.gz ILLUMINACLIP:TruSeq3-PE.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36</code><br>或者将<code>trimmomatic</code>改为<code>java -jar trimmomatic-0.35.jar</code>运行</p>
<blockquote>
<p>常用参数：<br>-threads 线程数<br>-trimlog 生成日志名，log文件较多建议不开启<br>-quiet 静默模式</p>
</blockquote>
<blockquote>
<p>修整步骤：<br>ILLUMINACLIP：从reads中剪切adapter和其他Illumina特定序列。<br>SLIDINGWINDOW：执行滑动窗口修剪，一旦窗口内的平均质量低于阈值，则切割。<br>LEADING：如果低于阈值质量，则在reads起始处剪切碱基<br>TRAILING：如果低于阈值质量，则在reads末尾处剪切碱基<br>CROP：将reads从末尾切割为指定长度<br>HEADCROP：从reads剪切后低于指定长度，则删除<br>MINLEN：如果reads低于指定长度，则删除<br>TOPHRED33：将质量得分转换为Phred-33<br>TOPHRED64：将质量得分转换为Phred-64</p>
</blockquote>
<p>若数据有很多对转录组，建议批量操作</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">cat 文件名 | while read f; do time java -jar &#x2F;home&#x2F;你的路径&#x2F;software&#x2F;Trimmomatic-0.39&#x2F;trimmomatic-0.39.jar PE -threads 20 -phred33 $&#123;f&#125;&quot;_1.cor.fq.gz&quot; $&#123;f&#125;&quot;_2.cor.fq.gz&quot; $&#123;f&#125;&quot;_1.cor.p.fq.gz&quot; $&#123;f&#125;&quot;_1.cor.u.fq.gz&quot; $&#123;f&#125;&quot;_2.cor.p.fq.gz&quot; $&#123;f&#125;&quot;_2.cor.u.fq.gz&quot; ILLUMINACLIP:&#x2F;home&#x2F;lixingze&#x2F;software&#x2F;Trimmomatic-0.39&#x2F;adapters&#x2F;TruSeq2-PE.fa:2:30:10 SLIDINGWINDOW:4:15 LEADING:5 TRAILING:4 MINLEN:80; done</span><br></pre></td></tr></table></figure>
<p><code>$&#123;f&#125;&quot;_1.cor.fq.gz&quot; $&#123;f&#125;&quot;_2.cor.fq.gz&quot; </code>为输入的文件<br><code>$&#123;f&#125;&quot;_1.cor.p.fq.gz&quot; $&#123;f&#125;&quot;_1.cor.u.fq.gz&quot; $&#123;f&#125;&quot;_2.cor.p.fq.gz&quot; $&#123;f&#125;&quot;_2.cor.u.fq.gz&quot; </code>为输出的文件<br>后缀一致，将名称放入文件中 <code>cat 文件</code> 即可,和上面形式一致。</p>
<h1 id="转录组De-novo组装"><a href="#转录组De-novo组装" class="headerlink" title="转录组De novo组装"></a>转录组De novo组装</h1><p>用<code>Trinity</code>进行转录组组装<br><img src="https://z3.ax1x.com/2021/01/24/sbRNvQ.png" alt="Trinity"></p>
<blockquote>
</blockquote>
<p><code>Trinity</code>包括三个模块，能处理大型的RNA数据<br>Inchworn: 将RNA-seq数据组装成单个转录本，通常是主要转录亚型的全长转录本<br>Chrysalis: 这一步将上一步得到contig进行聚类，对于每个聚类构建完整的德布罗意图(de Bruijin graph)每个转录本表示的是给定基因或者一组有着共同序列的基因的全部转录组成。 之后会根据图中不相交的点对全部短读数据进行拆分<br>Butterfly: 并行处理各个图(graph), 追踪每个图中的短读和配对短读的路径，最后报告可变剪切亚型的全长转录本，并且区分出旁系同源基因的转录本</p>
<h2 id="安装方法"><a href="#安装方法" class="headerlink" title="安装方法"></a>安装方法</h2><p>1、<code>conda install -c bioconda trinity</code><br>2、<a target="_blank" rel="noopener" href="https://github.com/trinityrnaseq/trinityrnaseq/">trinity官网</a><br>因为trinity需要的依赖包很多，这里建议用<code>conda</code>安装,目前最新版本为Trinity-v2.11.0</p>
<h2 id="操作步骤"><a href="#操作步骤" class="headerlink" title="操作步骤"></a>操作步骤</h2><p>安装成功后输入<code>Trinity</code>弹出如下</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br></pre></td><td class="code"><pre><span class="line">$ Trinity</span><br><span class="line"></span><br><span class="line"><span class="comment">###############################################################################</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"></span><br><span class="line">     ______  ____   ____  ____   ____  ______  __ __</span><br><span class="line">    |      ||    \ |    ||    \ |    ||      ||  |  |</span><br><span class="line">    |      ||  D  ) |  | |  _  | |  | |      ||  |  |</span><br><span class="line">    |_|  |_||    /  |  | |  |  | |  | |_|  |_||  ~  |</span><br><span class="line">      |  |  |    \  |  | |  |  | |  |   |  |  |___, |</span><br><span class="line">      |  |  |  .  \ |  | |  |  | |  |   |  |  |     |</span><br><span class="line">      |__|  |__|\_||____||__|__||____|  |__|  |____/</span><br><span class="line"></span><br><span class="line">    Trinity-v2.11.0</span><br><span class="line"></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment"># Required:</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment">#  --seqType &lt;string&gt;      :type of reads: (&#x27;fa&#x27; or &#x27;fq&#x27;)</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment">#  --max_memory &lt;string&gt;      :suggested max memory to use by Trinity where limiting can be enabled. (jellyfish, sorting, etc)</span></span><br><span class="line"><span class="comment">#                            provided in Gb of RAM, ie.  &#x27;--max_memory 10G&#x27;</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment">#  If paired reads:</span></span><br><span class="line"><span class="comment">#      --left  &lt;string&gt;    :left reads, one or more file names (separated by commas, no spaces)</span></span><br><span class="line"><span class="comment">#      --right &lt;string&gt;    :right reads, one or more file names (separated by commas, no spaces)</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment">#  Or, if unpaired reads:</span></span><br><span class="line"><span class="comment">#      --single &lt;string&gt;   :single reads, one or more file names, comma-delimited (note, if single file contains pairs, can use flag: --run_as_paired )</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment">#  Or,</span></span><br><span class="line"><span class="comment">#      --samples_file &lt;string&gt;         tab-delimited text file indicating biological replicate relationships.</span></span><br><span class="line"><span class="comment">#                                   ex.</span></span><br><span class="line"><span class="comment">#                                        cond_A    cond_A_rep1    A_rep1_left.fq    A_rep1_right.fq</span></span><br><span class="line"><span class="comment">#                                        cond_A    cond_A_rep2    A_rep2_left.fq    A_rep2_right.fq</span></span><br><span class="line"><span class="comment">#                                        cond_B    cond_B_rep1    B_rep1_left.fq    B_rep1_right.fq</span></span><br><span class="line"><span class="comment">#                                        cond_B    cond_B_rep2    B_rep2_left.fq    B_rep2_right.fq</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment">#                      # if single-end instead of paired-end, then leave the 4th column above empty.</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment">####################################</span></span><br><span class="line"><span class="comment">##  Misc:  #########################</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment">#  --include_supertranscripts      :yield supertranscripts fasta and gtf files as outputs.</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment">#  --SS_lib_type &lt;string&gt;          :Strand-specific RNA-Seq read orientation.</span></span><br><span class="line"><span class="comment">#                                   if paired: RF or FR,</span></span><br><span class="line"><span class="comment">#                                   if single: F or R.   (dUTP method = RF)</span></span><br><span class="line"><span class="comment">#                                   See web documentation.</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment">#  --CPU &lt;int&gt;                     :number of CPUs to use, default: 2</span></span><br><span class="line"><span class="comment">#  --min_contig_length &lt;int&gt;       :minimum assembled contig length to report</span></span><br><span class="line"><span class="comment">#                                   (def=200)</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment">#  --long_reads &lt;string&gt;           :fasta file containing error-corrected or circular consensus (CCS) pac bio reads</span></span><br><span class="line"><span class="comment">#                                   (** note: experimental parameter **, this functionality continues to be under development)</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment">#  --genome_guided_bam &lt;string&gt;    :genome guided mode, provide path to coordinate-sorted bam file.</span></span><br><span class="line"><span class="comment">#                                   (see genome-guided param section under --show_full_usage_info)</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment">#  --jaccard_clip                  :option, set if you have paired reads and</span></span><br><span class="line"><span class="comment">#                                   you expect high gene density with UTR</span></span><br><span class="line"><span class="comment">#                                   overlap (use FASTQ input file format</span></span><br><span class="line"><span class="comment">#                                   for reads).</span></span><br><span class="line"><span class="comment">#                                   (note: jaccard_clip is an expensive</span></span><br><span class="line"><span class="comment">#                                   operation, so avoid using it unless</span></span><br><span class="line"><span class="comment">#                                   necessary due to finding excessive fusion</span></span><br><span class="line"><span class="comment">#                                   transcripts w/o it.)</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment">#  --trimmomatic                   :run Trimmomatic to quality trim reads</span></span><br><span class="line"><span class="comment">#                                        see &#x27;--quality_trimming_params&#x27; under full usage info for tailored settings.</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment">#  --output &lt;string&gt;               :name of directory for output (will be</span></span><br><span class="line"><span class="comment">#                                   created if it doesn&#x27;t already exist)</span></span><br><span class="line"><span class="comment">#                                   default( your current working directory: &quot;/home/lixingze/trinity_out_dir&quot; </span></span><br><span class="line"><span class="comment">#                                    note: must include &#x27;trinity&#x27; in the name as a safety precaution! )</span></span><br><span class="line"><span class="comment">#  </span></span><br><span class="line"><span class="comment">#  --full_cleanup                  :only retain the Trinity fasta file, rename as $&#123;output_dir&#125;.Trinity.fasta</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment">#  --cite                          :show the Trinity literature citation</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment">#  --verbose                       :provide additional job status info during the run.</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment">#  --version                       :reports Trinity version (Trinity-v2.11.0) and exits.</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment">#  --show_full_usage_info          :show the many many more options available for running Trinity (expert usage).</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment">###############################################################################</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment">#  *Note, a typical Trinity command might be:</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment">#        Trinity --seqType fq --max_memory 50G --left reads_1.fq  --right reads_2.fq --CPU 6</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment">#            (if you have multiple samples, use --samples_file ... see above for details)</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment">#    and for Genome-guided Trinity, provide a coordinate-sorted bam:</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment">#        Trinity --genome_guided_bam rnaseq_alignments.csorted.bam --max_memory 50G</span></span><br><span class="line"><span class="comment">#                --genome_guided_max_intron 10000 --CPU 6</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment">#     see: /home/lixingze/software/trinityrnaseq-v2.11.0/sample_data/test_Trinity_Assembly/</span></span><br><span class="line"><span class="comment">#          for sample data and &#x27;runMe.sh&#x27; for example Trinity execution</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment">#     For more details, visit: http://trinityrnaseq.github.io</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment">###############################################################################</span></span><br></pre></td></tr></table></figure>
<p>上面给的信息很详细运行只需要下面一条命令，接下来就是等待结果～</p>
<p><strong>几个常用命令介绍</strong><br><code>Trinity --seqType fq --samples_file 文件名称 --CPU 20 --max_memory 70G --output 输出文件夹</code></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line">Trinity </span><br><span class="line">--seqType reads类型fq or fa</span><br><span class="line">--samples_file 如果文件过多，可以将信息写入samples文件中</span><br><span class="line">文件格式已给出：</span><br><span class="line">cond_A    cond_A_rep1    A_rep1_left.fq    A_rep1_right.fq</span><br><span class="line">cond_A    cond_A_rep2    A_rep2_left.fq    A_rep2_right.fq</span><br><span class="line">cond_B    cond_B_rep1    B_rep1_left.fq    B_rep1_right.fq</span><br><span class="line">cond_B    cond_B_rep2    B_rep2_left.fq    B_rep2_right.fq</span><br><span class="line">--CPU 线程数</span><br><span class="line">--max_memory 最大内存数根据自己服务器内存自己定</span><br><span class="line">--output 输出文件名</span><br><span class="line">--left : left reads，多个文件以逗号隔开</span><br><span class="line">--right : right reads，多个文件以逗号隔开</span><br></pre></td></tr></table></figure>
<p>查看trinity的完整的全部参数 <code>Trinity  --show_full_usage_info</code> </p>
<h2 id="数据解读"><a href="#数据解读" class="headerlink" title="数据解读"></a>数据解读</h2><p>组装完成之后会在输出文件夹中输出一个<code>Trinity.fasta</code>这个就是最终组装结果！</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">&gt;TRINITY_DN59_c0_g1_i10 len=4860 path=[0:0-614 2:615-757 4:758-815 5:816-882 7:883-919 9:920-1052 10:1053-1080 11:1081-1408 13:1409-1434 15:1435-1614 16:1615-1638 18:1639-1773 20:1774-1960 22:1961-2333 24:2334-2374 26:2375-2404 27:2405-2991 29:2992-4149 30:4150-4212 32:4213-4318 33:4319-4526 34:4527-4559 36:4560-4859]</span><br><span class="line">CTTTGCCTCTCCTCAGAAGAGATGACAAGTGCATCAAGGAGTAGTTTTGCATCAACTACACATCATGCTTCGATGTTGCAGATAAGAGATTTCTCAGCATACCATATGGAAAATGTTTCCTACTATGCCTCCATGCATACTCAAAACGCCTCCTGAATTGGGGATAAATAGTTCTCGTATAGAATATAGGTTAATTATACAACAAATAAAGAACAAAGCTGTTAAACTGATTTTATGCATGAACACACTTTTTGA略</span><br></pre></td></tr></table></figure>
<p>其中<code>TRINITY_DN59_c0</code>是Trinity聚类编号，<code>g1</code>是基因编号，<code>i10</code>是转录组亚型编号</p>
<h1 id="转录组后续分析"><a href="#转录组后续分析" class="headerlink" title="转录组后续分析"></a>转录组后续分析</h1><p>组装完成之后，就需要对其进行注释等后续操作，将会在下一篇中提到。</p>
<a class="btn-beautify button--animated purple larger" target="_blank" rel="noopener" href="https://lixingze.xyz" title="点击返回主页"><i class="far fa-hand-point-right"></i><span>点击返回主页</span></a>
</article><div class="post-copyright"><div class="post-copyright__author"><span class="post-copyright-meta">文章作者: </span><span class="post-copyright-info"><a href="mailto:lixingzee@gmail.com">李星泽</a></span></div><div class="post-copyright__type"><span class="post-copyright-meta">文章链接: </span><span class="post-copyright-info"><a href="http://lxz9.com/2021/01/18/Trinity/">http://lxz9.com/2021/01/18/Trinity/</a></span></div><div class="post-copyright__notice"><span class="post-copyright-meta">版权声明: </span><span class="post-copyright-info">本博客所有文章除特别声明外，均采用 <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" target="_blank">CC BY-NC-SA 4.0</a> 许可协议。转载请注明来自 <a href="http://lxz9.com" target="_blank">生信技术</a>！</span></div></div><div class="tag_share"><div class="post-meta__tag-list"><a class="post-meta__tags" href="/tags/%E6%95%99%E7%A8%8B/">教程</a><a class="post-meta__tags" href="/tags/%E8%BD%AC%E5%BD%95%E7%BB%84%E7%BB%84%E8%A3%85/">转录组组装</a><a class="post-meta__tags" href="/tags/%E8%BD%AF%E4%BB%B6/">软件</a><a class="post-meta__tags" href="/tags/Trinity/">Trinity</a><a class="post-meta__tags" href="/tags/Rcorrector/">Rcorrector</a><a class="post-meta__tags" href="/tags/Trimmomatic/">Trimmomatic</a></div><div class="post_share"><div class="social-share" data-image="https://z3.ax1x.com/2021/01/31/yE3IzV.jpg" data-sites="facebook,twitter,wechat,weibo,qq"></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/social-share.js/dist/css/share.min.css" media="print" onload="this.media='all'"><script src="https://cdn.jsdelivr.net/npm/social-share.js/dist/js/social-share.min.js" defer></script></div></div><div class="post-reward"><div class="reward-button"><i class="fas fa-qrcode"></i> 打赏<div class="reward-main"><ul class="reward-all"><li class="reward-item"><a href="https://z3.ax1x.com/2021/01/19/sgZ5y6.jpg" target="_blank"><img class="post-qr-code-img" src="https://z3.ax1x.com/2021/01/19/sgZ5y6.jpg" alt="微信"/></a><div class="post-qr-code-desc">微信</div></li><li class="reward-item"><a href="https://z3.ax1x.com/2021/01/19/sgZIOK.jpg" target="_blank"><img class="post-qr-code-img" src="https://z3.ax1x.com/2021/01/19/sgZIOK.jpg" alt="支付宝"/></a><div class="post-qr-code-desc">支付宝</div></li></ul></div></div></div><nav class="pagination-post" id="pagination"><div class="prev-post pull-left"><a href="/2021/01/19/TransDecoder/"><img class="prev-cover" src="https://z3.ax1x.com/2021/01/24/sb4KRf.jpg" onerror="onerror=null;src='/img/404.jpg'" alt="cover of previous post"><div class="pagination-info"><div class="label">上一篇</div><div class="prev_info">转录组分析流程｜TransDecoder预测转录本的开放阅读框（二）</div></div></a></div><div class="next-post pull-right"><a href="/2021/01/17/mcscanX/"><img class="next-cover" src="https://z3.ax1x.com/2021/02/04/y1558I.jpg" onerror="onerror=null;src='/img/404.jpg'" alt="cover of next post"><div class="pagination-info"><div class="label">下一篇</div><div class="next_info">物种内共线性分析——思路以及踩坑总结（二）</div></div></a></div></nav><div class="relatedPosts"><div class="headline"><i class="fas fa-thumbs-up fa-fw"></i><span> 相关推荐</span></div><div class="relatedPosts-list"><div><a href="/2021/01/19/TransDecoder/" title="转录组分析流程｜TransDecoder预测转录本的开放阅读框（二）"><img class="cover" src="https://z3.ax1x.com/2021/01/24/sb4KRf.jpg" alt="cover"><div class="content is-center"><div class="date"><i class="far fa-calendar-alt fa-fw"></i> 2021-01-19</div><div class="title">转录组分析流程｜TransDecoder预测转录本的开放阅读框（二）</div></div></a></div><div><a href="/2021/01/20/salmon/" title="转录组分析流程｜基于salmon转录组批量定量流程（三）"><img class="cover" src="https://z3.ax1x.com/2021/01/31/yE3WIs.jpg" alt="cover"><div class="content is-center"><div class="date"><i class="far fa-calendar-alt fa-fw"></i> 2021-01-20</div><div class="title">转录组分析流程｜基于salmon转录组批量定量流程（三）</div></div></a></div><div><a href="/2021/07/18/3d-DNA/" title="3d-DNA的使用及juicebox调整挂载到染色体水平 ｜ HiC辅助基因组组装（二）"><img class="cover" src="https://z3.ax1x.com/2021/07/18/W8VU5F.jpg" alt="cover"><div class="content is-center"><div class="date"><i class="far fa-calendar-alt fa-fw"></i> 2021-07-18</div><div class="title">3d-DNA的使用及juicebox调整挂载到染色体水平 ｜ HiC辅助基因组组装（二）</div></div></a></div><div><a href="/2021/08/05/ALLHIC/" title="ALLHIC使用 ｜ HiC辅助基因组组装（三）"><img class="cover" src="https://z3.ax1x.com/2021/08/05/feYdmD.png" alt="cover"><div class="content is-center"><div class="date"><i class="far fa-calendar-alt fa-fw"></i> 2021-08-05</div><div class="title">ALLHIC使用 ｜ HiC辅助基因组组装（三）</div></div></a></div><div><a href="/2021/02/08/Annoation/" title="如何进行基因组注释"><img class="cover" src="https://z3.ax1x.com/2021/02/08/yapB1H.jpg" alt="cover"><div class="content is-center"><div class="date"><i class="far fa-calendar-alt fa-fw"></i> 2021-02-08</div><div class="title">如何进行基因组注释</div></div></a></div><div><a href="/2021/07/18/HiC-Pro/" title="HiC-Pro的使用 ｜ HiC辅助基因组组装（一）"><img class="cover" src="https://z3.ax1x.com/2021/07/18/W8VNUU.jpg" alt="cover"><div class="content is-center"><div class="date"><i class="far fa-calendar-alt fa-fw"></i> 2021-07-18</div><div class="title">HiC-Pro的使用 ｜ HiC辅助基因组组装（一）</div></div></a></div></div></div><hr/><div id="post-comment"><div class="comment-head"><div class="comment-headline"><i class="fas fa-comments fa-fw"></i><span> 评论</span></div></div><div class="comment-wrap"><div><div class="vcomment" id="vcomment"></div></div></div></div></div><div class="aside-content" id="aside-content"><div class="card-widget card-info"><div class="card-info-avatar is-center"><img class="avatar-img" src="https://z3.ax1x.com/2021/03/10/6JSyoq.png" onerror="this.onerror=null;this.src='/img/friend_404.gif'" alt="avatar"/><div class="author-info__name">李星泽</div><div class="author-info__description"></div></div><div class="card-info-data"><div class="card-info-data-item is-center"><a href="/archives/"><div class="headline">文章</div><div class="length-num">20</div></a></div><div class="card-info-data-item is-center"><a href="/tags/"><div class="headline">标签</div><div class="length-num">38</div></a></div><div class="card-info-data-item is-center"><a href="/categories/"><div class="headline">分类</div><div class="length-num">10</div></a></div></div><a class="button--animated" id="card-info-btn" target="_blank" rel="noopener" href="https://github.com/xxxxxx"><i class="fab fa-github"></i><span>Follow Me</span></a><div class="card-info-social-icons is-center"><a class="social-icon" href="https://z3.ax1x.com/2021/02/07/ytWWB6.jpg" target="_blank" title="QQ"><i class="iconfont icon-QQ"></i></a><a class="social-icon" href="https://z3.ax1x.com/2021/10/31/ISELWt.jpg" target="_blank" title="wechat"><i class="iconfont icon-weixin"></i></a><a class="social-icon" href="https://space.bilibili.com/113291922?from=search&amp;seid=3526734520747213504&amp;spm_id_from=333.337.0.0" target="_blank" title="bilibili"><i class="iconfont icon-bilibili-fill"></i></a><a class="social-icon" href="mailto:lixingzee@gmail.com" target="_blank" title="Email"><i class="fas fa-envelope"></i></a></div></div><div class="card-widget card-announcement"><div class="item-headline"><i class="fas fa-bullhorn card-announcement-animation"></i><span>公告</span></div><div class="announcement_content"></div></div><div class="sticky_layout"><div class="card-widget" id="card-toc"><div class="item-headline"><i class="fas fa-stream"></i><span>目录</span></div><div class="toc-content"><ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#%E5%AE%9A%E4%B9%89"><span class="toc-number">1.</span> <span class="toc-text">定义</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E7%BB%84%E8%A3%85%E8%BD%AF%E4%BB%B6%E5%8F%8A%E7%94%A8%E6%B3%95"><span class="toc-number">2.</span> <span class="toc-text">组装软件及用法</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#%E6%95%B0%E6%8D%AE%E7%9F%AB%E6%AD%A3"><span class="toc-number">2.1.</span> <span class="toc-text">数据矫正</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E5%8E%BB%E6%8E%A5%E5%A4%B4%E5%92%8C%E4%BD%8E%E8%B4%A8%E9%87%8F%E5%BA%8F%E5%88%97"><span class="toc-number">2.2.</span> <span class="toc-text">去接头和低质量序列</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E8%BD%AC%E5%BD%95%E7%BB%84De-novo%E7%BB%84%E8%A3%85"><span class="toc-number">3.</span> <span class="toc-text">转录组De novo组装</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#%E5%AE%89%E8%A3%85%E6%96%B9%E6%B3%95"><span class="toc-number">3.1.</span> <span class="toc-text">安装方法</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E6%93%8D%E4%BD%9C%E6%AD%A5%E9%AA%A4"><span class="toc-number">3.2.</span> <span class="toc-text">操作步骤</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E6%95%B0%E6%8D%AE%E8%A7%A3%E8%AF%BB"><span class="toc-number">3.3.</span> <span class="toc-text">数据解读</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E8%BD%AC%E5%BD%95%E7%BB%84%E5%90%8E%E7%BB%AD%E5%88%86%E6%9E%90"><span class="toc-number">4.</span> <span class="toc-text">转录组后续分析</span></a></li></ol></div></div><div class="card-widget card-recent-post"><div class="item-headline"><i class="fas fa-history"></i><span>最新文章</span></div><div class="aside-list"><div class="aside-list-item"><a class="thumbnail" href="/2021/10/31/mdBG/" title="rust-mdbg 一款用于基因组组装的高效率软件"><img src="https://z3.ax1x.com/2021/10/30/5zhbdI.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="rust-mdbg 一款用于基因组组装的高效率软件"/></a><div class="content"><a class="title" href="/2021/10/31/mdBG/" title="rust-mdbg 一款用于基因组组装的高效率软件">rust-mdbg 一款用于基因组组装的高效率软件</a><time datetime="2021-10-30T16:00:00.000Z" title="发表于 2021-10-31 00:00:00">2021-10-31</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2021/10/01/Perl01/" title="Perl语言基础(一)"><img src="https://z3.ax1x.com/2021/10/01/4TX9AI.jpg" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="Perl语言基础(一)"/></a><div class="content"><a class="title" href="/2021/10/01/Perl01/" title="Perl语言基础(一)">Perl语言基础(一)</a><time datetime="2021-09-30T16:00:00.000Z" title="发表于 2021-10-01 00:00:00">2021-10-01</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2021/09/06/wgdi/" title="WGDI 分析全基因组复制事件完整流程"><img src="https://z3.ax1x.com/2021/09/06/h4Hmo6.jpg" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="WGDI 分析全基因组复制事件完整流程"/></a><div class="content"><a class="title" href="/2021/09/06/wgdi/" title="WGDI 分析全基因组复制事件完整流程">WGDI 分析全基因组复制事件完整流程</a><time datetime="2021-09-06T13:00:00.000Z" title="发表于 2021-09-06 21:00:00">2021-09-06</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2021/08/31/lrzsz/" title="lrzsz(Linux服务器和Mac互传文件工具)"><img src="https://z3.ax1x.com/2021/08/31/haMYQJ.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="lrzsz(Linux服务器和Mac互传文件工具)"/></a><div class="content"><a class="title" href="/2021/08/31/lrzsz/" title="lrzsz(Linux服务器和Mac互传文件工具)">lrzsz(Linux服务器和Mac互传文件工具)</a><time datetime="2021-08-31T08:00:00.000Z" title="发表于 2021-08-31 16:00:00">2021-08-31</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2021/08/05/JGI/" title="JGI Phytozome 批量下载的几种方法"><img src="https://z3.ax1x.com/2021/08/05/feUpy6.jpg" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="JGI Phytozome 批量下载的几种方法"/></a><div class="content"><a class="title" href="/2021/08/05/JGI/" title="JGI Phytozome 批量下载的几种方法">JGI Phytozome 批量下载的几种方法</a><time datetime="2021-08-05T14:00:00.000Z" title="发表于 2021-08-05 22:00:00">2021-08-05</time></div></div></div></div></div></div></main><footer id="footer"><div id="footer-wrap"><div class="copyright">&copy;2020 - 2021&nbsp;<i style="color:#FF6A6A;animation: announ_animation 0.8s linear infinite;"class="fas fa-heart"></i> By 李星泽</div></div></footer></div><div id="rightside"><div id="rightside-config-hide"><button id="readmode" type="button" title="阅读模式"><i class="fas fa-book-open"></i></button><button id="darkmode" type="button" title="浅色和深色模式转换"><i class="fas fa-adjust"></i></button><button id="hide-aside-btn" type="button" title="单栏和双栏切换"><i class="fas fa-arrows-alt-h"></i></button></div><div id="rightside-config-show"><button id="rightside_config" type="button" title="设置"><i class="fas fa-cog fa-spin"></i></button><button class="close" id="mobile-toc-button" type="button" title="目录"><i class="fas fa-list-ul"></i></button><a id="to_comment" href="#post-comment" title="直达评论"><i class="fas fa-comments"></i></a><button id="go-up" type="button" title="回到顶部"><i class="fas fa-arrow-up"></i></button></div></div><div id="local-search"><div class="search-dialog"><div class="search-dialog__title" id="local-search-title">本地搜索</div><div id="local-input-panel"><div id="local-search-input"><div class="local-search-box"><input class="local-search-box--input" placeholder="搜索文章" type="text"/></div></div></div><hr/><div id="local-search-results"></div><span class="search-close-button"><i class="fas fa-times"></i></span></div><div id="search-mask"></div></div><div><script src="/js/utils.js"></script><script src="/js/main.js"></script><script src="https://cdn.jsdelivr.net/npm/medium-zoom/dist/medium-zoom.min.js"></script><script src="/js/search/local-search.js"></script><div class="js-pjax"><script>function loadValine () {
  function initValine () {
    let initData = {
      el: '#vcomment',
      appId: '',
      appKey: '',
      placeholder: 'Please leave your footprints',
      avatar: 'monsterid',
      meta: 'nick,mail,link'.split(','),
      pageSize: '10',
      lang: 'en',
      recordIP: false,
      serverURLs: '',
      emojiCDN: '',
      emojiMaps: "",
      enableQQ: false,
      path: window.location.pathname,
    }

    if (true) { 
      initData.requiredFields= ('nick,mail'.split(','))
    }
    
    if (false) {
      const otherData = false
      initData = Object.assign(initData, otherData)
    }
    
    const valine = new Valine(initData)
  }

  if (typeof Valine === 'function') initValine() 
  else getScript('https://cdn.jsdelivr.net/npm/valine/dist/Valine.min.js').then(initValine)
}

if ('Valine' === 'Valine' || !true) {
  if (true) btf.loadComment(document.getElementById('vcomment'),loadValine)
  else setTimeout(loadValine, 0)
} else {
  function loadOtherComment () {
    loadValine()
  }
}</script><script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script></div><div class="aplayer no-destroy" data-id="000PeZCQ1i4XVs" data-server="tencent" data-type="artist" data-fixed="true" data-mini="true" data-listFolded="false" data-order="random" data-preload="none" data-autoplay="flase" muted></div><script defer="defer" id="fluttering_ribbon" mobile="false" src="https://cdn.jsdelivr.net/npm/butterfly-extsrc@1/dist/canvas-fluttering-ribbon.min.js"></script><script id="canvas_nest" defer="defer" color="0,0,255" opacity="0.7" zIndex="-1" count="99" mobile="false" src="https://cdn.jsdelivr.net/npm/butterfly-extsrc@1/dist/canvas-nest.min.js"></script><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/aplayer/dist/APlayer.min.css" media="print" onload="this.media='all'"><script src="https://cdn.jsdelivr.net/npm/aplayer/dist/APlayer.min.js"></script><script src="https://cdn.jsdelivr.net/gh/metowolf/MetingJS@1.2/dist/Meting.min.js"></script><script src="https://cdn.jsdelivr.net/npm/pjax/pjax.min.js"></script><script>let pjaxSelectors = [
  'title',
  '#config_change',
  '#body-wrap',
  '#rightside-config-hide',
  '#rightside-config-show',
  '.js-pjax'
]

if (false) {
  pjaxSelectors.unshift('meta[property="og:image"]', 'meta[property="og:title"]', 'meta[property="og:url"]')
}

var pjax = new Pjax({
  elements: 'a:not([target="_blank"])',
  selectors: pjaxSelectors,
  cacheBust: false,
  analytics: true,
  scrollRestoration: false
})

document.addEventListener('pjax:complete', function () {
  window.refreshFn()

  document.querySelectorAll('script[data-pjax]').forEach(item => {
    const newScript = document.createElement('script')
    const content = item.text || item.textContent || item.innerHTML || ""
    Array.from(item.attributes).forEach(attr => newScript.setAttribute(attr.name, attr.value))
    newScript.appendChild(document.createTextNode(content))
    item.parentNode.replaceChild(newScript, item)
  })

  GLOBAL_CONFIG.islazyload && window.lazyLoadInstance.update()

  typeof chatBtnFn === 'function' && chatBtnFn()
  typeof panguInit === 'function' && panguInit()

  if (typeof gtag === 'function') {
    gtag('config', 'G-QRG4NL4207', {'page_path': window.location.pathname});
  }

  typeof loadMeting === 'function' && document.getElementsByClassName('aplayer').length && loadMeting()

  // Analytics
  if (false) {
    MtaH5.pgv()
  }

  // prismjs
  typeof Prism === 'object' && Prism.highlightAll()

  typeof preloader === 'object' && preloader.endLoading()
})


document.addEventListener('pjax:send', function () {
  typeof preloader === 'object' && preloader.initLoading()
  
  if (window.aplayers) {
    for (let i = 0; i < window.aplayers.length; i++) {
      if (!window.aplayers[i].options.fixed) {
        window.aplayers[i].destroy()
      }
    }
  }

  typeof typed === 'object' && typed.destroy()

  //reset readmode
  const $bodyClassList = document.body.classList
  $bodyClassList.contains('read-mode') && $bodyClassList.remove('read-mode')

})</script><script>(function(){
  const bp = document.createElement('script');
  const curProtocol = window.location.protocol.split(':')[0];
  if (curProtocol === 'https') {
    bp.src = 'https://zz.bdstatic.com/linksubmit/push.js';
  }
  else{
    bp.src = 'http://push.zhanzhang.baidu.com/push.js';
  }
  bp.dataset.pjax = ''
  const s = document.getElementsByTagName("script")[0];
  s.parentNode.insertBefore(bp, s);
})()</script></div></body></html>