<!DOCTYPE html><html lang="zh-CN" data-theme="light"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width,initial-scale=1"><title>Snakemake 入门教程（创建一个简单的工作流） | 生信技术</title><meta name="keywords" content="snakemake"><meta name="author" content="李星泽,lixingzee@gmail.com"><meta name="copyright" content="李星泽"><meta name="format-detection" content="telephone=no"><meta name="theme-color" content="#ffffff"><meta http-equiv="Cache-Control" content="no-transform"><meta http-equiv="Cache-Control" content="no-siteapp"><meta name="description" content="写在前面 既然写了教程就需要具有普适性，能适合大多数人的胃口，我这部分的内容以及示例主要还是参考了官方教程，但是都是我一步一步跑过的流程，所以会更有印象，送给想学 Snkaemake 但是一直没有去学的朋友们，这些内容对于有生信基础的人来讲，上手会很快，因为很多的生信软件都使用过，写起来也就没有那么晦涩，下面开始～ Snakemake 定义 Snakemake 工作流管理系统是一种用于创建可重复和">
<meta property="og:type" content="article">
<meta property="og:title" content="Snakemake 入门教程（创建一个简单的工作流）">
<meta property="og:url" content="http://lxz9.com/2021/11/10/Snakemake-1/index.html">
<meta property="og:site_name" content="生信技术">
<meta property="og:description" content="写在前面 既然写了教程就需要具有普适性，能适合大多数人的胃口，我这部分的内容以及示例主要还是参考了官方教程，但是都是我一步一步跑过的流程，所以会更有印象，送给想学 Snkaemake 但是一直没有去学的朋友们，这些内容对于有生信基础的人来讲，上手会很快，因为很多的生信软件都使用过，写起来也就没有那么晦涩，下面开始～ Snakemake 定义 Snakemake 工作流管理系统是一种用于创建可重复和">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="https://z3.ax1x.com/2021/11/10/Idni7V.png">
<meta property="article:published_time" content="2021-11-10T15:00:00.000Z">
<meta property="article:modified_time" content="2021-11-10T14:53:54.000Z">
<meta property="article:author" content="李星泽">
<meta property="article:tag" content="snakemake">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://z3.ax1x.com/2021/11/10/Idni7V.png"><link rel="shortcut icon" href="https://z3.ax1x.com/2021/02/09/yawIzV.png"><link rel="canonical" href="http://lxz9.com/2021/11/10/Snakemake-1/"><link rel="preconnect" href="//cdn.jsdelivr.net"/><link rel="preconnect" href="//www.google-analytics.com" crossorigin=""/><link rel="preconnect" href="//hm.baidu.com"/><link rel="preconnect" href="//busuanzi.ibruce.info"/><link rel="preconnect" href="//zz.bdstatic.com"/><link rel="stylesheet" href="/css/index.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free/css/all.min.css" media="print" onload="this.media='all'"><script>var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "https://hm.baidu.com/hm.js?72a556f842c8990127f397d4e0c54f2c";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script><script async="async" src="https://www.googletagmanager.com/gtag/js?id=G-QRG4NL4207"></script><script>window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-QRG4NL4207');
</script><script>var GLOBAL_CONFIG = { 
  root: '/',
  algolia: undefined,
  localSearch: {"path":"search.xml","languages":{"hits_empty":"找不到您查询的内容：${query}"}},
  translate: undefined,
  noticeOutdate: undefined,
  highlight: {"plugin":"highlighjs","highlightCopy":true,"highlightLang":true},
  copy: {
    success: '复制成功',
    error: '复制错误',
    noSupport: '浏览器不支持'
  },
  relativeDate: {
    homepage: false,
    post: false
  },
  runtime: '',
  date_suffix: {
    just: '刚刚',
    min: '分钟前',
    hour: '小时前',
    day: '天前',
    month: '个月前'
  },
  copyright: undefined,
  lightbox: 'mediumZoom',
  Snackbar: undefined,
  source: {
    jQuery: 'https://cdn.jsdelivr.net/npm/jquery@latest/dist/jquery.min.js',
    justifiedGallery: {
      js: 'https://cdn.jsdelivr.net/npm/justifiedGallery/dist/js/jquery.justifiedGallery.min.js',
      css: 'https://cdn.jsdelivr.net/npm/justifiedGallery/dist/css/justifiedGallery.min.css'
    },
    fancybox: {
      js: 'https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.js',
      css: 'https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.css'
    }
  },
  isPhotoFigcaption: false,
  islazyload: false,
  isanchor: false
};

var saveToLocal = {
  set: function setWithExpiry(key, value, ttl) {
    const now = new Date()
    const expiryDay = ttl * 86400000
    const item = {
      value: value,
      expiry: now.getTime() + expiryDay,
    }
    localStorage.setItem(key, JSON.stringify(item))
  },

  get: function getWithExpiry(key) {
    const itemStr = localStorage.getItem(key)

    if (!itemStr) {
      return undefined
    }
    const item = JSON.parse(itemStr)
    const now = new Date()

    if (now.getTime() > item.expiry) {
      localStorage.removeItem(key)
      return undefined
    }
    return item.value
  }
}

// https://stackoverflow.com/questions/16839698/jquery-getscript-alternative-in-native-javascript
const getScript = url => new Promise((resolve, reject) => {
  const script = document.createElement('script')
  script.src = url
  script.async = true
  script.onerror = reject
  script.onload = script.onreadystatechange = function() {
    const loadState = this.readyState
    if (loadState && loadState !== 'loaded' && loadState !== 'complete') return
    script.onload = script.onreadystatechange = null
    resolve()
  }
  document.head.appendChild(script)
})</script><script id="config_change">var GLOBAL_CONFIG_SITE = { 
  isPost: true,
  isHome: false,
  isHighlightShrink: false,
  isToc: true,
  postUpdate: '2021-11-10 22:53:54'
}</script><noscript><style type="text/css">
  #nav {
    opacity: 1
  }
  .justified-gallery img {
    opacity: 1
  }

  #recent-posts time,
  #post-meta time {
    display: inline !important
  }
</style></noscript><script>(function () {  window.activateDarkMode = function () {
    document.documentElement.setAttribute('data-theme', 'dark')
    if (document.querySelector('meta[name="theme-color"]') !== null) {
      document.querySelector('meta[name="theme-color"]').setAttribute('content', '#0d0d0d')
    }
  }
  window.activateLightMode = function () {
    document.documentElement.setAttribute('data-theme', 'light')
   if (document.querySelector('meta[name="theme-color"]') !== null) {
      document.querySelector('meta[name="theme-color"]').setAttribute('content', '#ffffff')
    }
  }
  const autoChangeMode = 'false'
  const t = saveToLocal.get('theme')
  if (autoChangeMode === '1') {
    const isDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches
    const isLightMode = window.matchMedia('(prefers-color-scheme: light)').matches
    const isNotSpecified = window.matchMedia('(prefers-color-scheme: no-preference)').matches
    const hasNoSupport = !isDarkMode && !isLightMode && !isNotSpecified
    if (t === undefined) {
      if (isLightMode) activateLightMode()
      else if (isDarkMode) activateDarkMode()
      else if (isNotSpecified || hasNoSupport) {
        const now = new Date()
        const hour = now.getHours()
        const isNight = hour <= 6 || hour >= 18
        isNight ? activateDarkMode() : activateLightMode()
      }
      window.matchMedia('(prefers-color-scheme: dark)').addListener(function (e) {
        if (saveToLocal.get('theme') === undefined) {
          e.matches ? activateDarkMode() : activateLightMode()
        }
      })
    } else if (t === 'light') activateLightMode()
    else activateDarkMode()
  } else if (autoChangeMode === '2') {
    const now = new Date()
    const hour = now.getHours()
    const isNight = hour <= 6 || hour >= 18
    if (t === undefined) isNight ? activateDarkMode() : activateLightMode()
    else if (t === 'light') activateLightMode()
    else activateDarkMode()
  } else {
    if (t === 'dark') activateDarkMode()
    else if (t === 'light') activateLightMode()
  }const asideStatus = saveToLocal.get('aside-status')
if (asideStatus !== undefined) {
   if (asideStatus === 'hide') {
     document.documentElement.classList.add('hide-aside')
   } else {
     document.documentElement.classList.remove('hide-aside')
   }
}})()</script><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/aplayer/dist/APlayer.min.css"  media="defer" onload="this.media='all'"><meting-js server='tencent' type='playlist' id='2134037073' fixed='true'></meting-js><style>#article-container.post-content h1:before, h2:before, h3:before, h4:before, h5:before, h6:before { -webkit-animation: avatar_turn_around 1s linear infinite; -moz-animation: avatar_turn_around 1s linear infinite; -o-animation: avatar_turn_around 1s linear infinite; -ms-animation: avatar_turn_around 1s linear infinite; animation: avatar_turn_around 1s linear infinite; }</style><link rel="stylesheet" href="/css/iconfont.css"><link rel="stylesheet" href="/css/background.css"><link rel="stylesheet" href="/css/lxz9.css"><link rel="stylesheet" href="/css/icon.css" media="defer" onload="this.media='all'"><!-- hexo injector head_end start -->
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.12.0/dist/katex.min.css">

<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-math@4.0.0/dist/style.css">
<!-- hexo injector head_end end --><meta name="generator" content="Hexo 5.3.0"><link rel="alternate" href="/atom.xml" title="生信技术" type="application/atom+xml">
</head><body><div id="sidebar"><div id="menu-mask"></div><div id="sidebar-menus"><div class="author-avatar"><img class="avatar-img" src="https://z3.ax1x.com/2021/03/10/6JSyoq.png" onerror="onerror=null;src='/img/friend_404.gif'" alt="avatar"/></div><div class="site-data"><div class="data-item is-center"><div class="data-item-link"><a href="/archives/"><div class="headline">文章</div><div class="length-num">31</div></a></div></div><div class="data-item is-center"><div class="data-item-link"><a href="/tags/"><div class="headline">标签</div><div class="length-num">45</div></a></div></div><div class="data-item is-center"><div class="data-item-link"><a href="/categories/"><div class="headline">分类</div><div class="length-num">13</div></a></div></div></div><hr/><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fa fa-home"></i><span> 首页</span></a></div><div class="menus_item"><a class="site-page" href="/archives/"><i class="fa-fw fa fa-archive"></i><span> 时间轴</span></a></div><div class="menus_item"><a class="site-page" href="/tags/"><i class="fa-fw fa fa-tags"></i><span> 标签</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fa fa-folder-open"></i><span> 分类</span></a></div><div class="menus_item"><a class="site-page" href="javascript:void(0);" rel="external nofollow noreferrer"><i class="fa-fw fas fa-list"></i><span> 清单</span><i class="fas fa-chevron-down expand"></i></a><ul class="menus_item_child"><li><a class="site-page" href="/music/"><i class="fa-fw fas fa-music"></i><span> 音乐</span></a></li><li><a class="site-page" href="/Gallery/"><i class="fa-fw fas fa-images"></i><span> 照片</span></a></li><li><a class="site-page" href="/movies/"><i class="fa-fw fas fa-video"></i><span> 电影</span></a></li></ul></div><div class="menus_item"><a class="site-page" href="/link/"><i class="fa-fw fas fa-link"></i><span> 友链</span></a></div><div class="menus_item"><a class="site-page" href="/map/"><i class="fa-fw fas fa-heart"></i><span> 地图</span></a></div></div></div></div><div class="post" id="body-wrap"><header class="post-bg" id="page-header" style="background-image: url(https://z3.ax1x.com/2021/11/10/Idni7V.png)"><nav id="nav"><span id="blog_name"><a id="site-name" href="/">生信技术</a></span><div id="menus"><div id="search-button"><a class="site-page social-icon search"><i class="fas fa-search fa-fw"></i><span> 搜索</span></a></div><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fa fa-home"></i><span> 首页</span></a></div><div class="menus_item"><a class="site-page" href="/archives/"><i class="fa-fw fa fa-archive"></i><span> 时间轴</span></a></div><div class="menus_item"><a class="site-page" href="/tags/"><i class="fa-fw fa fa-tags"></i><span> 标签</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fa fa-folder-open"></i><span> 分类</span></a></div><div class="menus_item"><a class="site-page" href="javascript:void(0);" rel="external nofollow noreferrer"><i class="fa-fw fas fa-list"></i><span> 清单</span><i class="fas fa-chevron-down expand"></i></a><ul class="menus_item_child"><li><a class="site-page" href="/music/"><i class="fa-fw fas fa-music"></i><span> 音乐</span></a></li><li><a class="site-page" href="/Gallery/"><i class="fa-fw fas fa-images"></i><span> 照片</span></a></li><li><a class="site-page" href="/movies/"><i class="fa-fw fas fa-video"></i><span> 电影</span></a></li></ul></div><div class="menus_item"><a class="site-page" href="/link/"><i class="fa-fw fas fa-link"></i><span> 友链</span></a></div><div class="menus_item"><a class="site-page" href="/map/"><i class="fa-fw fas fa-heart"></i><span> 地图</span></a></div></div><div id="toggle-menu"><a class="site-page"><i class="fas fa-bars fa-fw"></i></a></div></div></nav><div id="post-info"><h1 class="post-title">Snakemake 入门教程（创建一个简单的工作流）</h1><div id="post-meta"><div class="meta-firstline"><span class="post-meta-date"><i class="far fa-calendar-alt fa-fw post-meta-icon"></i><span class="post-meta-label">发表于</span><time class="post-meta-date-created" datetime="2021-11-10T15:00:00.000Z" title="发表于 2021-11-10 23:00:00">2021-11-10</time><span class="post-meta-separator">|</span><i class="fas fa-history fa-fw post-meta-icon"></i><span class="post-meta-label">更新于</span><time class="post-meta-date-updated" datetime="2021-11-10T14:53:54.000Z" title="更新于 2021-11-10 22:53:54">2021-11-10</time></span><span class="post-meta-categories"><span class="post-meta-separator">|</span><i class="fas fa-inbox fa-fw post-meta-icon"></i><a class="post-meta-categories" href="/categories/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF%E5%AD%A6/">生物信息学</a></span></div><div class="meta-secondline"><span class="post-meta-separator">|</span><span class="post-meta-wordcount"><i class="far fa-file-word fa-fw post-meta-icon"></i><span class="post-meta-label">字数总计:</span><span class="word-count">4.5k</span><span class="post-meta-separator">|</span><i class="far fa-clock fa-fw post-meta-icon"></i><span class="post-meta-label">阅读时长:</span><span>16分钟</span></span><span class="post-meta-separator">|</span><span class="post-meta-pv-cv"><i class="far fa-eye fa-fw post-meta-icon"></i><span class="post-meta-label">阅读量:</span><span id="busuanzi_value_page_pv"></span></span></div></div></div></header><main class="layout" id="content-inner"><div id="post"><article class="post-content" id="article-container"><h1 id="写在前面">写在前面</h1>
<p>既然写了教程就需要具有普适性，能适合大多数人的胃口，我这部分的内容以及示例主要还是参考了官方教程，但是都是我一步一步跑过的流程，所以会更有印象，送给想学 Snkaemake 但是一直没有去学的朋友们，这些内容对于有生信基础的人来讲，上手会很快，因为很多的生信软件都使用过，写起来也就没有那么晦涩，下面开始～</p>
<h1 id="Snakemake-定义">Snakemake 定义</h1>
<p>Snakemake 工作流管理系统是一种用于创建可重复和可扩展的数据分析的工具。</p>
<p>工作流是通过一种人类可读的、基于 Python 的语言来描述的。它们可以无缝扩展到服务器、集群、grid和云环境，无需修改工作流。</p>
<p>最后，Snakemake 工作流可能需要对所需软件的准备，这些软件将自动部署到任何执行环境。</p>
<h1 id="安装">安装</h1>
<p>Snakemake 可在 PyPi 上以及通过 Bioconda 和源代码获得。可以使用任意方法安装 Snakemake，我们这里仅介绍使用 Conda 安装</p>
<h2 id="通过-Conda-Mamba-安装">通过 Conda/Mamba 安装</h2>
<p>这是安装 Snakemake的<strong>推荐方式</strong>，因为Conda安装比较简单。</p>
<p>首先，必须已经安装了一个基于 Conda 的 Python3 发行版。推荐的选择是<a target="_blank" rel="noopener external nofollow noreferrer" href="https://github.com/conda-forge/miniforge#mambaforge">Mambaforge</a>，它不仅提供所需的 Python 和 Conda 命令，而且包括 <strong><a target="_blank" rel="noopener external nofollow noreferrer" href="https://github.com/mamba-org/mamba">Mamba</a>，它是强烈推荐的Conda包管理器的极其快速和强大的替代品</strong>。默认的 conda 求解器有点慢，有时在选择最新的软件包版本时会出现问题。因此，建议在任何情况下都使用Mamba。</p>
<p>如果不安装 Mambaforge ，也可以直接安装 Mamba</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">conda install -n base -c conda-forge mamba</span><br></pre></td></tr></table></figure>
<h3 id="单独安装到一个小环境">单独安装到一个小环境</h3>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">$ conda activate base</span><br><span class="line">$ mamba create -c conda-forge -c bioconda -n snakemake snakemake</span><br></pre></td></tr></table></figure>
<p>安装到单独环境中比较好，为了避免与其他软件包冲突，并通过如下方式进行激活</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">$ conda activate snakemake</span><br><span class="line">$ snakemake --help</span><br></pre></td></tr></table></figure>
<h3 id="仅安装必备软件的-snakemake-版本">仅安装必备软件的 snakemake 版本</h3>
<p>可以安装仅依赖基本必需软件的 minimal 版本 Snakemake</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">$ conda activate base</span><br><span class="line">$ mamba create -c bioconda -c conda-forge -n snakemake snakemake-minimal</span><br></pre></td></tr></table></figure>
<h1 id="基础示例">基础示例</h1>
<p>首先先激活 Snakemake ，看各自下载的环境，我是单独创建了一个小环境 所以我进行如下操作：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$ conda activate snakemake</span><br></pre></td></tr></table></figure>
<p><strong>Snakemake 工作流是通过在 Snakefile 中指定命令来定义的</strong>。 <strong>命令</strong>通过指定如何从<strong>输入文件集</strong>创建<strong>输出文件集</strong>，<strong>将工作流分解为小步骤</strong>（例如，单个工具的应用）。Snakemake通过匹配文件名自动确定命令之间的<strong>依赖关系</strong>。</p>
<p>下面通过创建示例工作流来介绍 Snakemake 语法。 工作流程来自基因组分析领域。 它将测序reads映射到参考基因组，并在映射reads上调用变体。 本教程不需要您知道这是关于什么的。 尽管如此，我们在以下段落中提供了一些背景知识。</p>
<h2 id="测试数据下载">测试数据下载</h2>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line">git <span class="built_in">clone</span> https://bitbucket.org/snakemake/snakemake-tutorial.git</span><br><span class="line"></span><br><span class="line"><span class="built_in">cd</span> snakemake-tutorial</span><br><span class="line"></span><br><span class="line">├── data</span><br><span class="line">│   ├── genome.fa</span><br><span class="line">│   ├── genome.fa.amb</span><br><span class="line">│   ├── genome.fa.ann</span><br><span class="line">│   ├── genome.fa.bwt</span><br><span class="line">│   ├── genome.fa.fai</span><br><span class="line">│   ├── genome.fa.pac</span><br><span class="line">│   ├── genome.fa.sa</span><br><span class="line">│   └── samples</span><br><span class="line">│       ├── A.fastq</span><br><span class="line">│       ├── B.fastq</span><br><span class="line">│       └── C.fastq</span><br><span class="line"></span><br><span class="line"><span class="comment"># 开启 Snakemake 学习之旅</span></span><br></pre></td></tr></table></figure>
<h2 id="1-Mapping-reads">1. Mapping reads</h2>
<p>第一个 Snakemake 命令将给定样本的reads映射到给定的参考基因组。为此，我们将使用工具<a target="_blank" rel="noopener external nofollow noreferrer" href="http://bio-bwa.sourceforge.net/">bwa</a>，特别是 subcommand 。在工作目录中，创建一个使用您选择的编辑器调用的新文件。官方建议使用<a target="_blank" rel="noopener external nofollow noreferrer" href="https://atom.io/">Atom编辑器</a>，因为它为 Snakemake 提供了开箱即用的语法突出显示。在 Snakefile 中，定义以下命令：<code>bwa mem Snakefile</code></p>
<p>第一个 Snakemake 命令将给定样本的reads映射到给定的参考基因组。 为此，我们将使用工具 <a target="_blank" rel="noopener external nofollow noreferrer" href="http://bio-bwa.sourceforge.net/">bwa</a>，特别是子命令 <code>bwa mem</code>。 在工作目录中，使用编辑器创建一个名为 Snakefile 的新文件。 官方建议使用 <a target="_blank" rel="noopener external nofollow noreferrer" href="https://atom.io/">Atom 编辑器</a>，因为它为 Snakemake 提供了开箱即用的语法突出显示。 在 Snakefile 中，定义以下命令：</p>
<blockquote>
<p>首先创建一个名称为 Snakefile 的文件，并输入我下面的内容</p>
</blockquote>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">rule bwa_map:</span><br><span class="line">    <span class="built_in">input</span>:</span><br><span class="line">        <span class="string">&quot;data/genome.fa&quot;</span>,</span><br><span class="line">        <span class="string">&quot;data/samples/A.fastq&quot;</span></span><br><span class="line">    output:</span><br><span class="line">        <span class="string">&quot;mapped_reads/A.bam&quot;</span></span><br><span class="line">    shell:</span><br><span class="line">        <span class="string">&quot;bwa mem &#123;input&#125; | samtools view -Sb - &gt; &#123;output&#125;&quot;</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 一个常见的错误是忘记输入或输出项之间的逗号。 由于 Python 连接后续字符串，这可能会导致抱错</span></span><br></pre></td></tr></table></figure>
<p>Snakemake rule有一个名称（这里是 <code>bwa_map</code>）和许多指令，这里是 <code>input</code> 、 <code>output</code> 和 <code>shell</code> 。</p>
<p>在 <code>input</code> 和 <code>output</code> 指令之后是那些预计将在命令中使用或创建的文件列表。</p>
<p>在最简单的情况下，这些只是 Python 字符串。 <code>shell</code> 指令后跟一个包含要执行的 shell 命令的 Python 字符串。</p>
<p>在 shell 命令字符串中，我们可以通过**大括号表示法（类似于 Python 格式函数）**引用命令的元素。</p>
<p>在这里，我们通过指定 <code>&#123;output&#125;</code> 来引用<strong>输出文件</strong>，通过指定 <code>&#123;input&#125;</code> 来引用<strong>输入文件</strong>。由于命令有多个输入文件，Snakemake 将连接它们，用<strong>空格</strong>分隔。换句话说，Snakemake 会在执行命令之前将 <code>&#123;input&#125;</code> 替换为 <code>data/genome.fa data/samples/A.fastq</code>。</p>
<p>shell 命令使用参考基因​​组和reads调用 <code>bwa mem</code>，并将输出通过管道传输到 <strong>samtools</strong>，后者创建包含比对的压缩 <a target="_blank" rel="noopener external nofollow noreferrer" href="https://en.wikipedia.org/wiki/Binary_Alignment_Map"><em>BAM</em></a> 文件。 samtools 的输出被重定向到命令定义的输出文件中，并带有 <code>&gt;</code>。</p>
<blockquote>
<p>执行工作流时，Snakemake 会尝试生成给定的目标文件。可以通过命令行指定目标文件。通过执行</p>
</blockquote>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 试运行</span></span><br><span class="line">$ snakemake -np mapped_reads/A.bam</span><br></pre></td></tr></table></figure>
<p>在包含 <strong>Snakefile</strong> 的工作目录中， Snakemake 生成目标文件 <code>mapping_reads/A.bam</code>。 由于我们使用了 <code>-n</code>（或 <code>--dry-run</code>）标志，Snakemake 将<strong>只显示执行计划而不是实际执行步骤</strong>。 <code>-p</code> 标志指示 Snakemake 打印出生成的 shell 命令以供说明。</p>
<p>为了生成目标文件，Snakemake 以自上而下的方式应用 Snakefile 中给出的命令。 应用命令来生成一组输出文件称为作业。 对于作业的每个输入文件，Snakemake 再次（即递归地）确定可应用于生成它的命令。 这产生了作业的有向无环图 (DAG)，其中边代表依赖关系。 到目前为止，我们只有一个命令，作业的 DAG 由单个节点组成。 尽管如此，我们可以执行我们的工作流程</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$ snakemake --cores 1 mapped_reads&#x2F;A.bam</span><br></pre></td></tr></table></figure>
<p><strong>无论何时执行工作流，都需要指定要使用的核心数</strong>。 对于本教程，现在将使用单个内核。 稍后介绍并行化是如何工作的。 完成上述命令后，Snakemake 将不会再次尝试创建<code>mapped_reads/A.bam</code>，因为它已经存在于文件系统中。 Snakemake <strong>仅在输入文件之一比输出文件之一新 或 输入文件之一将被另一个作业更新时重新运行作业</strong>。</p>
<h2 id="2-Generalizing-the-read-mapping-rule">2. Generalizing the read mapping rule</h2>
<p>前面的rule仅适用于在文件 <code>data/samples/A.fastq</code> 中读取。 但是，Snakemake <strong>允许使用命名通配符</strong>。 只需用通配符 <code>&#123;sample&#125;</code> 替换第二个输入文件和输出文件中的 <code>A</code>，即可批量读取～ 举例：</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">rule bwa_map:</span><br><span class="line">    <span class="built_in">input</span>:</span><br><span class="line">        <span class="string">&quot;data/genome.fa&quot;</span>,</span><br><span class="line">        <span class="string">&quot;data/samples/&#123;sample&#125;.fastq&quot;</span></span><br><span class="line">    output:</span><br><span class="line">        <span class="string">&quot;mapped_reads/&#123;sample&#125;.bam&quot;</span></span><br><span class="line">    shell:</span><br><span class="line">        <span class="string">&quot;bwa mem &#123;input&#125; | samtools view -Sb - &gt; &#123;output&#125;&quot;</span></span><br></pre></td></tr></table></figure>
<p>当 Snakemake 通过用适当的值替换输出文件中的通配符 <code>&#123;sample&#125;</code> 确定可以应用此命令来生成目标文件时，它将该值传播到输入文件中所有出现的 <code>&#123;sample&#125;</code>，从而确定 结果工作的必要输入。</p>
<blockquote>
<p>注意，您的文件路径中可以有多个通配符，但是，为了避免与同一命令的其他作业发生冲突，命令的<strong>所有输出文件必须包含完全相同的通配符</strong>。</p>
</blockquote>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">$ snakemake -np mapped_reads/B.bam</span><br><span class="line"></span><br><span class="line"><span class="comment"># 运行之后输出</span></span><br><span class="line">rule bwa_map:</span><br><span class="line">    input: data/genome.fa, data/samples/B.fastq</span><br><span class="line">    output: mapped_reads/B.bam</span><br><span class="line">    jobid: 0</span><br><span class="line">    wildcards: sample=B</span><br><span class="line">    resources: tmpdir=/tmp</span><br><span class="line"><span class="comment"># 可以看到内容随着输入命令变化匹配到了B.bam</span></span><br></pre></td></tr></table></figure>
<p>Snakemake 将通过将通配符 <code>&#123;sample&#125;</code> 替换为值 <code>B</code> 来确定可以应用命令 <code>bwa_map</code> 来生成目标文件。在试运行的输出中，可以看到通配符值如何传播到输入文件和 shell 命令中的所有文件名。 还可以<strong>指定多个目标</strong>，例如：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$ snakemake -np mapped_reads&#x2F;A.bam mapped_reads&#x2F;B.bam</span><br></pre></td></tr></table></figure>
<p>一些<a target="_blank" rel="noopener external nofollow noreferrer" href="https://tldp.org/LDP/Bash-Beginners-Guide/html/">Bash</a>语法特别方便。例如，可以选择在一次组合多个目标</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">$ snakemake -np mapped_reads/&#123;A,B&#125;.bam</span><br><span class="line"><span class="comment"># Bash 只是将其大括号扩展应用于集合 &#123;A,B&#125;，为每个元素创建给定的路径并用空格分隔结果路径。</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># snakemake -np mapped_reads/&#123;1..10&#125;.bam</span></span><br><span class="line"><span class="comment"># 会匹配1.bam; 2.bam; ... ; 10.bam </span></span><br></pre></td></tr></table></figure>
<p>在这两种情况下， Snakemake 只创建输出文件 <code>mapping_reads/B.bam</code>。</p>
<p>这是因为之前已经执行过 <code>mapping_reads/A.bam</code> 并且没有比输出文件更新的输入文件。</p>
<blockquote>
<p>可以更新输入文件<code>data/samples/A.fastq</code>的文件修改日期</p>
</blockquote>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$ touch data&#x2F;samples&#x2F;A.fastq</span><br></pre></td></tr></table></figure>
<p>并运行 Snakemake 重新运行作业来创建文件 <code>mapping_reads/A.bam</code></p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$ snakemake -np mapped_reads&#x2F;A.bam mapped_reads&#x2F;B.bam</span><br></pre></td></tr></table></figure>
<h2 id="3-Sorting-read-alignments">3. Sorting read alignments</h2>
<p>对于后面的步骤，我们需要对 <code>BAM</code> 文件中的读取对齐进行排序。这可以通过 <a target="_blank" rel="noopener external nofollow noreferrer" href="https://www.htslib.org/">samtools</a> <code>sort</code> 命令实现。我们在 <code>bwa_map</code> rule下添加以下rule：</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">rule samtools_sort:</span><br><span class="line">    <span class="built_in">input</span>:</span><br><span class="line">        <span class="string">&quot;mapped_reads/&#123;sample&#125;.bam&quot;</span></span><br><span class="line">    output:</span><br><span class="line">        <span class="string">&quot;sorted_reads/&#123;sample&#125;.bam&quot;</span></span><br><span class="line">    shell:</span><br><span class="line">        <span class="string">&quot;samtools sort -T sorted_reads/&#123;wildcards.sample&#125; &quot;</span></span><br><span class="line">        <span class="string">&quot;-O bam &#123;input&#125; &gt; &#123;output&#125;&quot;</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment"># 在上面的 shell 命令中，我们将字符串分成两行，但是 Python 会自动将它们连接成一行。</span></span><br><span class="line"><span class="comment"># 分行写的话可以避免 shell 命令行过长。使用它时，需要在每行但最后一行中都有一个尾随空格，以避免参数无法正确分隔。</span></span><br></pre></td></tr></table></figure>
<p>此命令将从<code>mapped_reads</code> 文件夹中获取<strong>输入文件</strong>，并将排序后的版本存储在<code>sorted_reads</code> 目录中。</p>
<p>注意，Snakemake 会在作业执行前自动创建丢失的目录。 对于排序，samtools 需要使用标志 <code>-T</code> 指定的前缀。</p>
<p>在这里，我们需要通配符<code>sample</code>的值。 Snakemake 允许通过 <code>wildcards</code> 对象访问 shell 命令中的通配符，该对象具有带有每个通配符值的属性。</p>
<blockquote>
<p>wildcards指通配符，学过类 LINUX 系统的，应该都知道什么是通配符。<br>
<code>*</code>  代表任意多个字符<br>
<code>?</code>  代表任意单个字符<br>
<code>[ ]</code> 代表“[”和“]”之间的某一个字符，比如[0-9]可以代表0-9之间的任意一个数字，[a-zA-Z]可以代表a-z和A-Z之间的任意一个字母，字母区分大小写<br>
<code>–</code> 代表一个字符<br>
<code>~</code> 用户的根目录</p>
</blockquote>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$ snakemake -np sorted_reads&#x2F;B.bam</span><br></pre></td></tr></table></figure>
<p>看到 Snakemake 首先运行<code>bwa_map</code>，然后运行<code>samtools_sort</code>来创建所需的目标文件：如前所述，依赖项通过匹配文件名自动解析。</p>
<h2 id="4-Indexing-read-alignments-and-visualizing-the-DAG-of-jobs">4. Indexing read alignments and visualizing the DAG of jobs</h2>
<p>接下来，我们需要再次使用 samtools 来索引已排序的读取比对，以便我们可以通过它们映射到的基因组位置快速访问读取。这可以通过以下命令来完成：</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">rule samtools_index:</span><br><span class="line">    <span class="built_in">input</span>:</span><br><span class="line">        <span class="string">&quot;sorted_reads/&#123;sample&#125;.bam&quot;</span></span><br><span class="line">    output:</span><br><span class="line">        <span class="string">&quot;sorted_reads/&#123;sample&#125;.bam.bai&quot;</span></span><br><span class="line">    shell:</span><br><span class="line">        <span class="string">&quot;samtools index &#123;input&#125;&quot;</span></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<blockquote>
<p>Snakemake 使用Python 格式<a target="_blank" rel="noopener external nofollow noreferrer" href="https://docs.python.org/3/library/string.html#formatexamples">mini language</a>来格式化 shell 命令。在 shell 命令中使用大括号 (<code>&#123;&#125;</code>) 来表示其他内容。在这种情况下，必须加倍对我们上面提到的是bash括号扩展依靠时逃避它们，<br>
例如：<br>
<code>ls &#123;&#123;A,B&#125;&#125;.txt</code></p>
</blockquote>
<p>已经完成了三个步骤，现在可以查看生成的有向无环图 (DAG)</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$ snakemake --dag sorted_reads/&#123;A,B&#125;.bam.bai | dot -Tsvg &gt; dag.svg</span><br></pre></td></tr></table></figure>
<p>Snakemake 使用 <a target="_blank" rel="noopener external nofollow noreferrer" href="https://www.graphviz.org/">Graphviz</a> 提供的 <code>dot</code> 命令创建 DAG 的可视化。 对于给定的目标文件，Snakemake 以 dot 语言指定 DAG 并将其通过管道传输到 <code>dot</code> 命令，该命令将定义呈现为 <a target="_blank" rel="noopener external nofollow noreferrer" href="https://en.wikipedia.org/wiki/Scalable_Vector_Graphics">SVG 格式</a>。 渲染的 DAG 通过管道传输到文件 <code>dag.svg</code> 中，如下所示：</p>
<p><img src="https://z3.ax1x.com/2021/11/10/Idea2d.png" alt></p>
<h2 id="5-Calling-genomic-variants">5. Calling genomic variants</h2>
<p>我们工作流程的下一步将聚合所有样本的映射reads，并共同调用它们的基因组变异。 对于变体调用，我们将结合两个实用程序 <a target="_blank" rel="noopener external nofollow noreferrer" href="https://www.htslib.org/">samtools</a> 和 <a target="_blank" rel="noopener external nofollow noreferrer" href="https://www.htslib.org/">bcftools</a>。 Snakemake 提供了一个辅助函数来收集输入文件，帮助我们描述这一步中的聚合。</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">expand(<span class="string">&quot;sorted_reads/&#123;sample&#125;.bam&quot;</span>, sample=SAMPLES)</span><br></pre></td></tr></table></figure>
<p>获取文件列表，其中给定模式<code>sorted_reads/&#123;sample&#125;.bam</code>被格式化为给定样本列表<code>SAMPLES</code>中的值</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">[<span class="string">&quot;sorted_reads/A.bam&quot;</span>, <span class="string">&quot;sorted_reads/B.bam&quot;</span>]</span><br></pre></td></tr></table></figure>
<p>当模式包含多个通配符时</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">expand(<span class="string">&quot;sorted_reads/&#123;sample&#125;.&#123;replicate&#125;.bam&quot;</span>, sample=SAMPLES, replicate=[<span class="number">0</span>, <span class="number">1</span>])</span><br></pre></td></tr></table></figure>
<p>将创建 <code>SAMPLES</code> 的所有元素和列表 <code>[0, 1]</code> 的乘积</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">[<span class="string">&quot;sorted_reads/A.0.bam&quot;</span>, <span class="string">&quot;sorted_reads/A.1.bam&quot;</span>, <span class="string">&quot;sorted_reads/B.0.bam&quot;</span>, <span class="string">&quot;sorted_reads/B.1.bam&quot;</span>]</span><br></pre></td></tr></table></figure>
<p>在这里，仅使用<code>expand</code>这个简单的例子。<br>
首先让 Snakemake 知道我们要考虑哪些样本。<br>
Snakemake 从请求的输出反向工作，而不是从可用的输入反向工作。 因此，它不会自动推断所有可能的输出，例如数据文件夹中的 fastq 文件。</p>
<p>Snakefiles 原则上是 Python 代码，通过一些声明性语句来定义工作流。 因此，我们可以在 Snakefile 顶部的纯 Python 中临时定义示例列表：</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SAMPLES = [<span class="string">&quot;A&quot;</span>, <span class="string">&quot;B&quot;</span>]</span><br></pre></td></tr></table></figure>
<p>可以将以上命令添加到 Snakefile 中：</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">rule bcftools_call:</span><br><span class="line">    <span class="built_in">input</span>:</span><br><span class="line">        fa=<span class="string">&quot;data/genome.fa&quot;</span>,</span><br><span class="line">        bam=expand(<span class="string">&quot;sorted_reads/&#123;sample&#125;.bam&quot;</span>, sample=SAMPLES),</span><br><span class="line">        bai=expand(<span class="string">&quot;sorted_reads/&#123;sample&#125;.bam.bai&quot;</span>, sample=SAMPLES)</span><br><span class="line">    output:</span><br><span class="line">        <span class="string">&quot;calls/all.vcf&quot;</span></span><br><span class="line">    shell:</span><br><span class="line">        <span class="string">&quot;samtools mpileup -g -f &#123;input.fa&#125; &#123;input.bam&#125; | &quot;</span></span><br><span class="line">        <span class="string">&quot;bcftools call -mv - &gt; &#123;output&#125;&quot;</span></span><br></pre></td></tr></table></figure>
<p>对于<strong>多个输入或输出文件</strong>，有时在 shell 命令中分别引用它们会很方便。 这可以通过指定输入或输出文件的名称来完成，例如使用<code>fa=....</code>然后可以在shell命令中通过名称引用这些文件，例如使用<code>&#123;input.fa&#125;</code>。</p>
<p>对于像这样的长 shell 命令，建议将字符串拆分为多个缩进行。 Python 会自动将其合二为一。 此外，您会注意到输入或输出文件列表可以包含任意 Python 语句，只要它返回一个字符串或字符串列表。 在这里，我们调用我们的 <code>expand</code> 函数来聚合所有样本的对齐reads。</p>
<h2 id="6-Using-custom-scripts">6. Using custom scripts</h2>
<p>通常，工作流不仅包括调用各种工具，还包含<strong>自定义代码</strong>，例如计算汇总统计或创建绘图。虽然 Snakemake 还允许您直接在命令中编写 Python 代码。为此，Snakemake 提供了 <code>script</code> 指令。将以下规则添加到您的 Snakefile 中：</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">rule plot_quals:</span><br><span class="line">    <span class="built_in">input</span>:</span><br><span class="line">        <span class="string">&quot;calls/all.vcf&quot;</span></span><br><span class="line">    output:</span><br><span class="line">        <span class="string">&quot;plots/quals.svg&quot;</span></span><br><span class="line">    script:</span><br><span class="line">        <span class="string">&quot;scripts/plot-quals.py&quot;</span></span><br></pre></td></tr></table></figure>
<p>使用此规则，我们最终将生成已分配给文件 <code>calls/all.vcf</code> 中的variant calls的质量分数的直方图。 生成绘图的实际 Python 代码在脚本 <code>scripts/plot-quals.py</code> 中。 在脚本中，命令的所有属性，如 <code>input</code>、<code>output</code>、<code>wildcards</code> 等，都可以作为全局 <code>snakemake</code> 对象的属性使用。 创建文件 <code>scripts/plot-quals.py</code></p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">import</span> matplotlib</span><br><span class="line">matplotlib.use(<span class="string">&quot;Agg&quot;</span>)</span><br><span class="line"><span class="keyword">import</span> matplotlib.pyplot <span class="keyword">as</span> plt</span><br><span class="line"><span class="keyword">from</span> pysam <span class="keyword">import</span> VariantFile</span><br><span class="line"></span><br><span class="line">quals = [record.qual <span class="keyword">for</span> record <span class="keyword">in</span> VariantFile(snakemake.<span class="built_in">input</span>[<span class="number">0</span>])]</span><br><span class="line">plt.hist(quals)</span><br><span class="line"></span><br><span class="line">plt.savefig(snakemake.output[<span class="number">0</span>])</span><br></pre></td></tr></table></figure>
<p>除了Python脚本之外，还可以使用R脚本;有关详细信息和示例，可以阅读官方教程中的<a target="_blank" rel="noopener external nofollow noreferrer" href="https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#snakefiles-external-scripts">外部脚本部分</a></p>
<h2 id="7-Adding-a-target-rule">7. Adding a target rule</h2>
<p>到目前为止，我们总是通过在命令行指定目标文件来执行工作流。除了文件名，如果请求的规则没有通配符，Snakemake还接受规则名作为目标。</p>
<p>因此，可以编写目标规则来收集所需结果或所有结果的特定子集。此外，<strong>如果命令行中没有给出目标，Snakemake会将 Snakefile 的第一条规则定义为目标</strong>。因此，最好的做法是在工作流的顶部有一个规则<code>all</code>，该rule将所有通常需要的目标文件作为输入文件</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">rule <span class="built_in">all</span>:</span><br><span class="line">    <span class="built_in">input</span>:</span><br><span class="line">        <span class="string">&quot;plots/quals.svg&quot;</span></span><br></pre></td></tr></table></figure>
<p>把这个rule添加到我们工作流程的顶部。执行 Snakemake 时</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">$ snakemake -n</span><br><span class="line"></span><br><span class="line"><span class="comment"># 可以在 Snakefile 的顶部添加多个目标rules。虽然 Snakemake 将默认执行第一个，但可以通过命令行（例如，snakemake -n mytarget）定位其中的任何一个。</span></span><br></pre></td></tr></table></figure>
<p>执行此命令将显示用于创建文件的执行计划<code>plots/quals.svg</code>，其中包含并总结了我们所有的结果。</p>
<p>除了 Snakemake 将工作流的第一条规则视为默认目标之外，Snakefile 中的规则顺序是任意的，不会影响作业的 DAG。</p>
<h1 id="总结">总结</h1>
<p>生成的工作流程如下所示：</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br></pre></td><td class="code"><pre><span class="line">SAMPLES = [<span class="string">&quot;A&quot;</span>, <span class="string">&quot;B&quot;</span>]</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">rule <span class="built_in">all</span>:</span><br><span class="line">    <span class="built_in">input</span>:</span><br><span class="line">        <span class="string">&quot;plots/quals.svg&quot;</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line">rule bwa_map:</span><br><span class="line">    <span class="built_in">input</span>:</span><br><span class="line">        <span class="string">&quot;data/genome.fa&quot;</span>,</span><br><span class="line">        <span class="string">&quot;data/samples/&#123;sample&#125;.fastq&quot;</span></span><br><span class="line">    output:</span><br><span class="line">        <span class="string">&quot;mapped_reads/&#123;sample&#125;.bam&quot;</span></span><br><span class="line">    shell:</span><br><span class="line">        <span class="string">&quot;bwa mem &#123;input&#125; | samtools view -Sb - &gt; &#123;output&#125;&quot;</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line">rule samtools_sort:</span><br><span class="line">    <span class="built_in">input</span>:</span><br><span class="line">        <span class="string">&quot;mapped_reads/&#123;sample&#125;.bam&quot;</span></span><br><span class="line">    output:</span><br><span class="line">        <span class="string">&quot;sorted_reads/&#123;sample&#125;.bam&quot;</span></span><br><span class="line">    shell:</span><br><span class="line">        <span class="string">&quot;samtools sort -T sorted_reads/&#123;wildcards.sample&#125; &quot;</span></span><br><span class="line">        <span class="string">&quot;-O bam &#123;input&#125; &gt; &#123;output&#125;&quot;</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line">rule samtools_index:</span><br><span class="line">    <span class="built_in">input</span>:</span><br><span class="line">        <span class="string">&quot;sorted_reads/&#123;sample&#125;.bam&quot;</span></span><br><span class="line">    output:</span><br><span class="line">        <span class="string">&quot;sorted_reads/&#123;sample&#125;.bam.bai&quot;</span></span><br><span class="line">    shell:</span><br><span class="line">        <span class="string">&quot;samtools index &#123;input&#125;&quot;</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line">rule bcftools_call:</span><br><span class="line">    <span class="built_in">input</span>:</span><br><span class="line">        fa=<span class="string">&quot;data/genome.fa&quot;</span>,</span><br><span class="line">        bam=expand(<span class="string">&quot;sorted_reads/&#123;sample&#125;.bam&quot;</span>, sample=SAMPLES),</span><br><span class="line">        bai=expand(<span class="string">&quot;sorted_reads/&#123;sample&#125;.bam.bai&quot;</span>, sample=SAMPLES)</span><br><span class="line">    output:</span><br><span class="line">        <span class="string">&quot;calls/all.vcf&quot;</span></span><br><span class="line">    shell:</span><br><span class="line">        <span class="string">&quot;samtools mpileup -g -f &#123;input.fa&#125; &#123;input.bam&#125; | &quot;</span></span><br><span class="line">        <span class="string">&quot;bcftools call -mv - &gt; &#123;output&#125;&quot;</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line">rule plot_quals:</span><br><span class="line">    <span class="built_in">input</span>:</span><br><span class="line">        <span class="string">&quot;calls/all.vcf&quot;</span></span><br><span class="line">    output:</span><br><span class="line">        <span class="string">&quot;plots/quals.svg&quot;</span></span><br><span class="line">    script:</span><br><span class="line">        <span class="string">&quot;scripts/plot-quals.py&quot;</span></span><br><span class="line"></span><br></pre></td></tr></table></figure>
</article><div class="post-copyright"><div class="post-copyright__author"><span class="post-copyright-meta">文章作者: </span><span class="post-copyright-info"><a href="mailto:lixingzee@gmail.com" rel="external nofollow noreferrer">李星泽</a></span></div><div class="post-copyright__type"><span class="post-copyright-meta">文章链接: </span><span class="post-copyright-info"><a href="http://lxz9.com/2021/11/10/Snakemake-1/">http://lxz9.com/2021/11/10/Snakemake-1/</a></span></div><div class="post-copyright__notice"><span class="post-copyright-meta">版权声明: </span><span class="post-copyright-info">本博客所有文章除特别声明外，均采用 <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" rel="external nofollow noreferrer" target="_blank">CC BY-NC-SA 4.0</a> 许可协议。转载请注明来自 <a href="http://lxz9.com" target="_blank">生信技术</a>！</span></div></div><div class="tag_share"><div class="post-meta__tag-list"><a class="post-meta__tags" href="/tags/snakemake/">snakemake</a></div><div class="post_share"><div class="social-share" data-image="https://z3.ax1x.com/2021/11/10/Idni7V.png" data-sites="facebook,twitter,wechat,weibo,qq"></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/social-share.js/dist/css/share.min.css" media="print" onload="this.media='all'"><script src="https://cdn.jsdelivr.net/npm/social-share.js/dist/js/social-share.min.js" defer></script></div></div><div class="post-reward"><div class="reward-button"><i class="fas fa-qrcode"></i> 打赏<div class="reward-main"><ul class="reward-all"><li class="reward-item"><a href="https://z3.ax1x.com/2021/01/19/sgZ5y6.jpg" rel="external nofollow noreferrer" target="_blank"><img class="post-qr-code-img" src="https://z3.ax1x.com/2021/01/19/sgZ5y6.jpg" alt="微信"/></a><div class="post-qr-code-desc">微信</div></li><li class="reward-item"><a href="https://z3.ax1x.com/2021/01/19/sgZIOK.jpg" rel="external nofollow noreferrer" target="_blank"><img class="post-qr-code-img" src="https://z3.ax1x.com/2021/01/19/sgZIOK.jpg" alt="支付宝"/></a><div class="post-qr-code-desc">支付宝</div></li></ul></div></div></div><nav class="pagination-post" id="pagination"><div class="prev-post pull-left"><a href="/2021/11/12/Snakemake-2/"><img class="prev-cover" src="https://z3.ax1x.com/2021/11/10/Idni7V.png" onerror="onerror=null;src='/img/404.jpg'" alt="cover of previous post"><div class="pagination-info"><div class="label">上一篇</div><div class="prev_info">Snakemake 常用参数以及进阶用法介绍</div></div></a></div><div class="next-post pull-right"><a href="/2021/11/05/bioinformatics/"><img class="next-cover" src="https://z3.ax1x.com/2021/11/05/InNi1f.jpg" onerror="onerror=null;src='/img/404.jpg'" alt="cover of next post"><div class="pagination-info"><div class="label">下一篇</div><div class="next_info">学习生物信息学的十条准则</div></div></a></div></nav><div class="relatedPosts"><div class="headline"><i class="fas fa-thumbs-up fa-fw"></i><span> 相关推荐</span></div><div class="relatedPosts-list"><div><a href="/2021/11/12/Snakemake-2/" title="Snakemake 常用参数以及进阶用法介绍"><img class="cover" src="https://z3.ax1x.com/2021/11/10/Idni7V.png" alt="cover"><div class="content is-center"><div class="date"><i class="far fa-calendar-alt fa-fw"></i> 2021-11-12</div><div class="title">Snakemake 常用参数以及进阶用法介绍</div></div></a></div></div></div><hr/><div id="post-comment"><div class="comment-head"><div class="comment-headline"><i class="fas fa-comments fa-fw"></i><span> 评论</span></div></div><div class="comment-wrap"><div><div class="vcomment" id="vcomment"></div></div></div></div></div><div class="aside-content" id="aside-content"><div class="card-widget card-info"><div class="card-info-avatar is-center"><img class="avatar-img" src="https://z3.ax1x.com/2021/03/10/6JSyoq.png" onerror="this.onerror=null;this.src='/img/friend_404.gif'" alt="avatar"/><div class="author-info__name">李星泽</div><div class="author-info__description"></div></div><div class="card-info-data"><div class="card-info-data-item is-center"><a href="/archives/"><div class="headline">文章</div><div class="length-num">31</div></a></div><div class="card-info-data-item is-center"><a href="/tags/"><div class="headline">标签</div><div class="length-num">45</div></a></div><div class="card-info-data-item is-center"><a href="/categories/"><div class="headline">分类</div><div class="length-num">13</div></a></div></div><a class="button--animated" id="card-info-btn" target="_blank" rel="noopener external nofollow noreferrer" href="https://github.com/xxxxxx"><i class="fab fa-github"></i><span>Follow Me</span></a><div class="card-info-social-icons is-center"><a class="social-icon" href="https://blog.csdn.net/m0_49960764" rel="external nofollow noreferrer" target="_blank" title="CSDN"><i class="iconfont icon-csdn"></i></a><a class="social-icon" href="https://z3.ax1x.com/2021/11/25/oAI0h9.jpg" rel="external nofollow noreferrer" target="_blank" title="微信公众号"><i class="iconfont icon-weixin"></i></a><a class="social-icon" href="https://space.bilibili.com/113291922?from=search&amp;seid=3526734520747213504&amp;spm_id_from=333.337.0.0" rel="external nofollow noreferrer" target="_blank" title="bilibili"><i class="iconfont icon-bilibili-copy"></i></a><a class="social-icon" href="mailto:lixingzee@gmail.com" rel="external nofollow noreferrer" target="_blank" title="Email"><i class="fas fa-envelope"></i></a></div></div><div class="card-widget card-announcement"><div class="item-headline"><i class="fas fa-bullhorn card-announcement-animation"></i><span>公告</span></div><div class="announcement_content">如果你有更多想了解的内容，欢迎关注我的个人公众号：<b>生信技术</b><br> <br> <br></div><timing></timing></div><div class="sticky_layout"><div class="card-widget" id="card-toc"><div class="item-headline"><i class="fas fa-stream"></i><span>目录</span></div><div class="toc-content"><ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#%E5%86%99%E5%9C%A8%E5%89%8D%E9%9D%A2"><span class="toc-number">1.</span> <span class="toc-text">写在前面</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#Snakemake-%E5%AE%9A%E4%B9%89"><span class="toc-number">2.</span> <span class="toc-text">Snakemake 定义</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E5%AE%89%E8%A3%85"><span class="toc-number">3.</span> <span class="toc-text">安装</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#%E9%80%9A%E8%BF%87-Conda-Mamba-%E5%AE%89%E8%A3%85"><span class="toc-number">3.1.</span> <span class="toc-text">通过 Conda&#x2F;Mamba 安装</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#%E5%8D%95%E7%8B%AC%E5%AE%89%E8%A3%85%E5%88%B0%E4%B8%80%E4%B8%AA%E5%B0%8F%E7%8E%AF%E5%A2%83"><span class="toc-number">3.1.1.</span> <span class="toc-text">单独安装到一个小环境</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E4%BB%85%E5%AE%89%E8%A3%85%E5%BF%85%E5%A4%87%E8%BD%AF%E4%BB%B6%E7%9A%84-snakemake-%E7%89%88%E6%9C%AC"><span class="toc-number">3.1.2.</span> <span class="toc-text">仅安装必备软件的 snakemake 版本</span></a></li></ol></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E5%9F%BA%E7%A1%80%E7%A4%BA%E4%BE%8B"><span class="toc-number">4.</span> <span class="toc-text">基础示例</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#%E6%B5%8B%E8%AF%95%E6%95%B0%E6%8D%AE%E4%B8%8B%E8%BD%BD"><span class="toc-number">4.1.</span> <span class="toc-text">测试数据下载</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#1-Mapping-reads"><span class="toc-number">4.2.</span> <span class="toc-text">1. Mapping reads</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#2-Generalizing-the-read-mapping-rule"><span class="toc-number">4.3.</span> <span class="toc-text">2. Generalizing the read mapping rule</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#3-Sorting-read-alignments"><span class="toc-number">4.4.</span> <span class="toc-text">3. Sorting read alignments</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#4-Indexing-read-alignments-and-visualizing-the-DAG-of-jobs"><span class="toc-number">4.5.</span> <span class="toc-text">4. Indexing read alignments and visualizing the DAG of jobs</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#5-Calling-genomic-variants"><span class="toc-number">4.6.</span> <span class="toc-text">5. Calling genomic variants</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#6-Using-custom-scripts"><span class="toc-number">4.7.</span> <span class="toc-text">6. Using custom scripts</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#7-Adding-a-target-rule"><span class="toc-number">4.8.</span> <span class="toc-text">7. Adding a target rule</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E6%80%BB%E7%BB%93"><span class="toc-number">5.</span> <span class="toc-text">总结</span></a></li></ol></div></div><div class="card-widget card-recent-post"><div class="item-headline"><i class="fas fa-history"></i><span>最新文章</span></div><div class="aside-list"><div class="aside-list-item"><a class="thumbnail" href="/2021/12/15/Conserved_Sequences/" title="保守序列的生物学意义｜用 K-mers 寻找保守序列"><img src="https://s4.ax1x.com/2021/12/15/ozGUE9.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="保守序列的生物学意义｜用 K-mers 寻找保守序列"/></a><div class="content"><a class="title" href="/2021/12/15/Conserved_Sequences/" title="保守序列的生物学意义｜用 K-mers 寻找保守序列">保守序列的生物学意义｜用 K-mers 寻找保守序列</a><time datetime="2021-12-15T10:00:00.000Z" title="发表于 2021-12-15 18:00:00">2021-12-15</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2021/12/14/R/" title="R语言基础画图教程合集(长文总结)"><img src="https://s4.ax1x.com/2021/12/15/ozG2Ed.jpg" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="R语言基础画图教程合集(长文总结)"/></a><div class="content"><a class="title" href="/2021/12/14/R/" title="R语言基础画图教程合集(长文总结)">R语言基础画图教程合集(长文总结)</a><time datetime="2021-12-14T10:00:00.000Z" title="发表于 2021-12-14 18:00:00">2021-12-14</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2021/12/01/Protein_structure_prediction/" title="使用同源建模预测蛋白质结构"><img src="https://z3.ax1x.com/2021/12/02/oYdAsA.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="使用同源建模预测蛋白质结构"/></a><div class="content"><a class="title" href="/2021/12/01/Protein_structure_prediction/" title="使用同源建模预测蛋白质结构">使用同源建模预测蛋白质结构</a><time datetime="2021-12-01T15:00:00.000Z" title="发表于 2021-12-01 23:00:00">2021-12-01</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2021/11/25/python/" title="Python 学习教程总结"><img src="https://z3.ax1x.com/2021/11/26/oElpDJ.jpg" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="Python 学习教程总结"/></a><div class="content"><a class="title" href="/2021/11/25/python/" title="Python 学习教程总结">Python 学习教程总结</a><time datetime="2021-11-24T16:00:00.000Z" title="发表于 2021-11-25 00:00:00">2021-11-25</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2021/11/23/Gene_Expression/" title="基因表达数据的聚类分析方法"><img src="https://z3.ax1x.com/2021/11/23/opHDQU.jpg" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="基因表达数据的聚类分析方法"/></a><div class="content"><a class="title" href="/2021/11/23/Gene_Expression/" title="基因表达数据的聚类分析方法">基因表达数据的聚类分析方法</a><time datetime="2021-11-23T10:00:00.000Z" title="发表于 2021-11-23 18:00:00">2021-11-23</time></div></div></div></div></div></div></main><footer id="footer"><div id="footer-wrap"><div class="copyright">&copy;2020 - 2021&nbsp;<i style="color:#FF6A6A;animation: announ_animation 0.8s linear infinite;"class="fas fa-heart"></i> By 李星泽</div></div><div class="container" id="jsi-flying-fish-container"><script src="/js/fish.js"></script></div><style>   @media only screen and (max-width: 767px){
   #sidebar_search_box input[type=text]{width:calc(100% - 24px)}
}</style></footer></div><div id="rightside"><div id="rightside-config-hide"><button id="readmode" type="button" title="阅读模式"><i class="fas fa-book-open"></i></button><button id="darkmode" type="button" title="浅色和深色模式转换"><i class="fas fa-adjust"></i></button><button id="hide-aside-btn" type="button" title="单栏和双栏切换"><i class="fas fa-arrows-alt-h"></i></button></div><div id="rightside-config-show"><button id="rightside_config" type="button" title="设置"><i class="fas fa-cog fa-spin"></i></button><button class="close" id="mobile-toc-button" type="button" title="目录"><i class="fas fa-list-ul"></i></button><a id="to_comment" href="#post-comment" title="直达评论"><i class="fas fa-comments"></i></a><button id="go-up" type="button" title="回到顶部"><i class="fas fa-arrow-up"></i></button></div></div><div id="local-search"><div class="search-dialog"><div class="search-dialog__title" id="local-search-title">本地搜索</div><div id="local-input-panel"><div id="local-search-input"><div class="local-search-box"><input class="local-search-box--input" placeholder="搜索文章" type="text"/></div></div></div><hr/><div id="local-search-results"></div><span class="search-close-button"><i class="fas fa-times"></i></span></div><div id="search-mask"></div></div><div><script src="/js/utils.js"></script><script src="/js/main.js"></script><script src="https://cdn.jsdelivr.net/npm/medium-zoom/dist/medium-zoom.min.js"></script><script src="/js/search/local-search.js"></script><div class="js-pjax"><script>function loadValine () {
  function initValine () {
    let initData = {
      el: '#vcomment',
      appId: '',
      appKey: '',
      placeholder: 'Please leave your footprints',
      avatar: 'monsterid',
      meta: 'nick,mail,link'.split(','),
      pageSize: '10',
      lang: 'en',
      recordIP: false,
      serverURLs: '',
      emojiCDN: '',
      emojiMaps: "",
      enableQQ: false,
      path: window.location.pathname,
    }

    if (true) { 
      initData.requiredFields= ('nick,mail'.split(','))
    }
    
    if (false) {
      const otherData = false
      initData = Object.assign(initData, otherData)
    }
    
    const valine = new Valine(initData)
  }

  if (typeof Valine === 'function') initValine() 
  else getScript('https://cdn.jsdelivr.net/npm/valine/dist/Valine.min.js').then(initValine)
}

if ('Valine' === 'Valine' || !true) {
  if (true) btf.loadComment(document.getElementById('vcomment'),loadValine)
  else setTimeout(loadValine, 0)
} else {
  function loadOtherComment () {
    loadValine()
  }
}</script><script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script></div><script src="/js/jquery.min.js"></script><script src="/js/fish.js"></script><script src="/js/timing.js"></script><script async src="https://cdn.jsdelivr.net/npm/aplayer"></script><script async src="https://cdn.jsdelivr.net/npm/meting/dist/Meting.min.js"></script><div class="aplayer no-destroy" data-id="000PeZCQ1i4XVs" data-server="tencent" data-type="artist" data-fixed="true" data-mini="true" data-listFolded="false" data-order="random" data-preload="none" data-autoplay="flase" muted></div><script defer="defer" id="fluttering_ribbon" mobile="false" src="https://cdn.jsdelivr.net/npm/butterfly-extsrc@1/dist/canvas-fluttering-ribbon.min.js"></script><script id="canvas_nest" defer="defer" color="0,0,255" opacity="0.7" zIndex="-1" count="99" mobile="false" src="https://cdn.jsdelivr.net/npm/butterfly-extsrc@1/dist/canvas-nest.min.js"></script><script>(function(i,s,o,g,r,a,m){i["DaoVoiceObject"]=r;i[r]=i[r]||function(){(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;a.charset="utf-8";m.parentNode.insertBefore(a,m)})(window,document,"script",('https:' == document.location.protocol ? 'https:' : 'http:') + "//widget.daovoice.io/widget/f84bbafb.js","daovoice")
</script><script>var isChatBtn = false
daovoice('init', {
  app_id: 'f84bbafb',},{
  launcher: { 
     disableLauncherIcon: isChatBtn // 悬浮 ICON 是否显示
  },
});
daovoice('update');

if (isChatBtn) {
  var chatBtnFn = () => {
    var chatBtn = document.getElementById("chat_btn")
    chatBtn.addEventListener("click", function(){
      daovoice('show')
    });
  }
  chatBtnFn()
} else {
  if (false) {
    function chatBtnHide () {
      daovoice('update', {},{
        launcher: { 
        disableLauncherIcon: true // 悬浮 ICON 是否显示
        },
      });
    }
    function chatBtnShow () {
      daovoice('update', {},{
        launcher: { 
        disableLauncherIcon: false // 悬浮 ICON 是否显示
        },
      });
    }
  }
}</script><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/aplayer/dist/APlayer.min.css" media="print" onload="this.media='all'"><script src="https://cdn.jsdelivr.net/npm/aplayer/dist/APlayer.min.js"></script><script src="https://cdn.jsdelivr.net/gh/metowolf/MetingJS@1.2/dist/Meting.min.js"></script><script src="https://cdn.jsdelivr.net/npm/pjax/pjax.min.js"></script><script>let pjaxSelectors = [
  'title',
  '#config_change',
  '#body-wrap',
  '#rightside-config-hide',
  '#rightside-config-show',
  '.js-pjax'
]

if (false) {
  pjaxSelectors.unshift('meta[property="og:image"]', 'meta[property="og:title"]', 'meta[property="og:url"]')
}

var pjax = new Pjax({
  elements: 'a:not([target="_blank"])',
  selectors: pjaxSelectors,
  cacheBust: false,
  analytics: true,
  scrollRestoration: false
})

document.addEventListener('pjax:complete', function () {
  window.refreshFn()

  document.querySelectorAll('script[data-pjax]').forEach(item => {
    const newScript = document.createElement('script')
    const content = item.text || item.textContent || item.innerHTML || ""
    Array.from(item.attributes).forEach(attr => newScript.setAttribute(attr.name, attr.value))
    newScript.appendChild(document.createTextNode(content))
    item.parentNode.replaceChild(newScript, item)
  })

  GLOBAL_CONFIG.islazyload && window.lazyLoadInstance.update()

  typeof chatBtnFn === 'function' && chatBtnFn()
  typeof panguInit === 'function' && panguInit()

  if (typeof gtag === 'function') {
    gtag('config', 'G-QRG4NL4207', {'page_path': window.location.pathname});
  }

  typeof loadMeting === 'function' && document.getElementsByClassName('aplayer').length && loadMeting()

  // Analytics
  if (false) {
    MtaH5.pgv()
  }

  // prismjs
  typeof Prism === 'object' && Prism.highlightAll()

  typeof preloader === 'object' && preloader.endLoading()
})


document.addEventListener('pjax:send', function () {
  typeof preloader === 'object' && preloader.initLoading()
  
  if (window.aplayers) {
    for (let i = 0; i < window.aplayers.length; i++) {
      if (!window.aplayers[i].options.fixed) {
        window.aplayers[i].destroy()
      }
    }
  }

  typeof typed === 'object' && typed.destroy()

  //reset readmode
  const $bodyClassList = document.body.classList
  $bodyClassList.contains('read-mode') && $bodyClassList.remove('read-mode')

})</script><script>(function(){
  const bp = document.createElement('script');
  const curProtocol = window.location.protocol.split(':')[0];
  if (curProtocol === 'https') {
    bp.src = 'https://zz.bdstatic.com/linksubmit/push.js';
  }
  else{
    bp.src = 'http://push.zhanzhang.baidu.com/push.js';
  }
  bp.dataset.pjax = ''
  const s = document.getElementsByTagName("script")[0];
  s.parentNode.insertBefore(bp, s);
})()</script></div></body></html>