<!DOCTYPE html><html lang="zh-CN" data-theme="light"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width,initial-scale=1"><title>系统发育基因组学(Phylogenomics)的介绍以及实操 | 生信技术</title><meta name="keywords" content="Phylogenomics"><meta name="author" content="李星泽,lixingzee@gmail.com"><meta name="copyright" content="李星泽"><meta name="format-detection" content="telephone=no"><meta name="theme-color" content="#ffffff"><meta http-equiv="Cache-Control" content="no-transform"><meta http-equiv="Cache-Control" content="no-siteapp"><meta name="description" content="Phylogenomics 写在前面 关于系统发育基因组学的内容，本文参考了 Mike Lee 的文章，将分为两期，本文介绍基本相关原理。  有一个相关的视频，时长为32’51’’ 感兴趣的可以点击下方进入观看。              什么是系统基因组学？ 用一个容易理解但是不准确的概念来表示： 系统基因组学正试图在基因组水平上推断进化关系。 因为在实践中，我们并没有关注的所有生物体整个基因组">
<meta property="og:type" content="article">
<meta property="og:title" content="系统发育基因组学(Phylogenomics)的介绍以及实操">
<meta property="og:url" content="http://lxz9.com/2022/01/13/Phylogenomics/index.html">
<meta property="og:site_name" content="生信技术">
<meta property="og:description" content="Phylogenomics 写在前面 关于系统发育基因组学的内容，本文参考了 Mike Lee 的文章，将分为两期，本文介绍基本相关原理。  有一个相关的视频，时长为32’51’’ 感兴趣的可以点击下方进入观看。              什么是系统基因组学？ 用一个容易理解但是不准确的概念来表示： 系统基因组学正试图在基因组水平上推断进化关系。 因为在实践中，我们并没有关注的所有生物体整个基因组">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="https://s4.ax1x.com/2022/01/13/7Mt8UJ.png">
<meta property="article:published_time" content="2022-01-13T02:00:00.000Z">
<meta property="article:modified_time" content="2022-01-13T05:25:54.873Z">
<meta property="article:author" content="李星泽">
<meta property="article:tag" content="Phylogenomics">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://s4.ax1x.com/2022/01/13/7Mt8UJ.png"><link rel="shortcut icon" href="https://z3.ax1x.com/2021/02/09/yawIzV.png"><link rel="canonical" href="http://lxz9.com/2022/01/13/Phylogenomics/"><link rel="preconnect" href="//cdn.jsdelivr.net"/><link rel="preconnect" href="//www.google-analytics.com" crossorigin=""/><link rel="preconnect" href="//hm.baidu.com"/><link rel="preconnect" href="//busuanzi.ibruce.info"/><link rel="preconnect" href="//zz.bdstatic.com"/><link rel="stylesheet" href="/css/index.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free/css/all.min.css" media="print" onload="this.media='all'"><script>var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "https://hm.baidu.com/hm.js?72a556f842c8990127f397d4e0c54f2c";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script><script async="async" src="https://www.googletagmanager.com/gtag/js?id=G-QRG4NL4207"></script><script>window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-QRG4NL4207');
</script><script>var GLOBAL_CONFIG = { 
  root: '/',
  algolia: undefined,
  localSearch: {"path":"search.xml","languages":{"hits_empty":"找不到您查询的内容：${query}"}},
  translate: undefined,
  noticeOutdate: undefined,
  highlight: {"plugin":"highlighjs","highlightCopy":true,"highlightLang":true},
  copy: {
    success: '复制成功',
    error: '复制错误',
    noSupport: '浏览器不支持'
  },
  relativeDate: {
    homepage: false,
    post: false
  },
  runtime: '',
  date_suffix: {
    just: '刚刚',
    min: '分钟前',
    hour: '小时前',
    day: '天前',
    month: '个月前'
  },
  copyright: undefined,
  lightbox: 'mediumZoom',
  Snackbar: undefined,
  source: {
    jQuery: 'https://cdn.jsdelivr.net/npm/jquery@latest/dist/jquery.min.js',
    justifiedGallery: {
      js: 'https://cdn.jsdelivr.net/npm/justifiedGallery/dist/js/jquery.justifiedGallery.min.js',
      css: 'https://cdn.jsdelivr.net/npm/justifiedGallery/dist/css/justifiedGallery.min.css'
    },
    fancybox: {
      js: 'https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.js',
      css: 'https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.css'
    }
  },
  isPhotoFigcaption: false,
  islazyload: false,
  isanchor: false
};

var saveToLocal = {
  set: function setWithExpiry(key, value, ttl) {
    const now = new Date()
    const expiryDay = ttl * 86400000
    const item = {
      value: value,
      expiry: now.getTime() + expiryDay,
    }
    localStorage.setItem(key, JSON.stringify(item))
  },

  get: function getWithExpiry(key) {
    const itemStr = localStorage.getItem(key)

    if (!itemStr) {
      return undefined
    }
    const item = JSON.parse(itemStr)
    const now = new Date()

    if (now.getTime() > item.expiry) {
      localStorage.removeItem(key)
      return undefined
    }
    return item.value
  }
}

// https://stackoverflow.com/questions/16839698/jquery-getscript-alternative-in-native-javascript
const getScript = url => new Promise((resolve, reject) => {
  const script = document.createElement('script')
  script.src = url
  script.async = true
  script.onerror = reject
  script.onload = script.onreadystatechange = function() {
    const loadState = this.readyState
    if (loadState && loadState !== 'loaded' && loadState !== 'complete') return
    script.onload = script.onreadystatechange = null
    resolve()
  }
  document.head.appendChild(script)
})</script><script id="config_change">var GLOBAL_CONFIG_SITE = { 
  isPost: true,
  isHome: false,
  isHighlightShrink: false,
  isToc: true,
  postUpdate: '2022-01-13 13:25:54'
}</script><noscript><style type="text/css">
  #nav {
    opacity: 1
  }
  .justified-gallery img {
    opacity: 1
  }

  #recent-posts time,
  #post-meta time {
    display: inline !important
  }
</style></noscript><script>(function () {  window.activateDarkMode = function () {
    document.documentElement.setAttribute('data-theme', 'dark')
    if (document.querySelector('meta[name="theme-color"]') !== null) {
      document.querySelector('meta[name="theme-color"]').setAttribute('content', '#0d0d0d')
    }
  }
  window.activateLightMode = function () {
    document.documentElement.setAttribute('data-theme', 'light')
   if (document.querySelector('meta[name="theme-color"]') !== null) {
      document.querySelector('meta[name="theme-color"]').setAttribute('content', '#ffffff')
    }
  }
  const autoChangeMode = 'false'
  const t = saveToLocal.get('theme')
  if (autoChangeMode === '1') {
    const isDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches
    const isLightMode = window.matchMedia('(prefers-color-scheme: light)').matches
    const isNotSpecified = window.matchMedia('(prefers-color-scheme: no-preference)').matches
    const hasNoSupport = !isDarkMode && !isLightMode && !isNotSpecified
    if (t === undefined) {
      if (isLightMode) activateLightMode()
      else if (isDarkMode) activateDarkMode()
      else if (isNotSpecified || hasNoSupport) {
        const now = new Date()
        const hour = now.getHours()
        const isNight = hour <= 6 || hour >= 18
        isNight ? activateDarkMode() : activateLightMode()
      }
      window.matchMedia('(prefers-color-scheme: dark)').addListener(function (e) {
        if (saveToLocal.get('theme') === undefined) {
          e.matches ? activateDarkMode() : activateLightMode()
        }
      })
    } else if (t === 'light') activateLightMode()
    else activateDarkMode()
  } else if (autoChangeMode === '2') {
    const now = new Date()
    const hour = now.getHours()
    const isNight = hour <= 6 || hour >= 18
    if (t === undefined) isNight ? activateDarkMode() : activateLightMode()
    else if (t === 'light') activateLightMode()
    else activateDarkMode()
  } else {
    if (t === 'dark') activateDarkMode()
    else if (t === 'light') activateLightMode()
  }const asideStatus = saveToLocal.get('aside-status')
if (asideStatus !== undefined) {
   if (asideStatus === 'hide') {
     document.documentElement.classList.add('hide-aside')
   } else {
     document.documentElement.classList.remove('hide-aside')
   }
}})()</script><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/aplayer/dist/APlayer.min.css"  media="defer" onload="this.media='all'"><meting-js server='tencent' type='playlist' id='2134037073' fixed='true'></meting-js><style>#article-container.post-content h1:before, h2:before, h3:before, h4:before, h5:before, h6:before { -webkit-animation: avatar_turn_around 1s linear infinite; -moz-animation: avatar_turn_around 1s linear infinite; -o-animation: avatar_turn_around 1s linear infinite; -ms-animation: avatar_turn_around 1s linear infinite; animation: avatar_turn_around 1s linear infinite; }</style><link rel="stylesheet" href="/css/iconfont.css"><link rel="stylesheet" href="/css/background.css"><link rel="stylesheet" href="/css/lxz9.css"><link rel="stylesheet" href="/css/icon.css" media="defer" onload="this.media='all'"><!-- hexo injector head_end start -->
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.12.0/dist/katex.min.css">

<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-math@4.0.0/dist/style.css">
<!-- hexo injector head_end end --><meta name="generator" content="Hexo 5.3.0">
<style>.github-emoji { position: relative; display: inline-block; width: 1.2em; min-height: 1.2em; overflow: hidden; vertical-align: top; color: transparent; }  .github-emoji > span { position: relative; z-index: 10; }  .github-emoji img, .github-emoji .fancybox { margin: 0 !important; padding: 0 !important; border: none !important; outline: none !important; text-decoration: none !important; user-select: none !important; cursor: auto !important; }  .github-emoji img { height: 1.2em !important; width: 1.2em !important; position: absolute !important; left: 50% !important; top: 50% !important; transform: translate(-50%, -50%) !important; user-select: none !important; cursor: auto !important; } .github-emoji-fallback { color: inherit; } .github-emoji-fallback img { opacity: 0 !important; }</style>
<link rel="alternate" href="/atom.xml" title="生信技术" type="application/atom+xml">
</head><body><div id="sidebar"><div id="menu-mask"></div><div id="sidebar-menus"><div class="author-avatar"><img class="avatar-img" src="https://z3.ax1x.com/2021/03/10/6JSyoq.png" onerror="onerror=null;src='/img/friend_404.gif'" alt="avatar"/></div><div class="site-data"><div class="data-item is-center"><div class="data-item-link"><a href="/archives/"><div class="headline">文章</div><div class="length-num">44</div></a></div></div><div class="data-item is-center"><div class="data-item-link"><a href="/tags/"><div class="headline">标签</div><div class="length-num">55</div></a></div></div><div class="data-item is-center"><div class="data-item-link"><a href="/categories/"><div class="headline">分类</div><div class="length-num">16</div></a></div></div></div><hr/><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fa fa-home"></i><span> 首页</span></a></div><div class="menus_item"><a class="site-page" href="/archives/"><i class="fa-fw fa fa-archive"></i><span> 时间轴</span></a></div><div class="menus_item"><a class="site-page" href="/tags/"><i class="fa-fw fa fa-tags"></i><span> 标签</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fa fa-folder-open"></i><span> 分类</span></a></div><div class="menus_item"><a class="site-page" href="javascript:void(0);" rel="external nofollow noreferrer"><i class="fa-fw fas fa-list"></i><span> 清单</span><i class="fas fa-chevron-down expand"></i></a><ul class="menus_item_child"><li><a class="site-page" href="/music/"><i class="fa-fw fas fa-music"></i><span> 音乐</span></a></li><li><a class="site-page" href="/Gallery/"><i class="fa-fw fas fa-images"></i><span> 照片</span></a></li><li><a class="site-page" href="/movies/"><i class="fa-fw fas fa-video"></i><span> 电影</span></a></li></ul></div><div class="menus_item"><a class="site-page" href="/link/"><i class="fa-fw fas fa-link"></i><span> 友链</span></a></div><div class="menus_item"><a class="site-page" href="/map/"><i class="fa-fw fas fa-heart"></i><span> 地图</span></a></div></div></div></div><div class="post" id="body-wrap"><header class="post-bg" id="page-header" style="background-image: url(https://s4.ax1x.com/2022/01/13/7Mt8UJ.png)"><nav id="nav"><span id="blog_name"><a id="site-name" href="/">生信技术</a></span><div id="menus"><div id="search-button"><a class="site-page social-icon search"><i class="fas fa-search fa-fw"></i><span> 搜索</span></a></div><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fa fa-home"></i><span> 首页</span></a></div><div class="menus_item"><a class="site-page" href="/archives/"><i class="fa-fw fa fa-archive"></i><span> 时间轴</span></a></div><div class="menus_item"><a class="site-page" href="/tags/"><i class="fa-fw fa fa-tags"></i><span> 标签</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fa fa-folder-open"></i><span> 分类</span></a></div><div class="menus_item"><a class="site-page" href="javascript:void(0);" rel="external nofollow noreferrer"><i class="fa-fw fas fa-list"></i><span> 清单</span><i class="fas fa-chevron-down expand"></i></a><ul class="menus_item_child"><li><a class="site-page" href="/music/"><i class="fa-fw fas fa-music"></i><span> 音乐</span></a></li><li><a class="site-page" href="/Gallery/"><i class="fa-fw fas fa-images"></i><span> 照片</span></a></li><li><a class="site-page" href="/movies/"><i class="fa-fw fas fa-video"></i><span> 电影</span></a></li></ul></div><div class="menus_item"><a class="site-page" href="/link/"><i class="fa-fw fas fa-link"></i><span> 友链</span></a></div><div class="menus_item"><a class="site-page" href="/map/"><i class="fa-fw fas fa-heart"></i><span> 地图</span></a></div></div><div id="toggle-menu"><a class="site-page"><i class="fas fa-bars fa-fw"></i></a></div></div></nav><div id="post-info"><h1 class="post-title">系统发育基因组学(Phylogenomics)的介绍以及实操</h1><div id="post-meta"><div class="meta-firstline"><span class="post-meta-date"><i class="far fa-calendar-alt fa-fw post-meta-icon"></i><span class="post-meta-label">发表于</span><time class="post-meta-date-created" datetime="2022-01-13T02:00:00.000Z" title="发表于 2022-01-13 10:00:00">2022-01-13</time><span class="post-meta-separator">|</span><i class="fas fa-history fa-fw post-meta-icon"></i><span class="post-meta-label">更新于</span><time class="post-meta-date-updated" datetime="2022-01-13T05:25:54.873Z" title="更新于 2022-01-13 13:25:54">2022-01-13</time></span><span class="post-meta-categories"><span class="post-meta-separator">|</span><i class="fas fa-inbox fa-fw post-meta-icon"></i><a class="post-meta-categories" href="/categories/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF%E5%AD%A6/">生物信息学</a></span></div><div class="meta-secondline"><span class="post-meta-separator">|</span><span class="post-meta-wordcount"><i class="far fa-file-word fa-fw post-meta-icon"></i><span class="post-meta-label">字数总计:</span><span class="word-count">4k</span><span class="post-meta-separator">|</span><i class="far fa-clock fa-fw post-meta-icon"></i><span class="post-meta-label">阅读时长:</span><span>12分钟</span></span><span class="post-meta-separator">|</span><span class="post-meta-pv-cv"><i class="far fa-eye fa-fw post-meta-icon"></i><span class="post-meta-label">阅读量:</span><span id="busuanzi_value_page_pv"></span></span></div></div></div></header><main class="layout" id="content-inner"><div id="post"><article class="post-content" id="article-container"><h1 id="Phylogenomics"><em>Phylogenomics</em></h1>
<h2 id="写在前面">写在前面</h2>
<p>关于系统发育基因组学的内容，本文参考了 <em>Mike Lee</em> 的文章，将分为两期，本文介绍基本相关原理。</p>
<blockquote>
<p>有一个相关的视频，时长为32’51’’ 感兴趣的可以点击下方进入观看。</p>
</blockquote>
<div align="center" class="aspect-ratio">
    <iframe src="https://player.bilibili.com/player.html?aid=765410718&amp;bvid=BV13r4y1U7s7&amp;cid=478796425&amp;page=1" scrolling="no" border="0" frameborder="no" framespacing="0" high_quality="1" danmaku="1" allowfullscreen="true">
    </iframe>
</div>
<h2 id="什么是系统基因组学？">什么是系统基因组学？</h2>
<p>用一个容易理解但是不准确的概念来表示： <strong>系统基因组学正试图在基因组水平上推断进化关系。</strong></p>
<p>因为在实践中，我们并没有关注的所有生物体整个基因组。</p>
<p>并且根据我们考虑的多样性的广度，无论如何使用整个基因组是不可能或毫无意义的（因为它们可能太不同了）。</p>
<p>所以更恰当的说法是：<strong>系统基因组学试图在更接近基因组的水平上推断进化关系，而不是单个基因的系统发育（比如<em>16S rRNA</em> 基因树）。</strong></p>
<p><img src="https://s4.ax1x.com/2022/01/10/7E4UhD.png" alt=""></p>
<p>一般大家习惯于查看和使用的大多数系统发育树是单一基因类型（如 <em>16S rRNA</em> 基因）的不同拷贝去估计进化关系的视觉表示。</p>
<p>如果我们试图在生物体层面上思考，我们将使用这些基因拷贝作为代表生物体本身的代理，并且我们假设这些基因的进化关系告诉我们一些关于它们源生物进化关系的有意义的事情。</p>
<p>我们正在用系统基因组学做同样的事情，只是我们使用多个基因而不是单个基因。</p>
<p>最后，我们仍然只是使用序列作为代理。所以更合适的定义可能是：</p>
<p><strong>系统基因组学试图推断由多个串联基因组成的序列之间的进化关系，同时假设这些推断的进化关系告诉我们一些关于其源基因组进化关系的有意义的信息。</strong></p>
<h2 id="什么是单拷贝核心基因？">什么是单拷贝核心基因？</h2>
<p><strong>单拷贝核心基因，或单拷贝基因 (<em>single-copy genes SCGs</em>)</strong>，是我们目前正在谈论的大多数生物体中恰好存在 1 个拷贝的基因。</p>
<p>为了比较生物体之间的基因，我们需要考虑实际拥有这些基因的生物体。但是我们也需要单拷贝的基因（而不是在我们的目标生物体内往往存在多个拷贝的基因），因为系统发育学中的一个内置假设通常是所考虑的基因处于<strong>相似的进化压力下</strong>。</p>
<p>即使使用单拷贝基因，这也是微不足道的（并且可能永远不会真正完全正确），但是如果同一基因组中有多个基因拷贝（这些被称为旁系同源物），我们更有可能违反该假设。</p>
<p>所以这就是为什么当我们谈论系统基因组学时，SCG 扮演着如此重要的角色。</p>
<p><img src="https://s4.ax1x.com/2022/01/10/7EfPUS.jpg" alt=""></p>
<h2 id="我们应该使用哪些基因？">我们应该使用哪些基因？</h2>
<p>我们选择研究的生物体决定了应该使用的基因。</p>
<p>根据上面谈到的关于为什么 SCG 在系统基因组学中如此有用，如果我们专注于更密切相关的生物群，那么符合这些标准的基因数量将比关注一个更分化的有机体组会相对较大。</p>
<p><img src="https://s4.ax1x.com/2022/01/06/7S07IH.png" alt="ph01"></p>
<p>上图来自 <em>Hug et al. 2016</em> 的生命之树使用了 15 个目标基因，因为它旨在跨越 3 个域。</p>
<p>但是，例如，如果我们只关注蓝藻，那么这一群体中共享的单拷贝基因的数量会大得多。</p>
<p>例如，为了为 <em>GToTree</em> 设计一个特定于蓝藻的单拷贝基因集，需要在 NCBI 提供的至少 90% 中蓝藻基因组中找到恰好存在于 1 个拷贝中的所有基因。</p>
<p>该过程在执行时产生了 251 个基因靶点。相比之下，当对所有细菌基因组应用相同的过程时，结果是 74 个。</p>
<p>没有一套理想的目标基因可以一直使用，因为它完全取决于我们试图研究的多样性的广度。</p>
<h2 id="流程概览">流程概览</h2>
<p>假设有我们想要包含的基因组以及想要定位的一组单拷贝基因。下一步是什么？</p>
<p><strong>目前最常见的方法之一可能如下所示：</strong></p>
<div align="center"><img src="https://img-blog.csdnimg.cn/img_convert/b2fcc83423504f4aa4ee800dc58577b0.png" width="50%" height="50%">
<p><strong>1. 在我们所有的输入基因组中识别我们的目标基因</strong></p>
<ul>
<li>这些由右侧卡通中的颜色表示</li>
<li>请注意，基因组 “B” 缺少 3 个示例目标基因之一</li>
</ul>
<div align="center"><img src="https://s4.ax1x.com/2022/01/13/7MGhb4.png" width="50%" height="50%">
<p><strong>2. 单独比对每组识别的基因</strong></p>
<ul>
<li>例如 从所有输入基因组中识别出的目标基因“X”的所有拷贝都对齐在一起；对所有基因组单独完成相同的过程</li>
<li>对于缺少目标基因之一的任何基因组，将 gaps 插入到该基因组的该基因的比对中（如上图”Genome B”的基因“Y”的比对）</li>
</ul>
<div align="center"><img src="https://s4.ax1x.com/2022/01/13/7MGIa9.png" width="50%" height="50%">
<p><strong>3. 将所有这些序列水平对齐排列</strong></p>
<ul>
<li>所有的单个基因比对并水平对齐排列，如果是氨基酸比对，通常会有一些“spacing”字符，比如几个“X”</li>
</ul>
<div align="center"><img src="https://s4.ax1x.com/2022/01/13/7MG5VJ.png" width="50%" height="50%">
<p><strong>4. 推断这些组合序列的进化关系</strong></p>
<ul>
<li>构建进化树</li>
</ul>
<p>这些一般步骤中的<strong>大多数可以以不同的方式完成，并且有很多决定需要做出</strong>，例如我们如何尝试识别我们的目标基因，我们使用什么来对齐它们，以及我们使用什么来制作我们的树。还有很多我们可能需要注意的事情，例如：</p>
<ul>
<li>
<p>可能想要采取一些措施来进行某种形式的自动筛选，以确保我们不会让任何被确定为我们目标基因版本的基因通过，但可能不应该这样做（因为这可能会影响我们对该基因集的比对）。</p>
</li>
<li>
<p>如果一个给定的输入基因组中已识别出有目标基因的多个拷贝，我们可能不希望包含来自该基因组的该基因（具体可参见上面的SCG部分）。</p>
</li>
<li>
<p>如果输入基因组的目标基因总体上很少被识别，我们可能希望从分析中完全删除该基因组。</p>
</li>
</ul>
<p>下面我们将使用 <a target="_blank" rel="noopener external nofollow noreferrer" href="https://github.com/AstrobioMike/GToTree/wiki/what-is-gtotree%3F">GToTree</a> 做一些系统基因组学，我们将看到它如何为我们自动化这些过程。</p>
<h2 id="系统基因组学实操">系统基因组学实操</h2>
<p>这里要用到一款关于软件 <a target="_blank" rel="noopener external nofollow noreferrer" href="https://github.com/AstrobioMike/GToTree"><strong>GToTree</strong></a> 这是一个系统基因组学工作流程的软件，具体的内容可以通过 <a target="_blank" rel="noopener external nofollow noreferrer" href="https://github.com/AstrobioMike/GToTree/wiki"><em>Github</em> 访问</a>，当然这款软件也不是适用于所有的数据组，还需要去自己详细了解一下。</p>
<p><strong>GToTree</strong> 是一个用户友好的系统基因组学工作流程。</p>
<p>它为我们处理了所有潜在的计算量大的任务，例如：</p>
<ul>
<li>大规模访问基因组数据</li>
<li>整合来自不同文件格式的基因组</li>
<li>执行基因和基因组过滤</li>
<li>将 <em>input labels</em> 交换为 lineage 信息</li>
<li>为我们将流程中的不同工具拼接在一起；等等</li>
</ul>
<p>它使系统发育树的生成和迭代更加容易处理。 <strong>GToTree wiki</strong> 上有更多信息，<br>
这里展示其概览图：</p>
<p><img src="https://s4.ax1x.com/2022/01/06/7S05qO.png" alt="ph03"></p>
<h2 id="安装-GToTree">安装 GToTree</h2>
<p>使用 GToTree 创建一个 conda 环境，并像这样激活它：</p>
<figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">conda create -y -n gtotree -c conda-forge -c bioconda -c defaults -c astrobiomike gtotree</span><br><span class="line"></span><br><span class="line">conda activate gtotree</span><br></pre></td></tr></tbody></table></figure>
<h2 id="聚球藻-Synechococcus-实例">聚球藻(<em>Synechococcus</em>)实例</h2>
<p>聚球藻菌属（学名：<em>Synechococcus</em>）为聚球藻菌科的一个属。它是一种丰富的蓝藻，遍布全球海洋（和其他地方，但这里我们只讨论海洋 Synechococcus）</p>
<p>分类分配工具( <em>taxonomy assignment tool</em> )会告诉我们这些是聚球藻，<strong>这是必要的第一步</strong></p>
<p><strong>下一步</strong>是看看哪些聚球藻(<em>Synechococcus</em>)推断与新基因组最密切相关 —— 这就是从头系统发育树派上用场的地方（在使用新基因组时，无论是来自分离物测序或从宏基因组中恢复，一般在使用 GToTree 生成系统发育树之前使用<a target="_blank" rel="noopener external nofollow noreferrer" href="https://github.com/Ecogenomics/GTDBTk">GTDB-Tk</a> 进行分类分配。）</p>
<p>我们需要提供给 GToTree 两件必需的内容：</p>
<ol>
<li>我们想要研究哪些基因组；<br></li>
<li>我们想用哪些目标基因来构建我们的系统发育树。</li>
</ol>
<h2 id="输入基因组">输入基因组</h2>
<p>把新基因组（<em>FASTA</em>格式）放入一些海洋参考聚球藻中（只需要提供 <em>NCBI accessions</em> 之后 GToTree 会为我们获取它们）。</p>
<p>我们可以使用以下内容下载此起始示例数据：</p>
<figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">curl -L -o syn-gtotree-example.tar.gz https://ndownloader.figshare.com/files/23629763</span><br><span class="line"></span><br><span class="line">tar -xvzf syn-gtotree-example.tar.gz</span><br><span class="line"></span><br><span class="line"><span class="built_in">cd</span> syn-gtotree-example/</span><br></pre></td></tr></tbody></table></figure>
<p>如果我们查看该<code>ref-syn-accs.txt</code>文件，我们可以看到其中有 20 个 <em>NCBI</em> 程序集：</p>
<figure class="highlight plain"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">wc -l ref-syn-accs.txt</span><br><span class="line">head ref-syn-accs.txt</span><br></pre></td></tr></tbody></table></figure>
<figure class="highlight plain"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">GCF_000011385</span><br><span class="line">GCF_000012505</span><br><span class="line">GCF_000012625</span><br><span class="line">GCF_000014585</span><br><span class="line">GCF_000063505</span><br><span class="line">GCF_000063525</span><br><span class="line">GCF_000153065</span><br><span class="line">GCF_000153285</span><br><span class="line">GCF_000153805</span><br><span class="line">GCF_000153825</span><br></pre></td></tr></tbody></table></figure>
<p>其中包括 19 种海洋聚球藻参考，以及 <em>Gloeobacter violaceus</em>（一种更远的蓝藻）作为外群创建有根树🌲。</p>
<blockquote>
<p><strong>注意</strong><br><br>
要生成此参考 <em>accessions</em>，我们可以在 NCBI 网站的 <a target="_blank" rel="noopener external nofollow noreferrer" href="https://www.ncbi.nlm.nih.gov/assembly/"><strong>assembly</strong> 区域</a>中进行搜索，也可以使用 <a target="_blank" rel="noopener external nofollow noreferrer" href="https://dataguide.nlm.nih.gov/edirect/documentation.html"><em>EDirect</em></a> 从命令行搜索它们。在此处的 <a target="_blank" rel="noopener external nofollow noreferrer" href="https://github.com/AstrobioMike/GToTree/wiki/example-usage#generating-inputs"><em>GToTree wiki Alteromonas example</em></a> 中，有两种方式生成我们的参考 <em>accessions</em> 的示例</p>
</blockquote>
<p>我们可以看到我们有 12 个新的基因组序列作为 fasta 文件：</p>
<figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">ls *.fa | wc -l</span><br></pre></td></tr></tbody></table></figure>
<p>我们只需要将它们放入一个可以提供给 GToTree 的文件中，例如：</p>
<figure class="highlight plain"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">ls *.fa &gt; our-genome-fasta-files.txt</span><br><span class="line"></span><br><span class="line">head our-genome-fasta-files.txt</span><br></pre></td></tr></tbody></table></figure>
<figure class="highlight plain"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">GEYO.fa</span><br><span class="line">N19.fa</span><br><span class="line">N26.fa</span><br><span class="line">N32.fa</span><br><span class="line">N5.fa</span><br><span class="line">UW105.fa</span><br><span class="line">UW106.fa</span><br><span class="line">UW140.fa</span><br><span class="line">UW179A.fa</span><br><span class="line">UW179B.fa</span><br></pre></td></tr></tbody></table></figure>
<h2 id="指定我们的目标基因">指定我们的目标基因</h2>
<p>在这种情况下，我们正在处理所有聚球藻，它们是蓝藻目( <em>Cyanobacteria</em> )。</p>
<p>我们可以通过GToTree的 <code>gtt-hmms</code> 看到 <em>HMM</em> 目标基因集：</p>
<figure class="highlight plain"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">gtt-hmms</span><br></pre></td></tr></tbody></table></figure>
<p>它告诉我们 <em>HMM</em> 的存储位置（也可以根据需要自定义），然后列出当前可用的内容：</p>
<figure class="highlight plain"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br></pre></td><td class="code"><pre><span class="line">The environment variable GToTree_HMM_dir is set to:</span><br><span class="line">     /home/lxz/software/conda/envs/gtotree/share/gtotree/hmm_sets/</span><br><span class="line"></span><br><span class="line"> The 15 available HMM SCG-sets here include:</span><br><span class="line"></span><br><span class="line"> Actinobacteria.hmm                (138 genes)</span><br><span class="line"> Alphaproteobacteria.hmm           (117 genes)</span><br><span class="line"> Archaea.hmm                        (76 genes)</span><br><span class="line"> Bacteria_and_Archaea.hmm           (25 genes)</span><br><span class="line"> Bacteria.hmm                       (74 genes)</span><br><span class="line"> Bacteroidetes.hmm                  (90 genes)</span><br><span class="line"> Betaproteobacteria.hmm            (203 genes)</span><br><span class="line"> Chlamydiae.hmm                    (286 genes)</span><br><span class="line"> Cyanobacteria.hmm                 (251 genes)</span><br><span class="line"> Epsilonproteobacteria.hmm         (260 genes)</span><br><span class="line"> Firmicutes.hmm                    (119 genes)</span><br><span class="line"> Gammaproteobacteria.hmm           (172 genes)</span><br><span class="line"> Proteobacteria.hmm                (119 genes)</span><br><span class="line"> Tenericutes.hmm                    (99 genes)</span><br><span class="line"> Universal_Hug_et_al.hmm            (16 genes)</span><br></pre></td></tr></tbody></table></figure>
<p>有一组蓝藻的 HMM 目标基因集，其中包含 251 个基因，适用于对聚球藻的关注。</p>
<p>GToTree 中包含的 SCG-sets 是按此处所述生成的。</p>
<h2 id="运行-GToTree">运行 <em>GToTree</em></h2>
<p>GToTree 的帮助菜单可以通过 <code>GToTree -h</code> 看到</p>
<p>在这里，我们将启动命令，然后分解代码以及 GToTree 在运行时所做的事情。</p>
<p>在这里运行时，迄今为止花费的最大时间是进行比对（使用更多线程运行意味着可以同时运行更多比对，从而大大提高运行时）。</p>
<p>通过比对和制作进化树，蓝藻 HMM 集中的全部 251 个基因，这在笔记本电脑上花费大约 6 分钟。</p>
<p><strong>如果想跳过处理，下面会提供结果文件。</strong></p>
<figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">GToTree -a ref-syn-accs.txt -f our-genome-fasta-files.txt -H Cyanobacteria -t -L Species -j 4 -o Syn-GToTree-out</span><br></pre></td></tr></tbody></table></figure>
<h3 id="参数详解">参数详解</h3>
<figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">GToTree</span><br><span class="line"></span><br><span class="line">-a  <span class="comment">## 保存作为 NCBI assembly 加入提供的输入基因组的文件</span></span><br><span class="line">-f  <span class="comment">## 包含作为 fasta 文件提供的输入基因组的文件</span></span><br><span class="line">-H  <span class="comment">## 要搜索的目标基因（可以提供我们自己的或查看所有可用的gtt-hmms）</span></span><br><span class="line">-t  <span class="comment">## 这告诉 GToTree 使用 TaxonKit 将谱系信息添加到我们正在使用的参考基因组的标签中</span></span><br><span class="line">-L  <span class="comment">## 这允许我们在标签中指定我们想要多少谱系信息，因为对于这棵树，我们只关注聚球藻，我们将其设置为“物种”（实际上包括 NCBI 中的属、种和菌株名称 对于这些错误）</span></span><br><span class="line">-j  <span class="comment">## 设置在可能的情况下并行运行的作业数量（例如，如上所述，这将一次下载和搜索 4 个基因组，并一次运行 4 个比对）</span></span><br><span class="line">-o  <span class="comment">## 设置输出目录的名称</span></span><br><span class="line"><span class="comment">## 有关更多信息，请参阅 GToTree -h</span></span><br></pre></td></tr></tbody></table></figure>
<p>当我们执行该命令时，在此示例中为 <em>GToTree</em>：</p>
<ol>
<li>下载提供的每个参考的氨基酸序列</li>
<li>对所有输入的fasta文件调用开放阅读框架（使用 <em>prodigal</em> ）</li>
<li>识别目标基因（使用 <em>HMMER3</em> ）</li>
<li>根据目标基因估计完成度/冗余度</li>
<li>根据长度和基因组根据目标基因的命中数过滤掉 <em>gene-hits</em></li>
<li>为缺失任何基因的基因组添加所需的 <em>gap</em> 序列</li>
<li>将每个单独的基因集使用 <em>Muscle</em> 进行比对</li>
<li>使用 <em>Trimal</em> 执行比对的自动修剪</li>
<li>将所有比对( <em>alignments</em> )连接在一起</li>
<li>默认使用 <em>FastTree</em> 建树</li>
</ol>
<h2 id="制作-iToL-映射文件来为新基因组着色">制作 iToL 映射文件来为新基因组着色</h2>
<p>这一步是可选的，在这里要做的最后一件事是快速制作颜色映射文件，以便在 <a target="_blank" rel="noopener external nofollow noreferrer" href="https://itol.embl.de/">Interactive Tree of Life (iToL)</a>站点上显示新基因组，我们将在其中可视化我们的树。</p>
<p>这需要一个输入文件，该文件只包含我们想要着色的树上的标签。</p>
<p>在这里，可以从我们的输入 FASTA 列表文件删除后缀 “.fa” 快速制作：</p>
<figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line">$ cat our-genome-fasta-files.txt</span><br><span class="line">GEYO.fa</span><br><span class="line">N19.fa</span><br><span class="line">N26.fa</span><br><span class="line">N32.fa</span><br><span class="line">N5.fa</span><br><span class="line">UW105.fa</span><br><span class="line">UW106.fa</span><br><span class="line">UW140.fa</span><br><span class="line">UW179A.fa</span><br><span class="line">UW179B.fa</span><br><span class="line">UW69.fa</span><br><span class="line">UW86.fa</span><br></pre></td></tr></tbody></table></figure>
<figure class="highlight bash"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">cut -f 1 -d <span class="string">"."</span> our-genome-fasta-files.txt &gt; labels-to-color.txt</span><br></pre></td></tr></tbody></table></figure>
<p>现在我们可以使用 GToTree 帮助程序来快速制作 iToL 兼容文件：</p>
<figure class="highlight plain"><table><tbody><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">gtt-gen-itol-map -g labels-to-color.txt -o iToL-colors.txt</span><br></pre></td></tr></tbody></table></figure>
<h2 id="可视化">可视化</h2>
<p>我们需要在本地计算机上生成的输出文件是<code>Syn-GtoTree-out.tre</code>文件和<code>iToL-colors.txt</code> 文件，以便能够将它们上传到 iToL 网站。</p>
<p>可以使用左侧的文件导航器面板从活页夹环境下载这些文件，或者可以通过单击以下链接下载这两个文件：</p>
<ul>
<li><a target="_blank" rel="noopener external nofollow noreferrer" href="https://ndownloader.figshare.com/files/23305604">Syn-GtoTree-out.tre</a></li>
<li><a target="_blank" rel="noopener external nofollow noreferrer" href="https://ndownloader.figshare.com/files/23305610">iToL-colors.txt</a></li>
</ul>
<p>我们可以<a target="_blank" rel="noopener external nofollow noreferrer" href="https://itol.embl.de/upload.cgi">上传</a>我们的树以在 iToL 网站上对其进行可视化：</p>
<p><img src="https://s4.ax1x.com/2022/01/06/7S0oZD.png" alt="ph04"></p>
<p>我们可以将 <code>Syn-GtoTree-out.tre</code> 文件拖放到：</p>
<p><br><img src="https://s4.ax1x.com/2022/01/06/7S0xL8.gif" alt="ph05"></p>
<p>然后将加载我们的树：</p>
<p><br><img src="https://s4.ax1x.com/2022/01/06/7S0bid.png" alt="ph06"></p>
<p>现在我们可以将我们的树植根于 <em>Gloeobacter violaceus</em>：</p>
<p><br><img src="https://s4.ax1x.com/2022/01/06/7S0qJA.png" alt="ph07"></p>
<p>最后，我们可以将 <code>iToL-colors.txt</code> 文件拖放到树上以突出显示我们的新基因组：</p>
<p><br><img src="https://s4.ax1x.com/2022/01/06/7S0LRI.png" alt="ph08"></p>
<p>最终结果得出这样的树状图：</p>
<p><br><img src="https://s4.ax1x.com/2022/01/06/7S0jQP.png" alt="ph09"></p>
<blockquote>
<p>一般来说，对于给定的一组感兴趣的生物，<strong>最好使用与该组生物一样多的单拷贝核心基因</strong>。<br> 但是，为了获得满足我们当前需求的可信信号，<strong>这并不总是必要的</strong>。</p>
</blockquote>
<p><br><img src="https://s4.ax1x.com/2022/01/06/7SB9oQ.gif" alt="ph10"></p>
<p>在这里，蓝藻特异性 <em>SCG-set</em> 的 251 个基因给出了与其论文图 1 中的 1,002 个目标基因树中获得的几乎相同的树和关系（除了一些旋转、着色和标签调整）：</p>
<p><img src="https://s4.ax1x.com/2022/01/12/7u34zD.png" alt="ph11"></p>
<h2 id="其他有用的输出">其他有用的输出</h2>
<p>除了树文件之外，GToTree 还生成了几个文件。</p>
<ul>
<li><strong>比对文件</strong>（我们可以使用不同的建树软件制作我们的树）</li>
<li><strong>一个基因组汇总表</strong>，其中包含每个基因组的信息，例如估计的完成/冗余百分比、谱系信息、输入标签和树中的最终标签等。</li>
<li><strong>单拷贝基因命中表</strong>，每个基因组每个目标基因的单拷贝基因命中计数</li>
<li><strong>运行日志</strong></li>
<li><strong>一个分区文件</strong>，可以使用混合模型制作树（示例在下面的更多示例下链接）</li>
<li><strong>引用文件</strong>，列出所有使用的程序和相关参考文献</li>
</ul>
<h2 id="参考">参考</h2>
<p><em>Happy Belly Bioinformatics: an open-source resource dedicated to helping biologists utilize bioinformatics</em></p>
<p><em>Lee M D. GToTree: a user-friendly workflow for phylogenomics[J]. Bioinformatics, 2019, 35(20): 4162-4164.</em></p>
<p><em>Lee M D, Ahlgren N A, Kling J D, et al. Marine Synechococcus isolates representing globally abundant genomic lineages demonstrate a unique evolutionary path of genome reduction without a decrease in GC content[J]. Environmental microbiology, 2019, 21(5): 1677-1686.</em></p>
</div></div></div></div></article><div class="post-copyright"><div class="post-copyright__author"><span class="post-copyright-meta">文章作者: </span><span class="post-copyright-info"><a href="mailto:lixingzee@gmail.com" rel="external nofollow noreferrer">李星泽</a></span></div><div class="post-copyright__type"><span class="post-copyright-meta">文章链接: </span><span class="post-copyright-info"><a href="http://lxz9.com/2022/01/13/Phylogenomics/">http://lxz9.com/2022/01/13/Phylogenomics/</a></span></div><div class="post-copyright__notice"><span class="post-copyright-meta">版权声明: </span><span class="post-copyright-info">本博客所有文章除特别声明外，均采用 <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" rel="external nofollow noreferrer" target="_blank">CC BY-NC-SA 4.0</a> 许可协议。转载请注明来自 <a href="http://lxz9.com" target="_blank">生信技术</a>！</span></div></div><div class="tag_share"><div class="post-meta__tag-list"><a class="post-meta__tags" href="/tags/Phylogenomics/">Phylogenomics</a></div><div class="post_share"><div class="social-share" data-image="https://s4.ax1x.com/2022/01/13/7Mt8UJ.png" data-sites="facebook,twitter,wechat,weibo,qq"></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/social-share.js/dist/css/share.min.css" media="print" onload="this.media='all'"><script src="https://cdn.jsdelivr.net/npm/social-share.js/dist/js/social-share.min.js" defer></script></div></div><div class="post-reward"><div class="reward-button"><i class="fas fa-qrcode"></i> 打赏<div class="reward-main"><ul class="reward-all"><li class="reward-item"><a href="https://z3.ax1x.com/2021/01/19/sgZ5y6.jpg" rel="external nofollow noreferrer" target="_blank"><img class="post-qr-code-img" src="https://z3.ax1x.com/2021/01/19/sgZ5y6.jpg" alt="微信"/></a><div class="post-qr-code-desc">微信</div></li><li class="reward-item"><a href="https://z3.ax1x.com/2021/01/19/sgZIOK.jpg" rel="external nofollow noreferrer" target="_blank"><img class="post-qr-code-img" src="https://z3.ax1x.com/2021/01/19/sgZIOK.jpg" alt="支付宝"/></a><div class="post-qr-code-desc">支付宝</div></li></ul></div></div></div><nav class="pagination-post" id="pagination"><div class="prev-post pull-left"><a href="/2022/01/20/rgl/"><img class="prev-cover" src="https://s4.ax1x.com/2022/01/20/76joLV.jpg" onerror="onerror=null;src='/img/404.jpg'" alt="cover of previous post"><div class="pagination-info"><div class="label">上一篇</div><div class="prev_info">使用 RGL 制作交互式 3D 散点图</div></div></a></div><div class="next-post pull-right"><a href="/2022/01/05/Gviz/"><img class="next-cover" src="https://s4.ax1x.com/2022/01/05/TvKUv6.png" onerror="onerror=null;src='/img/404.jpg'" alt="cover of next post"><div class="pagination-info"><div class="label">下一篇</div><div class="next_info">Gviz - 实现基因组数据可视化</div></div></a></div></nav><hr/><div id="post-comment"><div class="comment-head"><div class="comment-headline"><i class="fas fa-comments fa-fw"></i><span> 评论</span></div></div><div class="comment-wrap"><div><div class="vcomment" id="vcomment"></div></div></div></div></div><div class="aside-content" id="aside-content"><div class="card-widget card-info"><div class="card-info-avatar is-center"><img class="avatar-img" src="https://z3.ax1x.com/2021/03/10/6JSyoq.png" onerror="this.onerror=null;this.src='/img/friend_404.gif'" alt="avatar"/><div class="author-info__name">李星泽</div><div class="author-info__description"></div></div><div class="card-info-data"><div class="card-info-data-item is-center"><a href="/archives/"><div class="headline">文章</div><div class="length-num">44</div></a></div><div class="card-info-data-item is-center"><a href="/tags/"><div class="headline">标签</div><div class="length-num">55</div></a></div><div class="card-info-data-item is-center"><a href="/categories/"><div class="headline">分类</div><div class="length-num">16</div></a></div></div><a class="button--animated" id="card-info-btn" target="_blank" rel="noopener external nofollow noreferrer" href="https://github.com/lxingze"><i class="fab fa-github"></i><span>Follow Me</span></a><div class="card-info-social-icons is-center"><a class="social-icon" href="https://blog.csdn.net/m0_49960764" rel="external nofollow noreferrer" target="_blank" title="CSDN"><i class="iconfont icon-csdn"></i></a><a class="social-icon" href="https://z3.ax1x.com/2021/11/25/oAI0h9.jpg" rel="external nofollow noreferrer" target="_blank" title="微信公众号"><i class="iconfont icon-weixin"></i></a><a class="social-icon" href="https://space.bilibili.com/113291922?from=search&amp;seid=3526734520747213504&amp;spm_id_from=333.337.0.0" rel="external nofollow noreferrer" target="_blank" title="bilibili"><i class="iconfont icon-bilibili-copy"></i></a><a class="social-icon" href="mailto:lixingzee@gmail.com" rel="external nofollow noreferrer" target="_blank" title="Email"><i class="fas fa-envelope"></i></a></div></div><div class="card-widget card-announcement"><div class="item-headline"><i class="fas fa-bullhorn card-announcement-animation"></i><span>公告</span></div><div class="announcement_content">如果你有更多想了解的内容，欢迎关注我的个人公众号：<b>生信技术</b> <br> <img src="https://s4.ax1x.com/2021/12/29/TcEUvq.gif"> <br> <br></div><timing></timing></div><div class="sticky_layout"><div class="card-widget" id="card-toc"><div class="item-headline"><i class="fas fa-stream"></i><span>目录</span></div><div class="toc-content"><ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#Phylogenomics"><span class="toc-number">1.</span> <span class="toc-text">Phylogenomics</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#%E5%86%99%E5%9C%A8%E5%89%8D%E9%9D%A2"><span class="toc-number">1.1.</span> <span class="toc-text">写在前面</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E4%BB%80%E4%B9%88%E6%98%AF%E7%B3%BB%E7%BB%9F%E5%9F%BA%E5%9B%A0%E7%BB%84%E5%AD%A6%EF%BC%9F"><span class="toc-number">1.2.</span> <span class="toc-text">什么是系统基因组学？</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E4%BB%80%E4%B9%88%E6%98%AF%E5%8D%95%E6%8B%B7%E8%B4%9D%E6%A0%B8%E5%BF%83%E5%9F%BA%E5%9B%A0%EF%BC%9F"><span class="toc-number">1.3.</span> <span class="toc-text">什么是单拷贝核心基因？</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E6%88%91%E4%BB%AC%E5%BA%94%E8%AF%A5%E4%BD%BF%E7%94%A8%E5%93%AA%E4%BA%9B%E5%9F%BA%E5%9B%A0%EF%BC%9F"><span class="toc-number">1.4.</span> <span class="toc-text">我们应该使用哪些基因？</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E6%B5%81%E7%A8%8B%E6%A6%82%E8%A7%88"><span class="toc-number">1.5.</span> <span class="toc-text">流程概览</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E7%B3%BB%E7%BB%9F%E5%9F%BA%E5%9B%A0%E7%BB%84%E5%AD%A6%E5%AE%9E%E6%93%8D"><span class="toc-number">1.6.</span> <span class="toc-text">系统基因组学实操</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E5%AE%89%E8%A3%85-GToTree"><span class="toc-number">1.7.</span> <span class="toc-text">安装 GToTree</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E8%81%9A%E7%90%83%E8%97%BB-Synechococcus-%E5%AE%9E%E4%BE%8B"><span class="toc-number">1.8.</span> <span class="toc-text">聚球藻(Synechococcus)实例</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E8%BE%93%E5%85%A5%E5%9F%BA%E5%9B%A0%E7%BB%84"><span class="toc-number">1.9.</span> <span class="toc-text">输入基因组</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E6%8C%87%E5%AE%9A%E6%88%91%E4%BB%AC%E7%9A%84%E7%9B%AE%E6%A0%87%E5%9F%BA%E5%9B%A0"><span class="toc-number">1.10.</span> <span class="toc-text">指定我们的目标基因</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E8%BF%90%E8%A1%8C-GToTree"><span class="toc-number">1.11.</span> <span class="toc-text">运行 GToTree</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#%E5%8F%82%E6%95%B0%E8%AF%A6%E8%A7%A3"><span class="toc-number">1.11.1.</span> <span class="toc-text">参数详解</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E5%88%B6%E4%BD%9C-iToL-%E6%98%A0%E5%B0%84%E6%96%87%E4%BB%B6%E6%9D%A5%E4%B8%BA%E6%96%B0%E5%9F%BA%E5%9B%A0%E7%BB%84%E7%9D%80%E8%89%B2"><span class="toc-number">1.12.</span> <span class="toc-text">制作 iToL 映射文件来为新基因组着色</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E5%8F%AF%E8%A7%86%E5%8C%96"><span class="toc-number">1.13.</span> <span class="toc-text">可视化</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E5%85%B6%E4%BB%96%E6%9C%89%E7%94%A8%E7%9A%84%E8%BE%93%E5%87%BA"><span class="toc-number">1.14.</span> <span class="toc-text">其他有用的输出</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E5%8F%82%E8%80%83"><span class="toc-number">1.15.</span> <span class="toc-text">参考</span></a></li></ol></li></ol></div></div><div class="card-widget card-recent-post"><div class="item-headline"><i class="fas fa-history"></i><span>最新文章</span></div><div class="aside-list"><div class="aside-list-item"><a class="thumbnail" href="/2022/04/02/JavaScript/" title="JavaScript"><img src="https://s1.ax1x.com/2022/04/11/LZizLV.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="JavaScript"/></a><div class="content"><a class="title" href="/2022/04/02/JavaScript/" title="JavaScript">JavaScript</a><time datetime="2022-04-02T14:19:09.000Z" title="发表于 2022-04-02 22:19:09">2022-04-02</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2022/04/01/html_css/" title="HTML &amp; CSS"><img src="https://s1.ax1x.com/2022/04/11/LZixs0.jpg" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="HTML &amp; CSS"/></a><div class="content"><a class="title" href="/2022/04/01/html_css/" title="HTML &amp; CSS">HTML &amp; CSS</a><time datetime="2022-04-01T14:19:09.000Z" title="发表于 2022-04-01 22:19:09">2022-04-01</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2022/03/30/git/" title="Git 用法总结"><img src="https://s4.ax1x.com/2022/01/20/76jHdU.jpg" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="Git 用法总结"/></a><div class="content"><a class="title" href="/2022/03/30/git/" title="Git 用法总结">Git 用法总结</a><time datetime="2022-03-29T16:00:00.000Z" title="发表于 2022-03-30 00:00:00">2022-03-30</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2022/03/26/TypeHints/" title="Python 类型提示和静态类型检查"><img src="https://s1.ax1x.com/2022/03/28/qsCkB6.jpg" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="Python 类型提示和静态类型检查"/></a><div class="content"><a class="title" href="/2022/03/26/TypeHints/" title="Python 类型提示和静态类型检查">Python 类型提示和静态类型检查</a><time datetime="2022-03-25T16:00:00.000Z" title="发表于 2022-03-26 00:00:00">2022-03-26</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2022/03/25/mypy/" title="使用 Mypy 进行 Python 静态类型检查"><img src="https://s1.ax1x.com/2022/03/28/qsi4AK.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="使用 Mypy 进行 Python 静态类型检查"/></a><div class="content"><a class="title" href="/2022/03/25/mypy/" title="使用 Mypy 进行 Python 静态类型检查">使用 Mypy 进行 Python 静态类型检查</a><time datetime="2022-03-24T16:00:00.000Z" title="发表于 2022-03-25 00:00:00">2022-03-25</time></div></div></div></div></div></div></main><footer id="footer"><div id="footer-wrap"><div class="copyright">&copy;2020 - 2022&nbsp;<i style="color:#FF6A6A;animation: announ_animation 0.8s linear infinite;"class="fas fa-heart"></i> By 李星泽</div></div><div class="container" id="jsi-flying-fish-container"><script src="/js/fish.js"></script></div><style>   @media only screen and (max-width: 767px){
   #sidebar_search_box input[type=text]{width:calc(100% - 24px)}
}</style></footer></div><div id="rightside"><div id="rightside-config-hide"><button id="readmode" type="button" title="阅读模式"><i class="fas fa-book-open"></i></button><button id="darkmode" type="button" title="浅色和深色模式转换"><i class="fas fa-adjust"></i></button><button id="hide-aside-btn" type="button" title="单栏和双栏切换"><i class="fas fa-arrows-alt-h"></i></button></div><div id="rightside-config-show"><button id="rightside_config" type="button" title="设置"><i class="fas fa-cog fa-spin"></i></button><button class="close" id="mobile-toc-button" type="button" title="目录"><i class="fas fa-list-ul"></i></button><a id="to_comment" href="#post-comment" title="直达评论"><i class="fas fa-comments"></i></a><button id="go-up" type="button" title="回到顶部"><i class="fas fa-arrow-up"></i></button></div></div><div id="local-search"><div class="search-dialog"><div class="search-dialog__title" id="local-search-title">本地搜索</div><div id="local-input-panel"><div id="local-search-input"><div class="local-search-box"><input class="local-search-box--input" placeholder="搜索文章" type="text"/></div></div></div><hr/><div id="local-search-results"></div><span class="search-close-button"><i class="fas fa-times"></i></span></div><div id="search-mask"></div></div><div><script src="/js/utils.js"></script><script src="/js/main.js"></script><script src="https://cdn.jsdelivr.net/npm/medium-zoom/dist/medium-zoom.min.js"></script><script src="/js/search/local-search.js"></script><div class="js-pjax"><script>function loadValine () {
  function initValine () {
    let initData = {
      el: '#vcomment',
      appId: '',
      appKey: '',
      placeholder: 'Please leave your footprints',
      avatar: 'monsterid',
      meta: 'nick,mail,link'.split(','),
      pageSize: '10',
      lang: 'en',
      recordIP: false,
      serverURLs: '',
      emojiCDN: '',
      emojiMaps: "",
      enableQQ: false,
      path: window.location.pathname,
    }

    if (true) { 
      initData.requiredFields= ('nick,mail'.split(','))
    }
    
    if (false) {
      const otherData = false
      initData = Object.assign(initData, otherData)
    }
    
    const valine = new Valine(initData)
  }

  if (typeof Valine === 'function') initValine() 
  else getScript('https://cdn.jsdelivr.net/npm/valine/dist/Valine.min.js').then(initValine)
}

if ('Valine' === 'Valine' || !true) {
  if (true) btf.loadComment(document.getElementById('vcomment'),loadValine)
  else setTimeout(loadValine, 0)
} else {
  function loadOtherComment () {
    loadValine()
  }
}</script><script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script></div><script src="/js/jquery.min.js"></script><script src="/js/fish.js"></script><script src="/js/timing.js"></script><script async src="https://cdn.jsdelivr.net/npm/aplayer"></script><script async src="https://cdn.jsdelivr.net/npm/meting/dist/Meting.min.js"></script><div class="aplayer no-destroy" data-id="000PeZCQ1i4XVs" data-server="tencent" data-type="artist" data-fixed="true" data-mini="true" data-listFolded="false" data-order="random" data-preload="none" data-autoplay="flase data-lrctype="3" " muted></div><script defer="defer" id="fluttering_ribbon" mobile="false" src="https://cdn.jsdelivr.net/npm/butterfly-extsrc@1/dist/canvas-fluttering-ribbon.min.js"></script><script id="canvas_nest" defer="defer" color="0,0,255" opacity="0.7" zIndex="-1" count="99" mobile="false" src="https://cdn.jsdelivr.net/npm/butterfly-extsrc@1/dist/canvas-nest.min.js"></script><script>(function(i,s,o,g,r,a,m){i["DaoVoiceObject"]=r;i[r]=i[r]||function(){(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;a.charset="utf-8";m.parentNode.insertBefore(a,m)})(window,document,"script",('https:' == document.location.protocol ? 'https:' : 'http:') + "//widget.daovoice.io/widget/f84bbafb.js","daovoice")
</script><script>var isChatBtn = false
daovoice('init', {
  app_id: 'f84bbafb',},{
  launcher: { 
     disableLauncherIcon: isChatBtn // 悬浮 ICON 是否显示
  },
});
daovoice('update');

if (isChatBtn) {
  var chatBtnFn = () => {
    var chatBtn = document.getElementById("chat_btn")
    chatBtn.addEventListener("click", function(){
      daovoice('show')
    });
  }
  chatBtnFn()
} else {
  if (false) {
    function chatBtnHide () {
      daovoice('update', {},{
        launcher: { 
        disableLauncherIcon: true // 悬浮 ICON 是否显示
        },
      });
    }
    function chatBtnShow () {
      daovoice('update', {},{
        launcher: { 
        disableLauncherIcon: false // 悬浮 ICON 是否显示
        },
      });
    }
  }
}</script><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/aplayer/dist/APlayer.min.css" media="print" onload="this.media='all'"><script src="https://cdn.jsdelivr.net/npm/aplayer/dist/APlayer.min.js"></script><script src="https://cdn.jsdelivr.net/gh/metowolf/MetingJS@1.2/dist/Meting.min.js"></script><script src="https://cdn.jsdelivr.net/npm/pjax/pjax.min.js"></script><script>let pjaxSelectors = [
  'title',
  '#config_change',
  '#body-wrap',
  '#rightside-config-hide',
  '#rightside-config-show',
  '.js-pjax'
]

if (false) {
  pjaxSelectors.unshift('meta[property="og:image"]', 'meta[property="og:title"]', 'meta[property="og:url"]')
}

var pjax = new Pjax({
  elements: 'a:not([target="_blank"])',
  selectors: pjaxSelectors,
  cacheBust: false,
  analytics: true,
  scrollRestoration: false
})

document.addEventListener('pjax:complete', function () {
  window.refreshFn()

  document.querySelectorAll('script[data-pjax]').forEach(item => {
    const newScript = document.createElement('script')
    const content = item.text || item.textContent || item.innerHTML || ""
    Array.from(item.attributes).forEach(attr => newScript.setAttribute(attr.name, attr.value))
    newScript.appendChild(document.createTextNode(content))
    item.parentNode.replaceChild(newScript, item)
  })

  GLOBAL_CONFIG.islazyload && window.lazyLoadInstance.update()

  typeof chatBtnFn === 'function' && chatBtnFn()
  typeof panguInit === 'function' && panguInit()

  if (typeof gtag === 'function') {
    gtag('config', 'G-QRG4NL4207', {'page_path': window.location.pathname});
  }

  typeof loadMeting === 'function' && document.getElementsByClassName('aplayer').length && loadMeting()

  // Analytics
  if (false) {
    MtaH5.pgv()
  }

  // prismjs
  typeof Prism === 'object' && Prism.highlightAll()

  typeof preloader === 'object' && preloader.endLoading()
})


document.addEventListener('pjax:send', function () {
  typeof preloader === 'object' && preloader.initLoading()
  
  if (window.aplayers) {
    for (let i = 0; i < window.aplayers.length; i++) {
      if (!window.aplayers[i].options.fixed) {
        window.aplayers[i].destroy()
      }
    }
  }

  typeof typed === 'object' && typed.destroy()

  //reset readmode
  const $bodyClassList = document.body.classList
  $bodyClassList.contains('read-mode') && $bodyClassList.remove('read-mode')

})</script><script>(function(){
  const bp = document.createElement('script');
  const curProtocol = window.location.protocol.split(':')[0];
  if (curProtocol === 'https') {
    bp.src = 'https://zz.bdstatic.com/linksubmit/push.js';
  }
  else{
    bp.src = 'http://push.zhanzhang.baidu.com/push.js';
  }
  bp.dataset.pjax = ''
  const s = document.getElementsByTagName("script")[0];
  s.parentNode.insertBefore(bp, s);
})()</script></div></body></html>