<!DOCTYPE html><html lang="zh-CN" data-theme="light"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width,initial-scale=1"><title>WGDI 分析全基因组复制事件完整流程 | 生信技术</title><meta name="keywords" content="教程,WGD,WGDI"><meta name="author" content="李星泽,lixingzee@gmail.com"><meta name="copyright" content="李星泽"><meta name="format-detection" content="telephone=no"><meta name="theme-color" content="#ffffff"><meta http-equiv="Cache-Control" content="no-transform"><meta http-equiv="Cache-Control" content="no-siteapp"><meta name="description" content="简介  WGDI（全基因组重复识别），一种基于 Python 的命令行工具，可让研究人员深入了解递归多倍化并进行跨物种基因组比对分析。    安装 12345678910## 1.使用conda安装conda install -c bioconda  wgdi## 2.使用pip安装pip install wgdi## 3.本地安装git clone https:&#x2F;&#x2F;github.com&#x2F;SunP">
<meta property="og:type" content="article">
<meta property="og:title" content="WGDI 分析全基因组复制事件完整流程">
<meta property="og:url" content="http://lxz9.com/2021/09/06/wgdi/index.html">
<meta property="og:site_name" content="生信技术">
<meta property="og:description" content="简介  WGDI（全基因组重复识别），一种基于 Python 的命令行工具，可让研究人员深入了解递归多倍化并进行跨物种基因组比对分析。    安装 12345678910## 1.使用conda安装conda install -c bioconda  wgdi## 2.使用pip安装pip install wgdi## 3.本地安装git clone https:&#x2F;&#x2F;github.com&#x2F;SunP">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="https://z3.ax1x.com/2021/09/06/h4Hmo6.jpg">
<meta property="article:published_time" content="2021-09-06T13:00:00.000Z">
<meta property="article:modified_time" content="2021-09-06T13:57:58.000Z">
<meta property="article:author" content="李星泽">
<meta property="article:tag" content="教程">
<meta property="article:tag" content="WGD">
<meta property="article:tag" content="WGDI">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://z3.ax1x.com/2021/09/06/h4Hmo6.jpg"><link rel="shortcut icon" href="https://z3.ax1x.com/2021/02/09/yawIzV.png"><link rel="canonical" href="http://lxz9.com/2021/09/06/wgdi/"><link rel="preconnect" href="//cdn.jsdelivr.net"/><link rel="preconnect" href="//www.google-analytics.com" crossorigin=""/><link rel="preconnect" href="//hm.baidu.com"/><link rel="preconnect" href="//busuanzi.ibruce.info"/><link rel="preconnect" href="//zz.bdstatic.com"/><link rel="stylesheet" href="/css/index.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free/css/all.min.css" media="print" onload="this.media='all'"><script>var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "https://hm.baidu.com/hm.js?72a556f842c8990127f397d4e0c54f2c";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script><script async="async" src="https://www.googletagmanager.com/gtag/js?id=G-QRG4NL4207"></script><script>window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-QRG4NL4207');
</script><script>var GLOBAL_CONFIG = { 
  root: '/',
  algolia: undefined,
  localSearch: {"path":"search.xml","languages":{"hits_empty":"找不到您查询的内容：${query}"}},
  translate: undefined,
  noticeOutdate: undefined,
  highlight: {"plugin":"highlighjs","highlightCopy":true,"highlightLang":true},
  copy: {
    success: '复制成功',
    error: '复制错误',
    noSupport: '浏览器不支持'
  },
  relativeDate: {
    homepage: false,
    post: false
  },
  runtime: '',
  date_suffix: {
    just: '刚刚',
    min: '分钟前',
    hour: '小时前',
    day: '天前',
    month: '个月前'
  },
  copyright: undefined,
  lightbox: 'mediumZoom',
  Snackbar: undefined,
  source: {
    jQuery: 'https://cdn.jsdelivr.net/npm/jquery@latest/dist/jquery.min.js',
    justifiedGallery: {
      js: 'https://cdn.jsdelivr.net/npm/justifiedGallery/dist/js/jquery.justifiedGallery.min.js',
      css: 'https://cdn.jsdelivr.net/npm/justifiedGallery/dist/css/justifiedGallery.min.css'
    },
    fancybox: {
      js: 'https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.js',
      css: 'https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.css'
    }
  },
  isPhotoFigcaption: false,
  islazyload: false,
  isanchor: false
};

var saveToLocal = {
  set: function setWithExpiry(key, value, ttl) {
    const now = new Date()
    const expiryDay = ttl * 86400000
    const item = {
      value: value,
      expiry: now.getTime() + expiryDay,
    }
    localStorage.setItem(key, JSON.stringify(item))
  },

  get: function getWithExpiry(key) {
    const itemStr = localStorage.getItem(key)

    if (!itemStr) {
      return undefined
    }
    const item = JSON.parse(itemStr)
    const now = new Date()

    if (now.getTime() > item.expiry) {
      localStorage.removeItem(key)
      return undefined
    }
    return item.value
  }
}

// https://stackoverflow.com/questions/16839698/jquery-getscript-alternative-in-native-javascript
const getScript = url => new Promise((resolve, reject) => {
  const script = document.createElement('script')
  script.src = url
  script.async = true
  script.onerror = reject
  script.onload = script.onreadystatechange = function() {
    const loadState = this.readyState
    if (loadState && loadState !== 'loaded' && loadState !== 'complete') return
    script.onload = script.onreadystatechange = null
    resolve()
  }
  document.head.appendChild(script)
})</script><script id="config_change">var GLOBAL_CONFIG_SITE = { 
  isPost: true,
  isHome: false,
  isHighlightShrink: false,
  isToc: true,
  postUpdate: '2021-09-06 21:57:58'
}</script><noscript><style type="text/css">
  #nav {
    opacity: 1
  }
  .justified-gallery img {
    opacity: 1
  }

  #recent-posts time,
  #post-meta time {
    display: inline !important
  }
</style></noscript><script>(function () {  window.activateDarkMode = function () {
    document.documentElement.setAttribute('data-theme', 'dark')
    if (document.querySelector('meta[name="theme-color"]') !== null) {
      document.querySelector('meta[name="theme-color"]').setAttribute('content', '#0d0d0d')
    }
  }
  window.activateLightMode = function () {
    document.documentElement.setAttribute('data-theme', 'light')
   if (document.querySelector('meta[name="theme-color"]') !== null) {
      document.querySelector('meta[name="theme-color"]').setAttribute('content', '#ffffff')
    }
  }
  const autoChangeMode = 'false'
  const t = saveToLocal.get('theme')
  if (autoChangeMode === '1') {
    const isDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches
    const isLightMode = window.matchMedia('(prefers-color-scheme: light)').matches
    const isNotSpecified = window.matchMedia('(prefers-color-scheme: no-preference)').matches
    const hasNoSupport = !isDarkMode && !isLightMode && !isNotSpecified
    if (t === undefined) {
      if (isLightMode) activateLightMode()
      else if (isDarkMode) activateDarkMode()
      else if (isNotSpecified || hasNoSupport) {
        const now = new Date()
        const hour = now.getHours()
        const isNight = hour <= 6 || hour >= 18
        isNight ? activateDarkMode() : activateLightMode()
      }
      window.matchMedia('(prefers-color-scheme: dark)').addListener(function (e) {
        if (saveToLocal.get('theme') === undefined) {
          e.matches ? activateDarkMode() : activateLightMode()
        }
      })
    } else if (t === 'light') activateLightMode()
    else activateDarkMode()
  } else if (autoChangeMode === '2') {
    const now = new Date()
    const hour = now.getHours()
    const isNight = hour <= 6 || hour >= 18
    if (t === undefined) isNight ? activateDarkMode() : activateLightMode()
    else if (t === 'light') activateLightMode()
    else activateDarkMode()
  } else {
    if (t === 'dark') activateDarkMode()
    else if (t === 'light') activateLightMode()
  }const asideStatus = saveToLocal.get('aside-status')
if (asideStatus !== undefined) {
   if (asideStatus === 'hide') {
     document.documentElement.classList.add('hide-aside')
   } else {
     document.documentElement.classList.remove('hide-aside')
   }
}})()</script><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/aplayer/dist/APlayer.min.css"  media="defer" onload="this.media='all'"><meting-js server='tencent' type='playlist' id='2134037073' fixed='true'></meting-js><style>#article-container.post-content h1:before, h2:before, h3:before, h4:before, h5:before, h6:before { -webkit-animation: avatar_turn_around 1s linear infinite; -moz-animation: avatar_turn_around 1s linear infinite; -o-animation: avatar_turn_around 1s linear infinite; -ms-animation: avatar_turn_around 1s linear infinite; animation: avatar_turn_around 1s linear infinite; }</style><link rel="stylesheet" href="/css/iconfont.css"><link rel="stylesheet" href="/css/background.css"><link rel="stylesheet" href="/css/lxz9.css"><link rel="stylesheet" href="/css/icon.css" media="defer" onload="this.media='all'"><!-- hexo injector head_end start -->
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.12.0/dist/katex.min.css">

<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-math@4.0.0/dist/style.css">
<!-- hexo injector head_end end --><meta name="generator" content="Hexo 5.3.0"><link rel="alternate" href="/atom.xml" title="生信技术" type="application/atom+xml">
</head><body><div id="sidebar"><div id="menu-mask"></div><div id="sidebar-menus"><div class="author-avatar"><img class="avatar-img" src="https://z3.ax1x.com/2021/03/10/6JSyoq.png" onerror="onerror=null;src='/img/friend_404.gif'" alt="avatar"/></div><div class="site-data"><div class="data-item is-center"><div class="data-item-link"><a href="/archives/"><div class="headline">文章</div><div class="length-num">32</div></a></div></div><div class="data-item is-center"><div class="data-item-link"><a href="/tags/"><div class="headline">标签</div><div class="length-num">46</div></a></div></div><div class="data-item is-center"><div class="data-item-link"><a href="/categories/"><div class="headline">分类</div><div class="length-num">14</div></a></div></div></div><hr/><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fa fa-home"></i><span> 首页</span></a></div><div class="menus_item"><a class="site-page" href="/archives/"><i class="fa-fw fa fa-archive"></i><span> 时间轴</span></a></div><div class="menus_item"><a class="site-page" href="/tags/"><i class="fa-fw fa fa-tags"></i><span> 标签</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fa fa-folder-open"></i><span> 分类</span></a></div><div class="menus_item"><a class="site-page" href="javascript:void(0);" rel="external nofollow noreferrer"><i class="fa-fw fas fa-list"></i><span> 清单</span><i class="fas fa-chevron-down expand"></i></a><ul class="menus_item_child"><li><a class="site-page" href="/music/"><i class="fa-fw fas fa-music"></i><span> 音乐</span></a></li><li><a class="site-page" href="/Gallery/"><i class="fa-fw fas fa-images"></i><span> 照片</span></a></li><li><a class="site-page" href="/movies/"><i class="fa-fw fas fa-video"></i><span> 电影</span></a></li></ul></div><div class="menus_item"><a class="site-page" href="/link/"><i class="fa-fw fas fa-link"></i><span> 友链</span></a></div><div class="menus_item"><a class="site-page" href="/map/"><i class="fa-fw fas fa-heart"></i><span> 地图</span></a></div></div></div></div><div class="post" id="body-wrap"><header class="post-bg" id="page-header" style="background-image: url(https://z3.ax1x.com/2021/09/06/h4Hmo6.jpg)"><nav id="nav"><span id="blog_name"><a id="site-name" href="/">生信技术</a></span><div id="menus"><div id="search-button"><a class="site-page social-icon search"><i class="fas fa-search fa-fw"></i><span> 搜索</span></a></div><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fa fa-home"></i><span> 首页</span></a></div><div class="menus_item"><a class="site-page" href="/archives/"><i class="fa-fw fa fa-archive"></i><span> 时间轴</span></a></div><div class="menus_item"><a class="site-page" href="/tags/"><i class="fa-fw fa fa-tags"></i><span> 标签</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fa fa-folder-open"></i><span> 分类</span></a></div><div class="menus_item"><a class="site-page" href="javascript:void(0);" rel="external nofollow noreferrer"><i class="fa-fw fas fa-list"></i><span> 清单</span><i class="fas fa-chevron-down expand"></i></a><ul class="menus_item_child"><li><a class="site-page" href="/music/"><i class="fa-fw fas fa-music"></i><span> 音乐</span></a></li><li><a class="site-page" href="/Gallery/"><i class="fa-fw fas fa-images"></i><span> 照片</span></a></li><li><a class="site-page" href="/movies/"><i class="fa-fw fas fa-video"></i><span> 电影</span></a></li></ul></div><div class="menus_item"><a class="site-page" href="/link/"><i class="fa-fw fas fa-link"></i><span> 友链</span></a></div><div class="menus_item"><a class="site-page" href="/map/"><i class="fa-fw fas fa-heart"></i><span> 地图</span></a></div></div><div id="toggle-menu"><a class="site-page"><i class="fas fa-bars fa-fw"></i></a></div></div></nav><div id="post-info"><h1 class="post-title">WGDI 分析全基因组复制事件完整流程</h1><div id="post-meta"><div class="meta-firstline"><span class="post-meta-date"><i class="far fa-calendar-alt fa-fw post-meta-icon"></i><span class="post-meta-label">发表于</span><time class="post-meta-date-created" datetime="2021-09-06T13:00:00.000Z" title="发表于 2021-09-06 21:00:00">2021-09-06</time><span class="post-meta-separator">|</span><i class="fas fa-history fa-fw post-meta-icon"></i><span class="post-meta-label">更新于</span><time class="post-meta-date-updated" datetime="2021-09-06T13:57:58.000Z" title="更新于 2021-09-06 21:57:58">2021-09-06</time></span><span class="post-meta-categories"><span class="post-meta-separator">|</span><i class="fas fa-inbox fa-fw post-meta-icon"></i><a class="post-meta-categories" href="/categories/%E6%AF%94%E8%BE%83%E5%9F%BA%E5%9B%A0%E7%BB%84%E5%AD%A6/">比较基因组学</a></span></div><div class="meta-secondline"><span class="post-meta-separator">|</span><span class="post-meta-wordcount"><i class="far fa-file-word fa-fw post-meta-icon"></i><span class="post-meta-label">字数总计:</span><span class="word-count">3.8k</span><span class="post-meta-separator">|</span><i class="far fa-clock fa-fw post-meta-icon"></i><span class="post-meta-label">阅读时长:</span><span>16分钟</span></span><span class="post-meta-separator">|</span><span class="post-meta-pv-cv"><i class="far fa-eye fa-fw post-meta-icon"></i><span class="post-meta-label">阅读量:</span><span id="busuanzi_value_page_pv"></span></span></div></div></div></header><main class="layout" id="content-inner"><div id="post"><article class="post-content" id="article-container"><h1 id="简介">简介</h1>
<blockquote>
<p>WGDI（全基因组重复识别），一种基于 Python 的命令行工具，可让研究人员深入了解递归多倍化并进行跨物种基因组比对分析。</p>
</blockquote>
<blockquote></blockquote>
<p><img src="https://z3.ax1x.com/2021/09/06/h47UxJ.png" alt></p>
<h1 id="安装">安装</h1>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">## 1.使用conda安装</span></span><br><span class="line">conda install -c bioconda  wgdi</span><br><span class="line"></span><br><span class="line"><span class="comment">## 2.使用pip安装</span></span><br><span class="line">pip install wgdi</span><br><span class="line"></span><br><span class="line"><span class="comment">## 3.本地安装</span></span><br><span class="line">git <span class="built_in">clone</span> https://github.com/SunPengChuan/wgdi.git</span><br><span class="line"><span class="built_in">cd</span> wgdi</span><br><span class="line">python setup.py install</span><br></pre></td></tr></table></figure>
<h1 id="依赖软件">依赖软件</h1>
<p><a target="_blank" rel="noopener external nofollow noreferrer" href="http://abacus.gene.ucl.ac.uk/software/paml.html">PAML</a> | <a target="_blank" rel="noopener external nofollow noreferrer" href="https://mafft.cbrc.jp/alignment/software/">MAFFT</a> | <a target="_blank" rel="noopener external nofollow noreferrer" href="http://www.drive5.com/muscle/downloads.htm">MUSCLE</a> | <a target="_blank" rel="noopener external nofollow noreferrer" href="http://www.bork.embl.de/pal2nal/#Download">PAL2NAL</a> | <a target="_blank" rel="noopener external nofollow noreferrer" href="http://www.iqtree.org/#download">IQTREE</a></p>
<blockquote>
<p>使用pip下载完成 需要配置文件目录<br>
<code>wgdi -conf /？ &gt;conf.ini</code><br>
里面是默认的文件路径，如果不对应打开修改即可<br>
再次输入 <code>wgdi -conf conf.ini</code><br>
就将配置环境导入到程序中了</p>
</blockquote>
<h1 id="分析">分析</h1>
<h2 id="数据预处理">数据预处理</h2>
<blockquote>
<p>数据处理是很有必要的，如果格式不正确，后面的分析很有可能会报错，大家可以自行处理数据得到gff文件以及基因组len文件</p>
</blockquote>
<div class="note purple no-icon simple"><p>下面提供wgdi作者以及写的处理脚本,具体脚本内容将放到本文末尾</p>
<p><a target="_blank" rel="noopener external nofollow noreferrer" href="http://01.getgff.py">01.getgff.py</a><br>
02.gff_lens.py<br>
03.seq_newname.py<br>
generate_conf.py  # 脚本由<a target="_blank" rel="noopener external nofollow noreferrer" href="https://xuzhougeng.top/">徐洲更</a>制作。<a target="_blank" rel="noopener external nofollow noreferrer" href="https://github.com/xuzhougeng/myscripts/blob/master/comparative/generate_conf.py">下载链接</a></p>
</div>
<h2 id="序列比对">序列比对</h2>
<blockquote>
<p>通过blastp或者diamond进行蛋白之间比对。</p>
</blockquote>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">makeblastdb -<span class="keyword">in</span> ath.pep.fa -dbtype prot</span><br><span class="line">blastp -num_threads 50 -db ath.pep.fa -query ath.pep.fa -outfmt 6 -evalue 1e-5 -num_alignments 20  -out ath.blastp.txt &amp; </span><br></pre></td></tr></table></figure>
<h2 id="点图">点图</h2>
<blockquote>
<p>使用命令进入文件夹取出参数文件，主要是通过前面的[] 去识别程序所以可以将下列所有步骤放到一个文本文件中。</p>
</blockquote>
<p><code>wgdi -d help &gt;&gt; total.conf</code></p>
<h3 id="修改配置文件">修改配置文件</h3>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br></pre></td><td class="code"><pre><span class="line">[dotplot]</span><br><span class="line">blast = blast file <span class="comment"># blast</span></span><br><span class="line">blast_reverse = <span class="literal">false</span></span><br><span class="line">gff1 =  gff1 file <span class="comment"># gff和len使用python程序可以提取，格式有要求</span></span><br><span class="line">gff2 =  gff2 file</span><br><span class="line">lens1 = lens1 file</span><br><span class="line">lens2 = lens2 file</span><br><span class="line">genome1_name =  Genome1 name</span><br><span class="line">genome2_name =  Genome2 name</span><br><span class="line">multiple  = 1 <span class="comment"># 最好同源的个数(用红色表示)，不清楚就写1</span></span><br><span class="line">score = 100</span><br><span class="line">evalue = 1e-5</span><br><span class="line">repeat_number = 10 <span class="comment"># 基因相对另一个基因组同源基因的数量限制</span></span><br><span class="line">position = order <span class="comment"># 与end相比，end的点图相对比较集中。</span></span><br><span class="line">blast_reverse = <span class="literal">false</span> <span class="comment"># 比对中基因对顺序是否翻转,</span></span><br><span class="line">ancestor_left = none <span class="comment"># 点图左侧物种的祖先染色体区域</span></span><br><span class="line">ancestor_top = none <span class="comment"># 点图顶部物种的祖先染色体区域</span></span><br><span class="line">markersize = 0.5 <span class="comment"># 调整图中点的大小</span></span><br><span class="line">figsize = 10,10 <span class="comment"># 保存图片大小比例</span></span><br><span class="line">savefig = savefile(.svg,.png,.pdf) <span class="comment"># 类型： 默认： *.PNG保存图片支持png、pdf、svg格式</span></span><br></pre></td></tr></table></figure>
<p>运行命令 <code>wgdi -d total.conf</code></p>
<blockquote>
<p>任意修改lens1 和lens2的染色体的数量和顺序，可以得到任意染色体间的点图。<br>
如果想修改图片中染色体的位置，则可以改动len文件进行操作。<br>
也可以单独选择几条染色体做点图观察，都是通过改动len文件进行的</p>
</blockquote>
<p><img src="https://z3.ax1x.com/2021/09/06/h5uEPP.png" alt="dotpot"></p>
<p><img src="https://z3.ax1x.com/2021/09/06/h5uZ28.png" alt="dotpot2"></p>
<blockquote>
<p>点图解读：<br>
点图规则：点图是根据左边基因组的基因，最好同源的点为红色，次好的四个基因为蓝色，剩余部分（同源基因有限制个数）为灰色，是根据打分结果进行排序的。<br>
（1）点图需要横向看和纵向看：这个点图是物种自身比对，所以只需横向看。片段深度应该为2，但最好同源个数为1，即红色点没有集中分布在某个片段上。常常可以认两个片段很相似，再加上自身。所以，认定为最近发生的加倍为三倍化。除此之外，蓝色或灰色的片段很少有，表明更古老的加倍很不明显。<br>
（2）对角线上，本不应该出现片段。自身比自身显然是最好匹配，这些点已经去掉了。而剩余的这些由于串联重复造成的。即该基因临近的位置同源性很高，打在了对角线附近。可以计算这些串联重复的ks值，估算重复片段的爆发时间。<br>
（3）这个点图是后续跑共线性的基石。所以，score， evalue， repeat_number判定的同源点的数量是共线性打分矩阵的来源。</p>
</blockquote>
<blockquote>
<p>可以看到点图的结果中，共线性片段，按照blast进行排序，红色，蓝色  ，灰色是最后剩下的，左边相对上面的义工只有十个同源基因对应<br>
看到点图中共线性片段大多是由红色和蓝色构成的。</p>
</blockquote>
<h2 id="共线性">共线性</h2>
<p>使用命令进入文件夹取出参数文件。<code>wgdi -icl ? &gt;&gt; total.conf</code></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br></pre></td><td class="code"><pre><span class="line">[collinearity]</span><br><span class="line">gff1 = gff1 file</span><br><span class="line">gff2 = gff2 file</span><br><span class="line">lens1 = lens1 file</span><br><span class="line">lens2 = lens2 file</span><br><span class="line">blast = blast file</span><br><span class="line">blast_reverse = <span class="literal">false</span></span><br><span class="line">multiple  = 1 <span class="comment"># 用红点显示的同源基因的最佳数量</span></span><br><span class="line">process = 8 <span class="comment"># 根据点图中的颜色分配不同的分数，红色默认为 50，蓝色为 40，灰色为 25。</span></span><br><span class="line">evalue = 1e-5</span><br><span class="line">score = 100 <span class="comment"># 和点图保持一致</span></span><br><span class="line">grading = 50,40,25 <span class="comment"># 红，蓝，灰的不同分值,按照分值会优先连红色的点，蓝色点次之</span></span><br><span class="line">mg = 40,40 <span class="comment"># 两个基因对被认为能连起来的最大距离（罚分）</span></span><br><span class="line">pvalue = 0.2 <span class="comment"># 评估共线性的指标，如果想全部提取可以选1，共线块的compactness and uniqueness，范围为0-1，较好的共线范围为0-0.2</span></span><br><span class="line">repeat_number = 10 <span class="comment"># 和点图保持一致 </span></span><br><span class="line">positon = order <span class="comment">#唯一选项</span></span><br><span class="line">savefile = collinearity file <span class="comment"># 输出文件</span></span><br></pre></td></tr></table></figure>
<p>运行 <code>wgdi -icl total.conf</code></p>
<h2 id="Ks值计算">Ks值计算</h2>
<p><code>wgdi -ks ? &gt;&gt; total.conf</code></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line">[ks]</span><br><span class="line">cds_file =      cds file <span class="comment"># 两个物种或物种自身的cds序列，如果要计算多个物种，可以将它们文件cat到一起</span></span><br><span class="line">pep_file =      pep file <span class="comment"># 两个物种或物种自身的pep序列</span></span><br><span class="line">align_software = muscle <span class="comment"># 选择mafft or muscle</span></span><br><span class="line">pairs_file = gene pairs file <span class="comment">#代表其中一种: colinearscan输出结果 / mcscanx输出结果 / collinearity输出结果 / tab分割的基因对</span></span><br><span class="line">ks_file = ks result <span class="comment"># 输出文件，格式有要求</span></span><br></pre></td></tr></table></figure>
<p>运行 <code>wgdi -ks total.conf</code></p>
<blockquote>
<p>输出的ks结果，基因对算不出来的ks值，默认为-1。</p>
</blockquote>
<h2 id="Blockinfo">Blockinfo</h2>
<p>因为共线性和ks值结果不容易展示，而且，常常需要对共线性结果筛选，因此，将这些信息汇总到一个表中，方便删选。<br>
这一步主要是将前面跑的共线性和ks值统一到一个文件当中。</p>
<p>首先将参数保存到文件当中<br>
<code>wgdi -bi ? &gt;&gt; total.conf</code></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line">[blockinfo]</span><br><span class="line">blast = blast file</span><br><span class="line">gff1 =  gff1 file</span><br><span class="line">gff2 =  gff2 file</span><br><span class="line">lens1 = lens1 file</span><br><span class="line">lens2 = lens2 file</span><br><span class="line">collinearity = collinearity file</span><br><span class="line">score = 100 <span class="comment"># 和之前设置的保持一致</span></span><br><span class="line">evalue = 1e-5 <span class="comment"># 和点图、共线性计算的配置文件保持一致</span></span><br><span class="line">repeat_number = 20 <span class="comment"># 和点图、共线性计算的配置文件保持一致</span></span><br><span class="line">position = order <span class="comment"># 仅有order选项</span></span><br><span class="line">ks = ks file</span><br><span class="line">ks_col = ks_NG86 <span class="comment"># 根据之前ks得出的列表选择，软件会生成ks_NG86或ks_YN00，也可以用其他方式计算的结果单独一列</span></span><br><span class="line">savefile = block_information.csv <span class="comment"># 将共线性和ks值汇总到一起</span></span><br></pre></td></tr></table></figure>
<p>运行 <code>wgdi -bi total.conf</code></p>
<p>结果生成一个表block_information.csv<br>
<img src="https://z3.ax1x.com/2021/09/06/h4WMRI.png" alt></p>
<blockquote>
<p>第1列: id  共线性的结果的唯一标识<br>
第2.4.5列: chr1,start1,end1 参考基因组（点图的左边）的共线性范围<br>
第3.6.7列: chr2,start2,end2 参考基因组（点图的上边）的共线性范围<br>
第8列: pvalue 评估共线性的结果，常常认为小于0.01的更合理些<br>
第9列: length 共线性片段长度<br>
ks_median 共线性片段上所有基因对ks的中位数（主要用来评判ks分布的）<br>
ks_average 共线性片段上所有基因对ks的平均值<br>
homo1   homo2   homo3   homo4   homo5 与multiple参数有关，即homo+multiple<br>
主要规则是：基因对如果在点图中为红色，赋值为1，蓝色赋值为0，灰色赋值为-1。共线性片段上所有基因对赋值后求平均值，这样就赋予该共线性一个-1,1的值。如果共线性的点大部分为红色，那么该值接近于1。可以作为共线性片段的筛选。<br>
block1,block2 分别为共线性片段上基因order的位置。<br>
ks 共线性片段的上基因对的ks值<br>
density1, density2 共线性片段的基因分布密集程度。值越小表示稀疏;如何计算的呢？就是length 除以对应的srart1 end1 差值</p>
</blockquote>
<h2 id="Correspondence">Correspondence</h2>
<p><code>wgdi -c ? &gt;&gt; total.conf</code></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">[correspondence]</span><br><span class="line">blockinfo =  block_information.csv <span class="comment">#上一步blockinfo的输出csv文件作为输入文件</span></span><br><span class="line">lens1 = lens1 file</span><br><span class="line">lens2 = lens2 file</span><br><span class="line">tandem = (<span class="literal">true</span>/<span class="literal">false</span>) <span class="comment">#前一步展示过了不需要分析改为false</span></span><br><span class="line">tandem_length = 200 <span class="comment">#看对角线有没有加倍造成的片段，如果对角线上的ks值趋于0，那就不属于。</span></span><br><span class="line">pvalue = 0.05 <span class="comment">#要提取好的共线性 作者设置为了0.2</span></span><br><span class="line">block_length = 5</span><br><span class="line">multiple  = 1</span><br><span class="line">homo = 0,1</span><br><span class="line">savefile = block_information.new.csv <span class="comment"># 名称自定义(*.csv即可)</span></span><br></pre></td></tr></table></figure>
<p><code>wgd -c total.conf</code></p>
<p>筛选完的结果，是将对角线上的片段去除，剩余一些需要的片段。<br>
得出筛选完之后的结果会发现新的csv文件会比之前的文件要小，主要就是将对角线上的片段给去除了。</p>
<h2 id="blockks">blockks</h2>
<p><code>wgdi -bk ? &gt; blockks.conf</code></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line">[blockks]</span><br><span class="line">lens1 = lens1 file</span><br><span class="line">lens2 = lens2 file</span><br><span class="line">genome1_name =  Genome1 name</span><br><span class="line">genome2_name =  Genome2 name</span><br><span class="line">blockinfo = block_information.new.csv</span><br><span class="line">pvalue = 0.05</span><br><span class="line">tandem = <span class="literal">true</span></span><br><span class="line">tandem_length = 200</span><br><span class="line">markersize = 1 <span class="comment">#点的大小</span></span><br><span class="line">area = 0,2 <span class="comment">#显示范围</span></span><br><span class="line">block_length =  minimum length</span><br><span class="line">figsize = 8,8 <span class="comment">#点图大小的比例值</span></span><br><span class="line">savefig = save image <span class="comment">#保存图片，默认： *.PNG保存图片支持png、pdf、svg格式</span></span><br></pre></td></tr></table></figure>
<p><code>wgdi -bk total.conf</code></p>
<p>最终得到的结果会发现对角线上的点被去除，但是还是会有一些点会没有去除。但是也可以继续进行后续操作。</p>
<h2 id="KsPeaks">KsPeaks</h2>
<p>计算Ks峰值</p>
<p><code>wgdi -kp ? &gt;&gt; total.conf</code></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line">[kspeaks]</span><br><span class="line"></span><br><span class="line"> blockinfo = block_information.new.csv</span><br><span class="line"> pvalue = 0.05 <span class="comment">#设置可自行调整</span></span><br><span class="line"> tandem = <span class="literal">true</span></span><br><span class="line"> block_ length = int number</span><br><span class="line"> ks_area = 0,10</span><br><span class="line"> multiple = 1</span><br><span class="line"> homo = 0,1 <span class="comment">#结合最开始的点图，红色点代表1，查看共线性片段如果红色居多，整个共线性片段的值就接近于1，可以写成0.3/0.5,1</span></span><br><span class="line"> fontsize = 9</span><br><span class="line"> area = 0,3 </span><br><span class="line"> figsize = 10,6.18</span><br><span class="line"> savefig = saving image</span><br><span class="line"> savefile = ks_median.distri.csv</span><br></pre></td></tr></table></figure>
<p><code>wgdi -kp total.conf</code></p>
<p><img src="https://z3.ax1x.com/2021/09/06/h5uV8f.png" alt="KsPeaks"></p>
<p>结果图片如果发现前面趋于0 的值比较多，查看前面得出的点图，原因是由于Correspondence这一步设置的tandom值长度偏小，<br>
可以继续拟合或者直接在ks_median.distri.csv文件中进行筛选，将ks_median和ks_average进行从小到大排序，将数值过低的去除，它们还有一个特性就是<br>
chr1以及chr2 都是自己对自己。<br>
将修改结果重新输入[kspeaks] blockinfo = 修改后的(<code>*.csv</code>)</p>
<h2 id="PeaksFit">PeaksFit</h2>
<p>峰拟合，需要结合KsPeaks和blockks的结果看峰为一个还是多个，<code>mode = median</code>建议用每个block的中位数的ks值去做拟合。</p>
<p><code>wgdi -pf ? &gt;&gt; total.conf</code></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">[peaksfit] </span><br><span class="line">blockinfo = block information  <span class="comment">#经过kspeaks处理的单个峰的输出文件(*.csv)</span></span><br><span class="line">mode = median  <span class="comment">#分为，对应Ks峰图的三块median，average，total</span></span><br><span class="line">bins_number = 200  <span class="comment">#柱状图的bins</span></span><br><span class="line">ks_area = 0,10  <span class="comment">#ks值的范围，可以去掉一些奇异的ks值</span></span><br><span class="line">fontsize = 9 </span><br><span class="line">area = 0,3  <span class="comment">#横轴的范围</span></span><br><span class="line">figsize = 10,6.18 </span><br><span class="line">savefig = saving image</span><br></pre></td></tr></table></figure>
<p><code>wgdi -pf total.conf</code></p>
<blockquote>
<p>跑完得到 拟合优度以及对应的曲线<br>
R-square: 0.9618516780863838<br>
The gaussian fitting curve parameters are :<br>
1.305477901404146  |  2.1185082504069626  |  0.41222580475422466<br>
其中：<br>
中间的第二个值就是峰值的大小，数据保存到单独文件中，就可以绘制多个物种种内和种间的ks峰值了。</p>
</blockquote>
<p>关于一个物种发生多次加倍事件如何获取Ks峰值的策略:<br>
每次加倍保留的共线性片段是一定的，是某一次产生的。可以通过Ks值进行区分。<br>
应该对其分开处理，对共线性片段筛选（去除对角线的Ks峰或者有多次加倍的颜色发生变化的需要去除之后在进行拟合）</p>
<p><img src="https://z3.ax1x.com/2021/09/06/h5uk5t.png" alt="PeaksFit"></p>
<h1 id="脚本">脚本</h1>
<h2 id="getgff-py"><a target="_blank" rel="noopener external nofollow noreferrer" href="http://getgff.py">getgff.py</a></h2>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">import</span> sys</span><br><span class="line"><span class="keyword">import</span> pandas <span class="keyword">as</span> pd</span><br><span class="line"></span><br><span class="line">data = pd.read_csv(sys.argv[<span class="number">1</span>], sep=<span class="string">&quot;\t&quot;</span>, header=<span class="literal">None</span>,skiprows=<span class="number">0</span>)</span><br><span class="line">data = data[data[<span class="number">2</span>] == <span class="string">&#x27;mRNA&#x27;</span>]</span><br><span class="line">data = data.loc[:, [<span class="number">0</span>, <span class="number">8</span>, <span class="number">3</span>, <span class="number">4</span>, <span class="number">6</span>]]</span><br><span class="line">data[<span class="number">8</span>] = data[<span class="number">8</span>].<span class="built_in">str</span>.split(<span class="string">&#x27;:|=|;|\|&#x27;</span>,expand=<span class="literal">True</span>)[<span class="number">1</span>]</span><br><span class="line"><span class="comment"># data.drop_duplicates(subset=[8], keep=&#x27;first&#x27;, inplace=True)</span></span><br><span class="line">data[<span class="number">0</span>] = data[<span class="number">0</span>].<span class="built_in">str</span>.replace(<span class="string">&#x27;At&#x27;</span>,<span class="string">&#x27;&#x27;</span>,regex=<span class="literal">True</span>)</span><br><span class="line">data.to_csv(sys.argv[<span class="number">2</span>], sep=<span class="string">&quot;\t&quot;</span>, header=<span class="literal">None</span>, index=<span class="literal">False</span>)</span><br></pre></td></tr></table></figure>
<h2 id="gff-lens-py">gff_lens.py</h2>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#!usr/bin/env python</span></span><br><span class="line"><span class="comment"># conding: utf-8</span></span><br><span class="line"><span class="keyword">import</span> sys</span><br><span class="line"></span><br><span class="line"><span class="keyword">import</span> pandas <span class="keyword">as</span> pd</span><br><span class="line"></span><br><span class="line">data = pd.read_csv(sys.argv[<span class="number">1</span>], sep=<span class="string">&quot;\t&quot;</span>, header=<span class="literal">None</span>)</span><br><span class="line"></span><br><span class="line">new = data[<span class="number">1</span>].<span class="built_in">str</span>.split(<span class="string">&#x27;.&#x27;</span>).<span class="built_in">str</span></span><br><span class="line">data[<span class="string">&#x27;id&#x27;</span>] = new[<span class="number">0</span>].values</span><br><span class="line">data[<span class="string">&#x27;cha&#x27;</span>] = data[<span class="number">3</span>]-data[<span class="number">2</span>]</span><br><span class="line"></span><br><span class="line"><span class="keyword">for</span> name, group <span class="keyword">in</span> data.groupby([<span class="string">&#x27;id&#x27;</span>]):</span><br><span class="line">    <span class="keyword">if</span> <span class="built_in">len</span>(group) == <span class="number">1</span>:</span><br><span class="line">        <span class="keyword">continue</span></span><br><span class="line">    ind = group.sort_values(by=<span class="string">&#x27;cha&#x27;</span>, ascending=<span class="literal">False</span>).index[<span class="number">1</span>:].values</span><br><span class="line">    data.drop(index=ind, inplace=<span class="literal">True</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># data[2] =data[2].astype(int)</span></span><br><span class="line"><span class="comment"># data[3] =data[3].astype(int)</span></span><br><span class="line"><span class="comment"># for name, group in data.groupby(0):</span></span><br><span class="line"><span class="comment">#     if len(group) == 1:</span></span><br><span class="line"><span class="comment">#         continue</span></span><br><span class="line"><span class="comment">#     start=0</span></span><br><span class="line"><span class="comment">#     # print(group.head())</span></span><br><span class="line"><span class="comment">#     group = group.sort_values(by=[2,3],ascending=[True,False])</span></span><br><span class="line"><span class="comment">#     for index, row in group.iterrows():</span></span><br><span class="line"><span class="comment">#         # print(row)</span></span><br><span class="line"><span class="comment">#         if row[3]&gt;start or row[2]&gt;start:</span></span><br><span class="line"><span class="comment">#             start=max([row[3],row[2]])</span></span><br><span class="line"><span class="comment">#             continue</span></span><br><span class="line"><span class="comment">#         data.drop(index=index, inplace=True)</span></span><br><span class="line">    <span class="comment"># ind = group.sort_values(by=&#x27;cha&#x27;, ascending=False).index[1:].values</span></span><br><span class="line">    <span class="comment">#print(name)</span></span><br><span class="line">    <span class="comment"># print(group.sort_values(by=&#x27;cha&#x27;,ascending=False))</span></span><br><span class="line"></span><br><span class="line">    </span><br><span class="line"></span><br><span class="line"><span class="comment"># data = data[data[1].str.contains(&#x27;\.mRNA1$&#x27;)]</span></span><br><span class="line">data[<span class="string">&#x27;order&#x27;</span>] = <span class="string">&#x27;&#x27;</span></span><br><span class="line">data[<span class="string">&#x27;newname&#x27;</span>] = <span class="string">&#x27;&#x27;</span></span><br><span class="line">print(data.head())</span><br><span class="line"><span class="keyword">for</span> name, group <span class="keyword">in</span> data.groupby([<span class="number">0</span>]):</span><br><span class="line">    number = <span class="built_in">len</span>(group)</span><br><span class="line">    group = group.sort_values(by=[<span class="number">2</span>])</span><br><span class="line">    data.loc[group.index, <span class="string">&#x27;order&#x27;</span>] = <span class="built_in">list</span>(<span class="built_in">range</span>(<span class="number">1</span>, <span class="built_in">len</span>(group)+<span class="number">1</span>))</span><br><span class="line">    data.loc[group.index, <span class="string">&#x27;newname&#x27;</span>] = <span class="built_in">list</span>(</span><br><span class="line">        [<span class="string">&#x27;ath2s&#x27;</span>+<span class="built_in">str</span>(name)+<span class="string">&#x27;g&#x27;</span>+<span class="built_in">str</span>(i).zfill(<span class="number">5</span>) <span class="keyword">for</span> i <span class="keyword">in</span> <span class="built_in">range</span>(<span class="number">1</span>, <span class="built_in">len</span>(group)+<span class="number">1</span>)])</span><br><span class="line">data[<span class="string">&#x27;order&#x27;</span>] = data[<span class="string">&#x27;order&#x27;</span>].astype(<span class="string">&#x27;int&#x27;</span>)</span><br><span class="line">data = data[[<span class="number">0</span>, <span class="string">&#x27;newname&#x27;</span>, <span class="number">2</span>, <span class="number">3</span>, <span class="number">4</span>, <span class="string">&#x27;order&#x27;</span>, <span class="number">1</span>]]</span><br><span class="line">print(data.head())</span><br><span class="line">data = data.sort_values(by=[<span class="number">0</span>, <span class="string">&#x27;order&#x27;</span>])</span><br><span class="line">data.to_csv(sys.argv[<span class="number">2</span>], sep=<span class="string">&quot;\t&quot;</span>, index=<span class="literal">False</span>, header=<span class="literal">None</span>)</span><br><span class="line">lens = data.groupby(<span class="number">0</span>).<span class="built_in">max</span>()[[<span class="number">3</span>, <span class="string">&#x27;order&#x27;</span>]]</span><br><span class="line">lens.to_csv(sys.argv[<span class="number">3</span>], sep=<span class="string">&quot;\t&quot;</span>, header=<span class="literal">None</span>)</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h2 id="seq-newname-py">seq_newname.py</h2>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">import</span> sys</span><br><span class="line"><span class="keyword">import</span> pandas <span class="keyword">as</span> pd</span><br><span class="line"><span class="keyword">from</span> Bio <span class="keyword">import</span> SeqIO</span><br><span class="line"></span><br><span class="line">data = pd.read_csv(sys.argv[<span class="number">1</span>], sep=<span class="string">&quot;\t&quot;</span>, header=<span class="literal">None</span>, index_col=<span class="number">6</span>)</span><br><span class="line">data.index = data.index.<span class="built_in">str</span>[:-<span class="number">3</span>]</span><br><span class="line">id_dict = data[<span class="number">1</span>].to_dict()</span><br><span class="line">print(data.head())</span><br><span class="line">seqs = []</span><br><span class="line">n = <span class="number">0</span></span><br><span class="line"><span class="keyword">for</span> seq_record <span class="keyword">in</span> SeqIO.parse(sys.argv[<span class="number">2</span>], <span class="string">&quot;fasta&quot;</span>):</span><br><span class="line">    <span class="keyword">if</span> seq_record.<span class="built_in">id</span> <span class="keyword">in</span> id_dict:</span><br><span class="line">        seq_record.<span class="built_in">id</span> = id_dict[seq_record.<span class="built_in">id</span>]</span><br><span class="line">        n += <span class="number">1</span></span><br><span class="line">    <span class="keyword">else</span>:</span><br><span class="line">        <span class="keyword">continue</span></span><br><span class="line">    seqs.append(seq_record)</span><br><span class="line">SeqIO.write(seqs, sys.argv[<span class="number">3</span>], <span class="string">&quot;fasta&quot;</span>)</span><br><span class="line">print(n)</span><br></pre></td></tr></table></figure>
<h2 id="generate-conf-py">generate_conf.py</h2>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#!/usr/bin/env python3</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># GFF must have CDS feature</span></span><br><span class="line"><span class="comment"># GFF must have ID and Parent in column 9</span></span><br><span class="line"></span><br><span class="line"><span class="keyword">import</span> re</span><br><span class="line"><span class="keyword">import</span> argparse</span><br><span class="line"><span class="keyword">from</span> collections <span class="keyword">import</span> defaultdict</span><br><span class="line"><span class="keyword">from</span> collections <span class="keyword">import</span> OrderedDict</span><br><span class="line"></span><br><span class="line"><span class="function"><span class="keyword">def</span> <span class="title">get_parser</span>():</span></span><br><span class="line">    <span class="string">&quot;&quot;&quot;Get options&quot;&quot;&quot;</span></span><br><span class="line">    parser = argparse.ArgumentParser()</span><br><span class="line">    parser.add_argument(<span class="string">&#x27;fasta&#x27;</span>,</span><br><span class="line">                        <span class="built_in">help</span>=<span class="string">&quot;fasta file name&quot;</span>)</span><br><span class="line">    parser.add_argument(<span class="string">&#x27;gff&#x27;</span>,</span><br><span class="line">                        <span class="built_in">help</span>=<span class="string">&quot;gff file name&quot;</span>)</span><br><span class="line">    parser.add_argument(<span class="string">&#x27;-p&#x27;</span>,<span class="string">&#x27;--prefix&#x27;</span>, <span class="built_in">type</span>=<span class="built_in">str</span>, default=<span class="string">&quot;output&quot;</span>,</span><br><span class="line">                        <span class="built_in">help</span>=<span class="string">&quot;prefix for ouput &quot;</span>)</span><br><span class="line"></span><br><span class="line">    <span class="keyword">return</span> parser</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment"># get the fasta  len</span></span><br><span class="line"><span class="function"><span class="keyword">def</span> <span class="title">get_fasta_len</span>(<span class="params">fasta</span>):</span></span><br><span class="line">    fasta_dict = OrderedDict()</span><br><span class="line">    handle = <span class="built_in">open</span>(fasta, <span class="string">&quot;r&quot;</span>)</span><br><span class="line">    active_sequence_name = <span class="string">&quot;&quot;</span></span><br><span class="line">    <span class="keyword">for</span> line <span class="keyword">in</span> handle:</span><br><span class="line">        line = line.strip()</span><br><span class="line">        <span class="keyword">if</span> <span class="keyword">not</span> line:</span><br><span class="line">            <span class="keyword">continue</span></span><br><span class="line">        <span class="keyword">if</span> line.startswith(<span class="string">&quot;&gt;&quot;</span>): </span><br><span class="line">            active_sequence_name = line[<span class="number">1</span>:]</span><br><span class="line">            active_sequence_name = active_sequence_name.split(<span class="string">&quot; &quot;</span>)[<span class="number">0</span>]</span><br><span class="line">        <span class="keyword">if</span> active_sequence_name <span class="keyword">not</span> <span class="keyword">in</span> fasta_dict:</span><br><span class="line">            fasta_dict[active_sequence_name] = <span class="number">0</span></span><br><span class="line">            <span class="keyword">continue</span></span><br><span class="line">        sequence = line</span><br><span class="line">        fasta_dict[active_sequence_name] += <span class="built_in">len</span>(sequence)</span><br><span class="line">    handle.close()</span><br><span class="line">    <span class="keyword">return</span> fasta_dict</span><br><span class="line"></span><br><span class="line"><span class="comment"># parse the gff </span></span><br><span class="line"><span class="function"><span class="keyword">def</span> <span class="title">parse_gff</span>(<span class="params">gff</span>):</span></span><br><span class="line"></span><br><span class="line">    gene_dict = OrderedDict()</span><br><span class="line">    tx_pos_dict = defaultdict(<span class="built_in">list</span>)</span><br><span class="line">    CDS_dict = defaultdict(<span class="built_in">list</span>)</span><br><span class="line"></span><br><span class="line">    handle = <span class="built_in">open</span>(gff, <span class="string">&quot;r&quot;</span>)</span><br><span class="line"></span><br><span class="line">    <span class="keyword">for</span> line <span class="keyword">in</span> handle:</span><br><span class="line">        <span class="keyword">if</span> line.startswith(<span class="string">&quot;#&quot;</span>):</span><br><span class="line">            <span class="keyword">continue</span></span><br><span class="line">        content = line.split(<span class="string">&quot;\t&quot;</span>)</span><br><span class="line">        <span class="keyword">if</span> <span class="built_in">len</span>(content) &lt;= <span class="number">8</span>:</span><br><span class="line">            <span class="keyword">continue</span></span><br><span class="line">        <span class="comment">#print(content)</span></span><br><span class="line">        <span class="keyword">if</span> content[<span class="number">2</span>] == <span class="string">&quot;transcript&quot;</span> <span class="keyword">or</span> content[<span class="number">2</span>] == <span class="string">&quot;mRNA&quot;</span>:</span><br><span class="line"></span><br><span class="line">            <span class="comment"># use regual expression to extract the gene ID</span></span><br><span class="line">            <span class="comment"># match the pattern ID=xxxxx; or ID=xxxxx</span></span><br><span class="line"></span><br><span class="line">            tx_id = re.search(<span class="string">r&#x27;ID=(.*?)[;\n]&#x27;</span>,content[<span class="number">8</span>]).group(<span class="number">1</span>)</span><br><span class="line">            tx_parent = re.search(<span class="string">r&#x27;Parent=(.*?)[;\n]&#x27;</span>,content[<span class="number">8</span>]).group(<span class="number">1</span>)</span><br><span class="line">            tx_parent = tx_parent.strip() <span class="comment"># remove the &#x27;r&#x27; and &#x27;\n&#x27;</span></span><br><span class="line"></span><br><span class="line">            <span class="comment"># if the parent of transcript is not in the gene_dict, create it rather than append</span></span><br><span class="line">            <span class="keyword">if</span> tx_parent <span class="keyword">in</span> gene_dict:</span><br><span class="line">                gene_dict[tx_parent].append(tx_id)</span><br><span class="line">            <span class="keyword">else</span>:</span><br><span class="line">                gene_dict[tx_parent] = [tx_id]</span><br><span class="line">            tx_pos_dict[tx_id] = [content[<span class="number">0</span>],content[<span class="number">3</span>], content[<span class="number">4</span>], content[<span class="number">6</span>] ]</span><br><span class="line">        <span class="comment"># GFF must have CDS feature</span></span><br><span class="line">        <span class="keyword">if</span> content[<span class="number">2</span>] == <span class="string">&#x27;CDS&#x27;</span>:</span><br><span class="line">            width = <span class="built_in">int</span>(content[<span class="number">4</span>]) - <span class="built_in">int</span>(content[<span class="number">3</span>])</span><br><span class="line">            CDS_parent = re.search(<span class="string">r&#x27;Parent=(.*?)[;\n]&#x27;</span>,content[<span class="number">8</span>]).group(<span class="number">1</span>)</span><br><span class="line">            CDS_parent = CDS_parent.strip() <span class="comment"># strip the &#x27;\r&#x27; and &#x27;\n&#x27;</span></span><br><span class="line">            CDS_dict[CDS_parent].append(width)</span><br><span class="line">    handle.close()</span><br><span class="line">    <span class="keyword">return</span> [gene_dict, tx_pos_dict, CDS_dict]</span><br><span class="line"></span><br><span class="line"><span class="keyword">if</span> __name__ == <span class="string">&quot;__main__&quot;</span>:</span><br><span class="line">    parser = get_parser()</span><br><span class="line">    args = parser.parse_args()</span><br><span class="line">    fa_dict = get_fasta_len( args.fasta)</span><br><span class="line">    gene_dict, tx_pos_dict, CDS_dict= parse_gff( args.gff )</span><br><span class="line">    gene_count = &#123;&#125;</span><br><span class="line"></span><br><span class="line">    <span class="comment"># outfile</span></span><br><span class="line">    len_file = <span class="built_in">open</span>(args.prefix + <span class="string">&quot;.len&quot;</span>, <span class="string">&quot;w&quot;</span>)</span><br><span class="line">    gff_file = <span class="built_in">open</span>(args.prefix + <span class="string">&quot;.gff&quot;</span>, <span class="string">&quot;w&quot;</span>)</span><br><span class="line"></span><br><span class="line">    <span class="keyword">for</span> gene, txs <span class="keyword">in</span> gene_dict.items():</span><br><span class="line">        tmp = <span class="number">0</span></span><br><span class="line">        <span class="keyword">for</span> tx <span class="keyword">in</span> txs:</span><br><span class="line">            tx_len = <span class="built_in">sum</span>(CDS_dict[tx])</span><br><span class="line">            <span class="keyword">if</span> tx_len &gt; tmp:</span><br><span class="line">                lst_tx = tx</span><br><span class="line">                tmp = tx_len</span><br><span class="line">        tx_chrom = tx_pos_dict[lst_tx][<span class="number">0</span>]</span><br><span class="line">        <span class="keyword">if</span> tx_chrom <span class="keyword">not</span> <span class="keyword">in</span> gene_count:</span><br><span class="line">            gene_count[tx_chrom] = <span class="number">1</span></span><br><span class="line">        <span class="keyword">else</span>:</span><br><span class="line">            gene_count[tx_chrom] += <span class="number">1</span></span><br><span class="line">        tx_start = tx_pos_dict[lst_tx][<span class="number">1</span>]</span><br><span class="line">        tx_end   = tx_pos_dict[lst_tx][<span class="number">2</span>]</span><br><span class="line">        tx_strand = tx_pos_dict[lst_tx][<span class="number">3</span>]</span><br><span class="line"></span><br><span class="line">        print(<span class="string">&quot;&#123;chrom&#125;\t&#123;gene&#125;\t&#123;start&#125;\t&#123;end&#125;\t&#123;strand&#125;\t&#123;order&#125;\t&#123;tx&#125;&quot;</span>.<span class="built_in">format</span>(\</span><br><span class="line">                chrom=tx_chrom,gene=gene,start=tx_start,end=tx_end,strand=tx_strand,order=gene_count[tx_chrom],tx=lst_tx), file=gff_file )</span><br><span class="line"></span><br><span class="line">    <span class="keyword">for</span> chrom,lens <span class="keyword">in</span> fa_dict.items():</span><br><span class="line">        print(<span class="string">&quot;&#123;chrom&#125;\t&#123;lens&#125;\t&#123;count&#125;&quot;</span>.<span class="built_in">format</span>(\</span><br><span class="line">                chrom=chrom,lens=lens,count=gene_count[chrom]), file=len_file)</span><br><span class="line">    len_file.close()</span><br><span class="line">    gff_file.close()</span><br></pre></td></tr></table></figure>
<h1 id="参考文献">参考文献</h1>
<p>Sun P, Jiao B, Yang Y, et al. WGDI: A user-friendly toolkit for evolutionary analyses of whole-genome duplications and ancestral karyotypes[J]. bioRxiv, 2021.</p>
</article><div class="post-copyright"><div class="post-copyright__author"><span class="post-copyright-meta">文章作者: </span><span class="post-copyright-info"><a href="mailto:lixingzee@gmail.com" rel="external nofollow noreferrer">李星泽</a></span></div><div class="post-copyright__type"><span class="post-copyright-meta">文章链接: </span><span class="post-copyright-info"><a href="http://lxz9.com/2021/09/06/wgdi/">http://lxz9.com/2021/09/06/wgdi/</a></span></div><div class="post-copyright__notice"><span class="post-copyright-meta">版权声明: </span><span class="post-copyright-info">本博客所有文章除特别声明外，均采用 <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" rel="external nofollow noreferrer" target="_blank">CC BY-NC-SA 4.0</a> 许可协议。转载请注明来自 <a href="http://lxz9.com" target="_blank">生信技术</a>！</span></div></div><div class="tag_share"><div class="post-meta__tag-list"><a class="post-meta__tags" href="/tags/%E6%95%99%E7%A8%8B/">教程</a><a class="post-meta__tags" href="/tags/WGD/">WGD</a><a class="post-meta__tags" href="/tags/WGDI/">WGDI</a></div><div class="post_share"><div class="social-share" data-image="https://z3.ax1x.com/2021/09/06/h4Hmo6.jpg" data-sites="facebook,twitter,wechat,weibo,qq"></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/social-share.js/dist/css/share.min.css" media="print" onload="this.media='all'"><script src="https://cdn.jsdelivr.net/npm/social-share.js/dist/js/social-share.min.js" defer></script></div></div><div class="post-reward"><div class="reward-button"><i class="fas fa-qrcode"></i> 打赏<div class="reward-main"><ul class="reward-all"><li class="reward-item"><a href="https://z3.ax1x.com/2021/01/19/sgZ5y6.jpg" rel="external nofollow noreferrer" target="_blank"><img class="post-qr-code-img" src="https://z3.ax1x.com/2021/01/19/sgZ5y6.jpg" alt="微信"/></a><div class="post-qr-code-desc">微信</div></li><li class="reward-item"><a href="https://z3.ax1x.com/2021/01/19/sgZIOK.jpg" rel="external nofollow noreferrer" target="_blank"><img class="post-qr-code-img" src="https://z3.ax1x.com/2021/01/19/sgZIOK.jpg" alt="支付宝"/></a><div class="post-qr-code-desc">支付宝</div></li></ul></div></div></div><nav class="pagination-post" id="pagination"><div class="prev-post pull-left"><a href="/2021/10/01/Perl01/"><img class="prev-cover" src="https://z3.ax1x.com/2021/10/01/4TX9AI.jpg" onerror="onerror=null;src='/img/404.jpg'" alt="cover of previous post"><div class="pagination-info"><div class="label">上一篇</div><div class="prev_info">Perl语言基础(一)</div></div></a></div><div class="next-post pull-right"><a href="/2021/08/31/lrzsz/"><img class="next-cover" src="https://z3.ax1x.com/2021/08/31/haMYQJ.png" onerror="onerror=null;src='/img/404.jpg'" alt="cover of next post"><div class="pagination-info"><div class="label">下一篇</div><div class="next_info">lrzsz(Linux服务器和Mac互传文件工具)</div></div></a></div></nav><div class="relatedPosts"><div class="headline"><i class="fas fa-thumbs-up fa-fw"></i><span> 相关推荐</span></div><div class="relatedPosts-list"><div><a href="/2021/07/18/3d-DNA/" title="3d-DNA的使用及juicebox调整挂载到染色体水平 ｜ HiC辅助基因组组装（二）"><img class="cover" src="https://z3.ax1x.com/2021/07/18/W8VU5F.jpg" alt="cover"><div class="content is-center"><div class="date"><i class="far fa-calendar-alt fa-fw"></i> 2021-07-18</div><div class="title">3d-DNA的使用及juicebox调整挂载到染色体水平 ｜ HiC辅助基因组组装（二）</div></div></a></div><div><a href="/2021/08/05/ALLHIC/" title="ALLHIC使用 ｜ HiC辅助基因组组装（三）"><img class="cover" src="https://z3.ax1x.com/2021/08/05/feYdmD.png" alt="cover"><div class="content is-center"><div class="date"><i class="far fa-calendar-alt fa-fw"></i> 2021-08-05</div><div class="title">ALLHIC使用 ｜ HiC辅助基因组组装（三）</div></div></a></div><div><a href="/2021/02/08/Annoation/" title="如何进行基因组注释"><img class="cover" src="https://z3.ax1x.com/2021/02/08/yapB1H.jpg" alt="cover"><div class="content is-center"><div class="date"><i class="far fa-calendar-alt fa-fw"></i> 2021-02-08</div><div class="title">如何进行基因组注释</div></div></a></div><div><a href="/2021/07/18/HiC-Pro/" title="HiC-Pro的使用 ｜ HiC辅助基因组组装（一）"><img class="cover" src="https://z3.ax1x.com/2021/07/18/W8VNUU.jpg" alt="cover"><div class="content is-center"><div class="date"><i class="far fa-calendar-alt fa-fw"></i> 2021-07-18</div><div class="title">HiC-Pro的使用 ｜ HiC辅助基因组组装（一）</div></div></a></div><div><a href="/2021/04/03/HiC/" title="Hi-C辅助基因组组装原理｜主流软件"><img class="cover" src="https://z3.ax1x.com/2021/04/03/cuFUdH.jpg" alt="cover"><div class="content is-center"><div class="date"><i class="far fa-calendar-alt fa-fw"></i> 2021-04-03</div><div class="title">Hi-C辅助基因组组装原理｜主流软件</div></div></a></div><div><a href="/2021/01/19/TransDecoder/" title="转录组分析流程｜TransDecoder预测转录本的开放阅读框（二）"><img class="cover" src="https://z3.ax1x.com/2021/01/24/sb4KRf.jpg" alt="cover"><div class="content is-center"><div class="date"><i class="far fa-calendar-alt fa-fw"></i> 2021-01-19</div><div class="title">转录组分析流程｜TransDecoder预测转录本的开放阅读框（二）</div></div></a></div></div></div><hr/><div id="post-comment"><div class="comment-head"><div class="comment-headline"><i class="fas fa-comments fa-fw"></i><span> 评论</span></div></div><div class="comment-wrap"><div><div class="vcomment" id="vcomment"></div></div></div></div></div><div class="aside-content" id="aside-content"><div class="card-widget card-info"><div class="card-info-avatar is-center"><img class="avatar-img" src="https://z3.ax1x.com/2021/03/10/6JSyoq.png" onerror="this.onerror=null;this.src='/img/friend_404.gif'" alt="avatar"/><div class="author-info__name">李星泽</div><div class="author-info__description"></div></div><div class="card-info-data"><div class="card-info-data-item is-center"><a href="/archives/"><div class="headline">文章</div><div class="length-num">32</div></a></div><div class="card-info-data-item is-center"><a href="/tags/"><div class="headline">标签</div><div class="length-num">46</div></a></div><div class="card-info-data-item is-center"><a href="/categories/"><div class="headline">分类</div><div class="length-num">14</div></a></div></div><a class="button--animated" id="card-info-btn" target="_blank" rel="noopener external nofollow noreferrer" href="https://github.com/xxxxxx"><i class="fab fa-github"></i><span>Follow Me</span></a><div class="card-info-social-icons is-center"><a class="social-icon" href="https://blog.csdn.net/m0_49960764" rel="external nofollow noreferrer" target="_blank" title="CSDN"><i class="iconfont icon-csdn"></i></a><a class="social-icon" href="https://z3.ax1x.com/2021/11/25/oAI0h9.jpg" rel="external nofollow noreferrer" target="_blank" title="微信公众号"><i class="iconfont icon-weixin"></i></a><a class="social-icon" href="https://space.bilibili.com/113291922?from=search&amp;seid=3526734520747213504&amp;spm_id_from=333.337.0.0" rel="external nofollow noreferrer" target="_blank" title="bilibili"><i class="iconfont icon-bilibili-copy"></i></a><a class="social-icon" href="mailto:lixingzee@gmail.com" rel="external nofollow noreferrer" target="_blank" title="Email"><i class="fas fa-envelope"></i></a></div></div><div class="card-widget card-announcement"><div class="item-headline"><i class="fas fa-bullhorn card-announcement-animation"></i><span>公告</span></div><div class="announcement_content">如果你有更多想了解的内容，欢迎关注我的个人公众号：<b>生信技术</b> <img src="https://s4.ax1x.com/2021/12/23/TGLCTS.png"> <br> <br></div><timing></timing></div><div class="sticky_layout"><div class="card-widget" id="card-toc"><div class="item-headline"><i class="fas fa-stream"></i><span>目录</span></div><div class="toc-content"><ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#%E7%AE%80%E4%BB%8B"><span class="toc-number">1.</span> <span class="toc-text">简介</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E5%AE%89%E8%A3%85"><span class="toc-number">2.</span> <span class="toc-text">安装</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E4%BE%9D%E8%B5%96%E8%BD%AF%E4%BB%B6"><span class="toc-number">3.</span> <span class="toc-text">依赖软件</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E5%88%86%E6%9E%90"><span class="toc-number">4.</span> <span class="toc-text">分析</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#%E6%95%B0%E6%8D%AE%E9%A2%84%E5%A4%84%E7%90%86"><span class="toc-number">4.1.</span> <span class="toc-text">数据预处理</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E5%BA%8F%E5%88%97%E6%AF%94%E5%AF%B9"><span class="toc-number">4.2.</span> <span class="toc-text">序列比对</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E7%82%B9%E5%9B%BE"><span class="toc-number">4.3.</span> <span class="toc-text">点图</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#%E4%BF%AE%E6%94%B9%E9%85%8D%E7%BD%AE%E6%96%87%E4%BB%B6"><span class="toc-number">4.3.1.</span> <span class="toc-text">修改配置文件</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E5%85%B1%E7%BA%BF%E6%80%A7"><span class="toc-number">4.4.</span> <span class="toc-text">共线性</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Ks%E5%80%BC%E8%AE%A1%E7%AE%97"><span class="toc-number">4.5.</span> <span class="toc-text">Ks值计算</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Blockinfo"><span class="toc-number">4.6.</span> <span class="toc-text">Blockinfo</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Correspondence"><span class="toc-number">4.7.</span> <span class="toc-text">Correspondence</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#blockks"><span class="toc-number">4.8.</span> <span class="toc-text">blockks</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#KsPeaks"><span class="toc-number">4.9.</span> <span class="toc-text">KsPeaks</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#PeaksFit"><span class="toc-number">4.10.</span> <span class="toc-text">PeaksFit</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E8%84%9A%E6%9C%AC"><span class="toc-number">5.</span> <span class="toc-text">脚本</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#getgff-py"><span class="toc-number">5.1.</span> <span class="toc-text">getgff.py</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#gff-lens-py"><span class="toc-number">5.2.</span> <span class="toc-text">gff_lens.py</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#seq-newname-py"><span class="toc-number">5.3.</span> <span class="toc-text">seq_newname.py</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#generate-conf-py"><span class="toc-number">5.4.</span> <span class="toc-text">generate_conf.py</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E5%8F%82%E8%80%83%E6%96%87%E7%8C%AE"><span class="toc-number">6.</span> <span class="toc-text">参考文献</span></a></li></ol></div></div><div class="card-widget card-recent-post"><div class="item-headline"><i class="fas fa-history"></i><span>最新文章</span></div><div class="aside-list"><div class="aside-list-item"><a class="thumbnail" href="/2021/12/21/pypubmed/" title="PyPubMed｜一款进行文献检索的工具"><img src="https://s4.ax1x.com/2021/12/21/TKoMjA.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="PyPubMed｜一款进行文献检索的工具"/></a><div class="content"><a class="title" href="/2021/12/21/pypubmed/" title="PyPubMed｜一款进行文献检索的工具">PyPubMed｜一款进行文献检索的工具</a><time datetime="2021-12-21T03:00:00.000Z" title="发表于 2021-12-21 11:00:00">2021-12-21</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2021/12/15/Conserved_Sequences/" title="保守序列的生物学意义｜用 K-mers 寻找保守序列"><img src="https://s4.ax1x.com/2021/12/15/ozGUE9.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="保守序列的生物学意义｜用 K-mers 寻找保守序列"/></a><div class="content"><a class="title" href="/2021/12/15/Conserved_Sequences/" title="保守序列的生物学意义｜用 K-mers 寻找保守序列">保守序列的生物学意义｜用 K-mers 寻找保守序列</a><time datetime="2021-12-15T10:00:00.000Z" title="发表于 2021-12-15 18:00:00">2021-12-15</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2021/12/14/R/" title="R语言基础画图教程合集(长文总结)"><img src="https://s4.ax1x.com/2021/12/15/ozG2Ed.jpg" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="R语言基础画图教程合集(长文总结)"/></a><div class="content"><a class="title" href="/2021/12/14/R/" title="R语言基础画图教程合集(长文总结)">R语言基础画图教程合集(长文总结)</a><time datetime="2021-12-14T10:00:00.000Z" title="发表于 2021-12-14 18:00:00">2021-12-14</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2021/12/01/Protein_structure_prediction/" title="使用同源建模预测蛋白质结构"><img src="https://z3.ax1x.com/2021/12/02/oYdAsA.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="使用同源建模预测蛋白质结构"/></a><div class="content"><a class="title" href="/2021/12/01/Protein_structure_prediction/" title="使用同源建模预测蛋白质结构">使用同源建模预测蛋白质结构</a><time datetime="2021-12-01T15:00:00.000Z" title="发表于 2021-12-01 23:00:00">2021-12-01</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2021/11/25/python/" title="Python 学习教程总结"><img src="https://z3.ax1x.com/2021/11/26/oElpDJ.jpg" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="Python 学习教程总结"/></a><div class="content"><a class="title" href="/2021/11/25/python/" title="Python 学习教程总结">Python 学习教程总结</a><time datetime="2021-11-24T16:00:00.000Z" title="发表于 2021-11-25 00:00:00">2021-11-25</time></div></div></div></div></div></div></main><footer id="footer"><div id="footer-wrap"><div class="copyright">&copy;2020 - 2021&nbsp;<i style="color:#FF6A6A;animation: announ_animation 0.8s linear infinite;"class="fas fa-heart"></i> By 李星泽</div></div><div class="container" id="jsi-flying-fish-container"><script src="/js/fish.js"></script></div><style>   @media only screen and (max-width: 767px){
   #sidebar_search_box input[type=text]{width:calc(100% - 24px)}
}</style></footer></div><div id="rightside"><div id="rightside-config-hide"><button id="readmode" type="button" title="阅读模式"><i class="fas fa-book-open"></i></button><button id="darkmode" type="button" title="浅色和深色模式转换"><i class="fas fa-adjust"></i></button><button id="hide-aside-btn" type="button" title="单栏和双栏切换"><i class="fas fa-arrows-alt-h"></i></button></div><div id="rightside-config-show"><button id="rightside_config" type="button" title="设置"><i class="fas fa-cog fa-spin"></i></button><button class="close" id="mobile-toc-button" type="button" title="目录"><i class="fas fa-list-ul"></i></button><a id="to_comment" href="#post-comment" title="直达评论"><i class="fas fa-comments"></i></a><button id="go-up" type="button" title="回到顶部"><i class="fas fa-arrow-up"></i></button></div></div><div id="local-search"><div class="search-dialog"><div class="search-dialog__title" id="local-search-title">本地搜索</div><div id="local-input-panel"><div id="local-search-input"><div class="local-search-box"><input class="local-search-box--input" placeholder="搜索文章" type="text"/></div></div></div><hr/><div id="local-search-results"></div><span class="search-close-button"><i class="fas fa-times"></i></span></div><div id="search-mask"></div></div><div><script src="/js/utils.js"></script><script src="/js/main.js"></script><script src="https://cdn.jsdelivr.net/npm/medium-zoom/dist/medium-zoom.min.js"></script><script src="/js/search/local-search.js"></script><div class="js-pjax"><script>function loadValine () {
  function initValine () {
    let initData = {
      el: '#vcomment',
      appId: '',
      appKey: '',
      placeholder: 'Please leave your footprints',
      avatar: 'monsterid',
      meta: 'nick,mail,link'.split(','),
      pageSize: '10',
      lang: 'en',
      recordIP: false,
      serverURLs: '',
      emojiCDN: '',
      emojiMaps: "",
      enableQQ: false,
      path: window.location.pathname,
    }

    if (true) { 
      initData.requiredFields= ('nick,mail'.split(','))
    }
    
    if (false) {
      const otherData = false
      initData = Object.assign(initData, otherData)
    }
    
    const valine = new Valine(initData)
  }

  if (typeof Valine === 'function') initValine() 
  else getScript('https://cdn.jsdelivr.net/npm/valine/dist/Valine.min.js').then(initValine)
}

if ('Valine' === 'Valine' || !true) {
  if (true) btf.loadComment(document.getElementById('vcomment'),loadValine)
  else setTimeout(loadValine, 0)
} else {
  function loadOtherComment () {
    loadValine()
  }
}</script><script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script></div><script src="/js/jquery.min.js"></script><script src="/js/fish.js"></script><script src="/js/timing.js"></script><script async src="https://cdn.jsdelivr.net/npm/aplayer"></script><script async src="https://cdn.jsdelivr.net/npm/meting/dist/Meting.min.js"></script><div class="aplayer no-destroy" data-id="000PeZCQ1i4XVs" data-server="tencent" data-type="artist" data-fixed="true" data-mini="true" data-listFolded="false" data-order="random" data-preload="none" data-autoplay="flase" muted></div><script defer="defer" id="fluttering_ribbon" mobile="false" src="https://cdn.jsdelivr.net/npm/butterfly-extsrc@1/dist/canvas-fluttering-ribbon.min.js"></script><script id="canvas_nest" defer="defer" color="0,0,255" opacity="0.7" zIndex="-1" count="99" mobile="false" src="https://cdn.jsdelivr.net/npm/butterfly-extsrc@1/dist/canvas-nest.min.js"></script><script>(function(i,s,o,g,r,a,m){i["DaoVoiceObject"]=r;i[r]=i[r]||function(){(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;a.charset="utf-8";m.parentNode.insertBefore(a,m)})(window,document,"script",('https:' == document.location.protocol ? 'https:' : 'http:') + "//widget.daovoice.io/widget/f84bbafb.js","daovoice")
</script><script>var isChatBtn = false
daovoice('init', {
  app_id: 'f84bbafb',},{
  launcher: { 
     disableLauncherIcon: isChatBtn // 悬浮 ICON 是否显示
  },
});
daovoice('update');

if (isChatBtn) {
  var chatBtnFn = () => {
    var chatBtn = document.getElementById("chat_btn")
    chatBtn.addEventListener("click", function(){
      daovoice('show')
    });
  }
  chatBtnFn()
} else {
  if (false) {
    function chatBtnHide () {
      daovoice('update', {},{
        launcher: { 
        disableLauncherIcon: true // 悬浮 ICON 是否显示
        },
      });
    }
    function chatBtnShow () {
      daovoice('update', {},{
        launcher: { 
        disableLauncherIcon: false // 悬浮 ICON 是否显示
        },
      });
    }
  }
}</script><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/aplayer/dist/APlayer.min.css" media="print" onload="this.media='all'"><script src="https://cdn.jsdelivr.net/npm/aplayer/dist/APlayer.min.js"></script><script src="https://cdn.jsdelivr.net/gh/metowolf/MetingJS@1.2/dist/Meting.min.js"></script><script src="https://cdn.jsdelivr.net/npm/pjax/pjax.min.js"></script><script>let pjaxSelectors = [
  'title',
  '#config_change',
  '#body-wrap',
  '#rightside-config-hide',
  '#rightside-config-show',
  '.js-pjax'
]

if (false) {
  pjaxSelectors.unshift('meta[property="og:image"]', 'meta[property="og:title"]', 'meta[property="og:url"]')
}

var pjax = new Pjax({
  elements: 'a:not([target="_blank"])',
  selectors: pjaxSelectors,
  cacheBust: false,
  analytics: true,
  scrollRestoration: false
})

document.addEventListener('pjax:complete', function () {
  window.refreshFn()

  document.querySelectorAll('script[data-pjax]').forEach(item => {
    const newScript = document.createElement('script')
    const content = item.text || item.textContent || item.innerHTML || ""
    Array.from(item.attributes).forEach(attr => newScript.setAttribute(attr.name, attr.value))
    newScript.appendChild(document.createTextNode(content))
    item.parentNode.replaceChild(newScript, item)
  })

  GLOBAL_CONFIG.islazyload && window.lazyLoadInstance.update()

  typeof chatBtnFn === 'function' && chatBtnFn()
  typeof panguInit === 'function' && panguInit()

  if (typeof gtag === 'function') {
    gtag('config', 'G-QRG4NL4207', {'page_path': window.location.pathname});
  }

  typeof loadMeting === 'function' && document.getElementsByClassName('aplayer').length && loadMeting()

  // Analytics
  if (false) {
    MtaH5.pgv()
  }

  // prismjs
  typeof Prism === 'object' && Prism.highlightAll()

  typeof preloader === 'object' && preloader.endLoading()
})


document.addEventListener('pjax:send', function () {
  typeof preloader === 'object' && preloader.initLoading()
  
  if (window.aplayers) {
    for (let i = 0; i < window.aplayers.length; i++) {
      if (!window.aplayers[i].options.fixed) {
        window.aplayers[i].destroy()
      }
    }
  }

  typeof typed === 'object' && typed.destroy()

  //reset readmode
  const $bodyClassList = document.body.classList
  $bodyClassList.contains('read-mode') && $bodyClassList.remove('read-mode')

})</script><script>(function(){
  const bp = document.createElement('script');
  const curProtocol = window.location.protocol.split(':')[0];
  if (curProtocol === 'https') {
    bp.src = 'https://zz.bdstatic.com/linksubmit/push.js';
  }
  else{
    bp.src = 'http://push.zhanzhang.baidu.com/push.js';
  }
  bp.dataset.pjax = ''
  const s = document.getElementsByTagName("script")[0];
  s.parentNode.insertBefore(bp, s);
})()</script></div></body></html>