<!DOCTYPE html><html lang="zh-CN" data-theme="light"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width,initial-scale=1"><title>Gviz - 实现基因组数据可视化 | 生信技术</title><meta name="keywords" content="Gviz"><meta name="author" content="李星泽,lixingzee@gmail.com"><meta name="copyright" content="李星泽"><meta name="format-detection" content="telephone=no"><meta name="theme-color" content="#ffffff"><meta http-equiv="Cache-Control" content="no-transform"><meta http-equiv="Cache-Control" content="no-siteapp"><meta name="description" content="Gviz包介绍 Gviz软件包简介 Gviz软件包旨在提供一个结构化的可视化框架，以沿着基因组坐标绘制任何类型的数据。它还允许整合来自UCSC或ENSEMBL等来源的公开基因组注释数据。 与大多数基因组浏览器一样，单独类型的基因组特征或数据由单独的Track表示。  默认情况下，Gviz 检查所有提供的染色体名称在 UCSC 上的有效性（染色体必须以 chr 字符串开头）。可以决定通过调用 opt">
<meta property="og:type" content="article">
<meta property="og:title" content="Gviz - 实现基因组数据可视化">
<meta property="og:url" content="http://lxz9.com/2022/01/05/Gviz/index.html">
<meta property="og:site_name" content="生信技术">
<meta property="og:description" content="Gviz包介绍 Gviz软件包简介 Gviz软件包旨在提供一个结构化的可视化框架，以沿着基因组坐标绘制任何类型的数据。它还允许整合来自UCSC或ENSEMBL等来源的公开基因组注释数据。 与大多数基因组浏览器一样，单独类型的基因组特征或数据由单独的Track表示。  默认情况下，Gviz 检查所有提供的染色体名称在 UCSC 上的有效性（染色体必须以 chr 字符串开头）。可以决定通过调用 opt">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="https://s4.ax1x.com/2022/01/05/TvKUv6.png">
<meta property="article:published_time" content="2022-01-05T15:00:00.000Z">
<meta property="article:modified_time" content="2022-01-05T13:30:16.000Z">
<meta property="article:author" content="李星泽">
<meta property="article:tag" content="Gviz">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://s4.ax1x.com/2022/01/05/TvKUv6.png"><link rel="shortcut icon" href="https://z3.ax1x.com/2021/02/09/yawIzV.png"><link rel="canonical" href="http://lxz9.com/2022/01/05/Gviz/"><link rel="preconnect" href="//cdn.jsdelivr.net"/><link rel="preconnect" href="//www.google-analytics.com" crossorigin=""/><link rel="preconnect" href="//hm.baidu.com"/><link rel="preconnect" href="//busuanzi.ibruce.info"/><link rel="preconnect" href="//zz.bdstatic.com"/><link rel="stylesheet" href="/css/index.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free/css/all.min.css" media="print" onload="this.media='all'"><script>var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "https://hm.baidu.com/hm.js?72a556f842c8990127f397d4e0c54f2c";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script><script async="async" src="https://www.googletagmanager.com/gtag/js?id=G-QRG4NL4207"></script><script>window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-QRG4NL4207');
</script><script>var GLOBAL_CONFIG = { 
  root: '/',
  algolia: undefined,
  localSearch: {"path":"search.xml","languages":{"hits_empty":"找不到您查询的内容：${query}"}},
  translate: undefined,
  noticeOutdate: undefined,
  highlight: {"plugin":"highlighjs","highlightCopy":true,"highlightLang":true},
  copy: {
    success: '复制成功',
    error: '复制错误',
    noSupport: '浏览器不支持'
  },
  relativeDate: {
    homepage: false,
    post: false
  },
  runtime: '',
  date_suffix: {
    just: '刚刚',
    min: '分钟前',
    hour: '小时前',
    day: '天前',
    month: '个月前'
  },
  copyright: undefined,
  lightbox: 'mediumZoom',
  Snackbar: undefined,
  source: {
    jQuery: 'https://cdn.jsdelivr.net/npm/jquery@latest/dist/jquery.min.js',
    justifiedGallery: {
      js: 'https://cdn.jsdelivr.net/npm/justifiedGallery/dist/js/jquery.justifiedGallery.min.js',
      css: 'https://cdn.jsdelivr.net/npm/justifiedGallery/dist/css/justifiedGallery.min.css'
    },
    fancybox: {
      js: 'https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.js',
      css: 'https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.css'
    }
  },
  isPhotoFigcaption: false,
  islazyload: false,
  isanchor: false
};

var saveToLocal = {
  set: function setWithExpiry(key, value, ttl) {
    const now = new Date()
    const expiryDay = ttl * 86400000
    const item = {
      value: value,
      expiry: now.getTime() + expiryDay,
    }
    localStorage.setItem(key, JSON.stringify(item))
  },

  get: function getWithExpiry(key) {
    const itemStr = localStorage.getItem(key)

    if (!itemStr) {
      return undefined
    }
    const item = JSON.parse(itemStr)
    const now = new Date()

    if (now.getTime() > item.expiry) {
      localStorage.removeItem(key)
      return undefined
    }
    return item.value
  }
}

// https://stackoverflow.com/questions/16839698/jquery-getscript-alternative-in-native-javascript
const getScript = url => new Promise((resolve, reject) => {
  const script = document.createElement('script')
  script.src = url
  script.async = true
  script.onerror = reject
  script.onload = script.onreadystatechange = function() {
    const loadState = this.readyState
    if (loadState && loadState !== 'loaded' && loadState !== 'complete') return
    script.onload = script.onreadystatechange = null
    resolve()
  }
  document.head.appendChild(script)
})</script><script id="config_change">var GLOBAL_CONFIG_SITE = { 
  isPost: true,
  isHome: false,
  isHighlightShrink: false,
  isToc: true,
  postUpdate: '2022-01-05 21:30:16'
}</script><noscript><style type="text/css">
  #nav {
    opacity: 1
  }
  .justified-gallery img {
    opacity: 1
  }

  #recent-posts time,
  #post-meta time {
    display: inline !important
  }
</style></noscript><script>(function () {  window.activateDarkMode = function () {
    document.documentElement.setAttribute('data-theme', 'dark')
    if (document.querySelector('meta[name="theme-color"]') !== null) {
      document.querySelector('meta[name="theme-color"]').setAttribute('content', '#0d0d0d')
    }
  }
  window.activateLightMode = function () {
    document.documentElement.setAttribute('data-theme', 'light')
   if (document.querySelector('meta[name="theme-color"]') !== null) {
      document.querySelector('meta[name="theme-color"]').setAttribute('content', '#ffffff')
    }
  }
  const autoChangeMode = 'false'
  const t = saveToLocal.get('theme')
  if (autoChangeMode === '1') {
    const isDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches
    const isLightMode = window.matchMedia('(prefers-color-scheme: light)').matches
    const isNotSpecified = window.matchMedia('(prefers-color-scheme: no-preference)').matches
    const hasNoSupport = !isDarkMode && !isLightMode && !isNotSpecified
    if (t === undefined) {
      if (isLightMode) activateLightMode()
      else if (isDarkMode) activateDarkMode()
      else if (isNotSpecified || hasNoSupport) {
        const now = new Date()
        const hour = now.getHours()
        const isNight = hour <= 6 || hour >= 18
        isNight ? activateDarkMode() : activateLightMode()
      }
      window.matchMedia('(prefers-color-scheme: dark)').addListener(function (e) {
        if (saveToLocal.get('theme') === undefined) {
          e.matches ? activateDarkMode() : activateLightMode()
        }
      })
    } else if (t === 'light') activateLightMode()
    else activateDarkMode()
  } else if (autoChangeMode === '2') {
    const now = new Date()
    const hour = now.getHours()
    const isNight = hour <= 6 || hour >= 18
    if (t === undefined) isNight ? activateDarkMode() : activateLightMode()
    else if (t === 'light') activateLightMode()
    else activateDarkMode()
  } else {
    if (t === 'dark') activateDarkMode()
    else if (t === 'light') activateLightMode()
  }const asideStatus = saveToLocal.get('aside-status')
if (asideStatus !== undefined) {
   if (asideStatus === 'hide') {
     document.documentElement.classList.add('hide-aside')
   } else {
     document.documentElement.classList.remove('hide-aside')
   }
}})()</script><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/aplayer/dist/APlayer.min.css"  media="defer" onload="this.media='all'"><meting-js server='tencent' type='playlist' id='2134037073' fixed='true'></meting-js><style>#article-container.post-content h1:before, h2:before, h3:before, h4:before, h5:before, h6:before { -webkit-animation: avatar_turn_around 1s linear infinite; -moz-animation: avatar_turn_around 1s linear infinite; -o-animation: avatar_turn_around 1s linear infinite; -ms-animation: avatar_turn_around 1s linear infinite; animation: avatar_turn_around 1s linear infinite; }</style><link rel="stylesheet" href="/css/iconfont.css"><link rel="stylesheet" href="/css/background.css"><link rel="stylesheet" href="/css/lxz9.css"><link rel="stylesheet" href="/css/icon.css" media="defer" onload="this.media='all'"><!-- hexo injector head_end start -->
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.12.0/dist/katex.min.css">

<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-math@4.0.0/dist/style.css">
<!-- hexo injector head_end end --><meta name="generator" content="Hexo 5.3.0">
<style>.github-emoji { position: relative; display: inline-block; width: 1.2em; min-height: 1.2em; overflow: hidden; vertical-align: top; color: transparent; }  .github-emoji > span { position: relative; z-index: 10; }  .github-emoji img, .github-emoji .fancybox { margin: 0 !important; padding: 0 !important; border: none !important; outline: none !important; text-decoration: none !important; user-select: none !important; cursor: auto !important; }  .github-emoji img { height: 1.2em !important; width: 1.2em !important; position: absolute !important; left: 50% !important; top: 50% !important; transform: translate(-50%, -50%) !important; user-select: none !important; cursor: auto !important; } .github-emoji-fallback { color: inherit; } .github-emoji-fallback img { opacity: 0 !important; }</style>
<link rel="alternate" href="/atom.xml" title="生信技术" type="application/atom+xml">
</head><body><div id="sidebar"><div id="menu-mask"></div><div id="sidebar-menus"><div class="author-avatar"><img class="avatar-img" src="https://z3.ax1x.com/2021/03/10/6JSyoq.png" onerror="onerror=null;src='/img/friend_404.gif'" alt="avatar"/></div><div class="site-data"><div class="data-item is-center"><div class="data-item-link"><a href="/archives/"><div class="headline">文章</div><div class="length-num">42</div></a></div></div><div class="data-item is-center"><div class="data-item-link"><a href="/tags/"><div class="headline">标签</div><div class="length-num">51</div></a></div></div><div class="data-item is-center"><div class="data-item-link"><a href="/categories/"><div class="headline">分类</div><div class="length-num">14</div></a></div></div></div><hr/><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fa fa-home"></i><span> 首页</span></a></div><div class="menus_item"><a class="site-page" href="/archives/"><i class="fa-fw fa fa-archive"></i><span> 时间轴</span></a></div><div class="menus_item"><a class="site-page" href="/tags/"><i class="fa-fw fa fa-tags"></i><span> 标签</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fa fa-folder-open"></i><span> 分类</span></a></div><div class="menus_item"><a class="site-page" href="javascript:void(0);" rel="external nofollow noreferrer"><i class="fa-fw fas fa-list"></i><span> 清单</span><i class="fas fa-chevron-down expand"></i></a><ul class="menus_item_child"><li><a class="site-page" href="/music/"><i class="fa-fw fas fa-music"></i><span> 音乐</span></a></li><li><a class="site-page" href="/Gallery/"><i class="fa-fw fas fa-images"></i><span> 照片</span></a></li><li><a class="site-page" href="/movies/"><i class="fa-fw fas fa-video"></i><span> 电影</span></a></li></ul></div><div class="menus_item"><a class="site-page" href="/link/"><i class="fa-fw fas fa-link"></i><span> 友链</span></a></div><div class="menus_item"><a class="site-page" href="/map/"><i class="fa-fw fas fa-heart"></i><span> 地图</span></a></div></div></div></div><div class="post" id="body-wrap"><header class="post-bg" id="page-header" style="background-image: url(https://s4.ax1x.com/2022/01/05/TvKUv6.png)"><nav id="nav"><span id="blog_name"><a id="site-name" href="/">生信技术</a></span><div id="menus"><div id="search-button"><a class="site-page social-icon search"><i class="fas fa-search fa-fw"></i><span> 搜索</span></a></div><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fa fa-home"></i><span> 首页</span></a></div><div class="menus_item"><a class="site-page" href="/archives/"><i class="fa-fw fa fa-archive"></i><span> 时间轴</span></a></div><div class="menus_item"><a class="site-page" href="/tags/"><i class="fa-fw fa fa-tags"></i><span> 标签</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fa fa-folder-open"></i><span> 分类</span></a></div><div class="menus_item"><a class="site-page" href="javascript:void(0);" rel="external nofollow noreferrer"><i class="fa-fw fas fa-list"></i><span> 清单</span><i class="fas fa-chevron-down expand"></i></a><ul class="menus_item_child"><li><a class="site-page" href="/music/"><i class="fa-fw fas fa-music"></i><span> 音乐</span></a></li><li><a class="site-page" href="/Gallery/"><i class="fa-fw fas fa-images"></i><span> 照片</span></a></li><li><a class="site-page" href="/movies/"><i class="fa-fw fas fa-video"></i><span> 电影</span></a></li></ul></div><div class="menus_item"><a class="site-page" href="/link/"><i class="fa-fw fas fa-link"></i><span> 友链</span></a></div><div class="menus_item"><a class="site-page" href="/map/"><i class="fa-fw fas fa-heart"></i><span> 地图</span></a></div></div><div id="toggle-menu"><a class="site-page"><i class="fas fa-bars fa-fw"></i></a></div></div></nav><div id="post-info"><h1 class="post-title">Gviz - 实现基因组数据可视化</h1><div id="post-meta"><div class="meta-firstline"><span class="post-meta-date"><i class="far fa-calendar-alt fa-fw post-meta-icon"></i><span class="post-meta-label">发表于</span><time class="post-meta-date-created" datetime="2022-01-05T15:00:00.000Z" title="发表于 2022-01-05 23:00:00">2022-01-05</time><span class="post-meta-separator">|</span><i class="fas fa-history fa-fw post-meta-icon"></i><span class="post-meta-label">更新于</span><time class="post-meta-date-updated" datetime="2022-01-05T13:30:16.000Z" title="更新于 2022-01-05 21:30:16">2022-01-05</time></span><span class="post-meta-categories"><span class="post-meta-separator">|</span><i class="fas fa-inbox fa-fw post-meta-icon"></i><a class="post-meta-categories" href="/categories/R%E8%AF%AD%E8%A8%80/">R语言</a></span></div><div class="meta-secondline"><span class="post-meta-separator">|</span><span class="post-meta-wordcount"><i class="far fa-file-word fa-fw post-meta-icon"></i><span class="post-meta-label">字数总计:</span><span class="word-count">4.1k</span><span class="post-meta-separator">|</span><i class="far fa-clock fa-fw post-meta-icon"></i><span class="post-meta-label">阅读时长:</span><span>17分钟</span></span><span class="post-meta-separator">|</span><span class="post-meta-pv-cv"><i class="far fa-eye fa-fw post-meta-icon"></i><span class="post-meta-label">阅读量:</span><span id="busuanzi_value_page_pv"></span></span></div></div></div></header><main class="layout" id="content-inner"><div id="post"><article class="post-content" id="article-container"><h1 id="Gviz包介绍">Gviz包介绍</h1>
<p><strong>Gviz软件包简介</strong></p>
<p>Gviz软件包旨在提供一个结构化的可视化框架，以沿着基因组坐标绘制任何类型的数据。它还允许整合来自<em>UCSC</em>或<em>ENSEMBL</em>等来源的公开基因组注释数据。</p>
<p>与大多数基因组浏览器一样，单独类型的基因组特征或数据由单独的<em>Track</em>表示。</p>
<blockquote>
<p>默认情况下，Gviz 检查所有提供的染色体名称在 <em>UCSC</em> 上的有效性（染色体必须以 chr 字符串开头）。<br>可以决定通过调用 <em>options(ucscChromosomeNames=FALSE)</em> 来关此功能</p>
</blockquote>
<p><strong>在以下示例中，将利用小鼠 mm9 基因组上的 <em>UCSC</em> 基因组和 7 号染色体 (chr7)</strong></p>
<h2 id="绘制-AnnotationTrack-图">绘制 <em>AnnotationTrack</em> 图</h2>
<p>请注意，<em><strong>AnnotationTrack</strong></em> 构造函数可以容纳许多不同类型的输入。</p>
<p>例如，注释功能的开始和结束坐标可以作为单个参数<br>
start和end作为 data.frame 或甚至作为 IRanges 或 GRangesList 对象。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br></pre></td><td class="code"><pre><span class="line">library(Gviz)</span><br><span class="line">library(GenomicRanges)</span><br><span class="line"><span class="comment">#加载数据 : class = GRanges</span></span><br><span class="line">data(cpgIslands)</span><br><span class="line">cpgIslands</span><br><span class="line"></span><br><span class="line"><span class="comment">## GRanges object with 10 ranges and 0 metadata columns:</span></span><br><span class="line"><span class="comment">##        seqnames            ranges strand</span></span><br><span class="line"><span class="comment">##           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt;</span></span><br><span class="line"><span class="comment">##    [1]     chr7 26549019-26550183      *</span></span><br><span class="line"><span class="comment">##    [2]     chr7 26564119-26564500      *</span></span><br><span class="line"><span class="comment">##    [3]     chr7 26585667-26586158      *</span></span><br><span class="line"><span class="comment">##    [4]     chr7 26591772-26593309      *</span></span><br><span class="line"><span class="comment">##    [5]     chr7 26594192-26594570      *</span></span><br><span class="line"><span class="comment">##    [6]     chr7 26623835-26624150      *</span></span><br><span class="line"><span class="comment">##    [7]     chr7 26659284-26660352      *</span></span><br><span class="line"><span class="comment">##    [8]     chr7 26721294-26721717      *</span></span><br><span class="line"><span class="comment">##    [9]     chr7 26821518-26823297      *</span></span><br><span class="line"><span class="comment">##   [10]     chr7 26991322-26991841      *</span></span><br><span class="line"><span class="comment">##   -------</span></span><br><span class="line"><span class="comment">##   seqinfo: 1 sequence from hg19 genome; no seqlengths</span></span><br></pre></td></tr></table></figure>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#Annotation track, 标题命名为&quot;CpG&quot;</span></span><br><span class="line">atrack &lt;- AnnotationTrack(cpgIslands, name = <span class="string">&quot;CpG&quot;</span>)</span><br><span class="line">p&lt;-plotTracks(atrack)</span><br></pre></td></tr></table></figure>
<p><br><img src="https://s4.ax1x.com/2022/01/04/TOOfoR.png" alt="G-1"></p>
<h2 id="添加基因组坐标">添加基因组坐标</h2>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">## 基因组坐标</span></span><br><span class="line">gtrack &lt;- GenomeAxisTrack()</span><br><span class="line">plotTracks(<span class="built_in">list</span>(gtrack, atrack))</span><br></pre></td></tr></table></figure>
<p><br><img src="https://s4.ax1x.com/2022/01/04/TOO4F1.png" alt="G-2"></p>
<h2 id="添加染色体表意文字">添加染色体表意文字</h2>
<p>要添加染色体表意文字 (<em>ideogram</em>)，我们必须指明有效的 <em>UCSC</em> 基因组（例如：“hg19”）以及染色体名称（例如：“chr7”）</p>
<blockquote>
<p>由于该功能从 <em>UCSC</em> 获取数据，因此需要 <em>Internet</em> 连接，这可能需要很长时间。</p>
</blockquote>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># genome : &quot;hg19&quot; </span></span><br><span class="line">gen&lt;-genome(cpgIslands)</span><br><span class="line"><span class="comment"># Chromosme name : &quot;chr7&quot;</span></span><br><span class="line">chr &lt;- <span class="built_in">as.character</span>(unique(seqnames(cpgIslands)))</span><br><span class="line"><span class="comment"># Ideogram track</span></span><br><span class="line">itrack &lt;- IdeogramTrack(genome = gen, chromosome = chr)</span><br><span class="line">plotTracks(<span class="built_in">list</span>(itrack, gtrack, atrack))</span><br></pre></td></tr></table></figure>
<p><br><img src="https://s4.ax1x.com/2022/01/04/TOOIW6.png" alt="G-3"></p>
<blockquote>
<p><strong>表意文字轨迹( <em>ideogram track</em> )</strong> 是所有Gviz轨迹对象中的一个例外，因为它们不像所有其他轨迹那样真正显示在同一坐标系上。<br>取而代之的是，当前的基因组位置在染色体上用一个<strong>红色方框表示</strong>（或者，在本例中，如果宽度太小而无法容纳方框，则用红线表示）。</p>
</blockquote>
<h2 id="添加基因模型">添加基因模型</h2>
<p>我们可以利用现有本地来源的基因模型信息。或者，可以从可用的在线资源之一（如 <em>UCSC</em> 或 <em>ENSEBML</em>）下载此类数据，并且有构造函数来处理这些任务。</p>
<p>对于这个例子，我们将从存储的 <code>data.frame</code> 加载基因模型数据。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 加载数据</span></span><br><span class="line">data(geneModels)</span><br><span class="line">head(geneModels)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 画图</span></span><br><span class="line">grtrack &lt;- GeneRegionTrack(geneModels, genome = gen,</span><br><span class="line">                           chromosome = chr, name = <span class="string">&quot;Gene Model&quot;</span>)</span><br><span class="line">plotTracks(<span class="built_in">list</span>(itrack, gtrack, atrack, grtrack))</span><br></pre></td></tr></table></figure>
<p><br><img src="https://s4.ax1x.com/2022/01/04/TOO5Jx.png" alt="G-4"></p>
<h2 id="放大绘图">放大绘图</h2>
<p>我们经常想<strong>放大或缩小特定的绘图区域</strong>以查看更多细节或获得更广泛的概览。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 使用 from 和 to 参数进行缩放</span></span><br><span class="line">plotTracks(<span class="built_in">list</span>(itrack, gtrack, atrack, grtrack),</span><br><span class="line">           from = <span class="number">26700000</span>, to = <span class="number">26750000</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 使用 extend.left 和 extend.right 进行缩放</span></span><br><span class="line"><span class="comment"># 这些参数是相对于当前显示的范围，</span></span><br><span class="line"><span class="comment"># 并且可用于在绘图的一端或两端快速扩展视图。</span></span><br><span class="line">plotTracks(<span class="built_in">list</span>(itrack, gtrack, atrack, grtrack),</span><br><span class="line">           extend.left = <span class="number">0.5</span>, extend.right = <span class="number">500000</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 删除外显子的边界,更好的展示图片</span></span><br><span class="line">plotTracks(<span class="built_in">list</span>(itrack, gtrack, atrack, grtrack),</span><br><span class="line">           extend.left = <span class="number">0.5</span>, extend.right = <span class="number">500000</span>, col = <span class="literal">NULL</span>)</span><br></pre></td></tr></table></figure>
<p><br><img src="https://s4.ax1x.com/2022/01/05/TXG9Ig.png" alt="G-5-6-7"></p>
<blockquote>
<p>值为 0.5 时将放大到当前显示范围的一半</p>
</blockquote>
<h2 id="添加序列track并放大查看序列">添加序列track并放大查看序列</h2>
<p>必要的序列信息来自 <strong>BSgenome 包</strong></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">if</span> (!requireNamespace(<span class="string">&quot;BiocManager&quot;</span>, quietly = <span class="literal">TRUE</span>))</span><br><span class="line">    install.packages(<span class="string">&quot;BiocManager&quot;</span>)</span><br><span class="line">BiocManager::install(<span class="string">&quot;BSgenome.Hsapiens.UCSC.hg19&quot;</span>)</span><br><span class="line"></span><br><span class="line">library(BSgenome.Hsapiens.UCSC.hg19)</span><br><span class="line">strack &lt;- SequenceTrack(Hsapiens, chromosome = chr)</span><br><span class="line">plotTracks(<span class="built_in">list</span>(itrack, gtrack, atrack, grtrack,</span><br><span class="line">                strack), from = <span class="number">26591822</span>, to = <span class="number">26591852</span>, cex = <span class="number">0.8</span>)</span><br></pre></td></tr></table></figure>
<p><br><img src="https://s4.ax1x.com/2022/01/04/TOOLeH.png" alt="G-8"></p>
<h2 id="添加data-track">添加data track</h2>
<p><strong>DataTrack</strong> 对象本质上是运行长度编码 (Rle) 数字向量或矩阵，我们可以使用它们将各种数字数据添加到我们的基因组坐标图中。</p>
<p>这些 track 的不同可视化选项可用，包括点图、直方图和盒须图表示。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 出于演示目的，我们可以从随机采样的数据创建一个简单的DataTrack对象</span></span><br><span class="line">set.seed(<span class="number">255</span>)</span><br><span class="line">lim &lt;- <span class="built_in">c</span>(<span class="number">26700000</span>, <span class="number">26750000</span>)</span><br><span class="line">coords &lt;- sort(<span class="built_in">c</span>(lim[<span class="number">1</span>], sample(seq(from = lim[<span class="number">1</span>], </span><br><span class="line">                                    to = lim[<span class="number">2</span>]), <span class="number">99</span>), lim[<span class="number">2</span>]))</span><br><span class="line">dat &lt;- runif(<span class="number">100</span>, <span class="built_in">min</span> = -<span class="number">10</span>, <span class="built_in">max</span> = <span class="number">10</span>)</span><br><span class="line">head(dat)</span><br></pre></td></tr></table></figure>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">##data track</span></span><br><span class="line">dtrack &lt;- DataTrack(data = dat, start = coords[-<span class="built_in">length</span>(coords)],</span><br><span class="line">                    end = coords[-<span class="number">1</span>], chromosome = chr, genome = gen,</span><br><span class="line">                    name = <span class="string">&quot;Uniform&quot;</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 绘制 data track 图</span></span><br><span class="line">plotTracks(<span class="built_in">list</span>(itrack, gtrack, atrack, grtrack,</span><br><span class="line">                dtrack), from = lim[<span class="number">1</span>], to = lim[<span class="number">2</span>])</span><br><span class="line"></span><br><span class="line"><span class="comment"># 将绘图类型更改为直方图</span></span><br><span class="line">plotTracks(<span class="built_in">list</span>(itrack, gtrack, atrack, grtrack,dtrack),</span><br><span class="line">           from = lim[<span class="number">1</span>], to = lim[<span class="number">2</span>], type = <span class="string">&quot;histogram&quot;</span>)</span><br></pre></td></tr></table></figure>
<p><br><img src="https://s4.ax1x.com/2022/01/04/TOOvFI.png" alt="G-9-10"></p>
<blockquote>
<p>这种可视化在显示例如沿染色体的 NGS reads 的覆盖范围时特别有用，或显示来自微阵列实验的映射探针的测量值。</p>
</blockquote>
<h1 id="绘图参数">绘图参数</h1>
<h2 id="设置参数">设置参数</h2>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#Annotation of transcript</span></span><br><span class="line"><span class="comment"># 更改面板和标题背景颜色</span></span><br><span class="line">grtrack &lt;- GeneRegionTrack(geneModels, genome = gen,</span><br><span class="line">                           chromosome = chr, name = <span class="string">&quot;Gene Model&quot;</span>, </span><br><span class="line">                           transcriptAnnotation = <span class="string">&quot;symbol&quot;</span>,</span><br><span class="line">                           background.panel = <span class="string">&quot;#dbeeff&quot;</span>,</span><br><span class="line">                           background.title = <span class="string">&quot;darkblue&quot;</span>)</span><br><span class="line">plotTracks(<span class="built_in">list</span>(itrack, gtrack, atrack, grtrack))</span><br></pre></td></tr></table></figure>
<p><br><img src="https://s4.ax1x.com/2022/01/04/TOOOwd.png" alt="G-11"></p>
<h2 id="绘图方向">绘图方向</h2>
<p>默认情况下，所有轨迹都将以 <strong>5’ -&gt; 3’</strong> 方向绘制。有时实际显示相对于相反链的数据会很有用。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">plotTracks(<span class="built_in">list</span>(itrack, gtrack, atrack, grtrack),</span><br><span class="line">           reverseStrand = <span class="literal">TRUE</span>)</span><br></pre></td></tr></table></figure>
<p><br><img src="https://s4.ax1x.com/2022/01/04/TOOxYt.png" alt="G-12"></p>
<blockquote>
<p>数据已绘制在反向链上这也反映在 <strong>GenomeAxis track</strong> 中。</p>
</blockquote>
<h1 id="Track-classes">Track classes</h1>
<p>多个参数可用于更改不同轨道的外观。</p>
<h2 id="GenomeAxisTrack-的显示参数">GenomeAxisTrack 的显示参数</h2>
<p>将标签的位置设置为下方，显示 ID，更改颜色</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">axisTrack &lt;- GenomeAxisTrack(<span class="built_in">range</span> = IRanges(start = <span class="built_in">c</span>(<span class="number">2000000</span>,<span class="number">4000000</span>), end = <span class="built_in">c</span>(<span class="number">3000000</span>, <span class="number">7000000</span>), <span class="built_in">names</span> = <span class="built_in">rep</span>(<span class="string">&quot;N-stretch&quot;</span>, <span class="number">2</span>)))</span><br><span class="line">plotTracks(axisTrack, from = <span class="number">1000000</span>, to = <span class="number">9000000</span>, </span><br><span class="line">           labelPos = <span class="string">&quot;below&quot;</span>,showId=<span class="literal">TRUE</span>, col=<span class="string">&quot;#9400D3&quot;</span>)</span><br></pre></td></tr></table></figure>
<p><br><img src="https://s4.ax1x.com/2022/01/04/TOOXTA.png" alt="G-13"></p>
<h2 id="IdeogramTrack">IdeogramTrack</h2>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#Ideogram</span></span><br><span class="line">ideoTrack &lt;- IdeogramTrack(genome = <span class="string">&quot;hg19&quot;</span>, chromosome = <span class="string">&quot;chrX&quot;</span>)</span><br><span class="line">plotTracks(ideoTrack, from = <span class="number">85000000</span>, to = <span class="number">129000000</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment">#显示染色体条带 ID</span></span><br><span class="line">plotTracks(ideoTrack, from = <span class="number">85000000</span>, to = <span class="number">129000000</span>,</span><br><span class="line">           showId = <span class="literal">FALSE</span>, showBandId = <span class="literal">TRUE</span>, cex.bands = <span class="number">0.4</span>)</span><br></pre></td></tr></table></figure>
<p><br><img src="https://s4.ax1x.com/2022/01/04/TOOzfP.png" alt="G-14-15"></p>
<h2 id="DataTrack">DataTrack</h2>
<p>本质上，它们构成了与特定基因组坐标范围相关联的运行长度编码的数字向量或矩阵。</p>
<p>单个数据集中可以有多个样本，绘图方法提供了合并样本组信息的工具。</p>
<p>因此，创建 DataTrack 对象的起点将始终是一组范围，以 IRanges 或 GRanges 对象的形式，或单独作为开始和结束坐标或宽度。第二个成分是长度与范围数相同的数值向量，或具有相同列数的数值矩阵。</p>
<p><strong>接下来将从作为 Gviz 包的一部分的 GRanges 对象加载我们的示例数据</strong></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 加载数据</span></span><br><span class="line">data(twoGroups)</span><br><span class="line">head(twoGroups)</span><br></pre></td></tr></table></figure>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># data track 图（ DataTrack 的默认可视化是点图 ）</span></span><br><span class="line"> dTrack &lt;- DataTrack(twoGroups, name = <span class="string">&quot;uniform&quot;</span>)</span><br><span class="line"> plotTracks(dTrack)</span><br></pre></td></tr></table></figure>
<p><br><img src="https://s4.ax1x.com/2022/01/04/TOXpSf.png" alt="G-16"></p>
<h2 id="不同图片类型">不同图片类型</h2>
<p><strong>可能的绘图类型是：</strong></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 点图</span></span><br><span class="line">plotTracks(DataTrack(twoGroups, name = <span class="string">&quot;p&quot;</span>), type=<span class="string">&quot;p&quot;</span>)</span><br><span class="line"><span class="comment"># 线图</span></span><br><span class="line">plotTracks(DataTrack(twoGroups, name = <span class="string">&quot;l&quot;</span>), type=<span class="string">&quot;l&quot;</span>)</span><br><span class="line"><span class="comment"># 线图和点图</span></span><br><span class="line">plotTracks(DataTrack(twoGroups, name = <span class="string">&quot;b&quot;</span>), type=<span class="string">&quot;b&quot;</span>)</span><br><span class="line"><span class="comment"># 平均线图</span></span><br><span class="line">plotTracks(DataTrack(twoGroups, name = <span class="string">&quot;a&quot;</span>), type=<span class="string">&quot;a&quot;</span>)</span><br><span class="line"><span class="comment"># 直方线图</span></span><br><span class="line">plotTracks(DataTrack(twoGroups, name = <span class="string">&quot;h&quot;</span>), type=<span class="string">&quot;h&quot;</span>)</span><br><span class="line"><span class="comment"># 直方图（条宽等于范围）</span></span><br><span class="line">plotTracks(DataTrack(twoGroups, name = <span class="string">&quot;histogram&quot;</span>), type=<span class="string">&quot;histogram&quot;</span>)</span><br><span class="line"><span class="comment"># &#x27;polygon-type&#x27; 图</span></span><br><span class="line">plotTracks(DataTrack(twoGroups, name = <span class="string">&quot;polygon&quot;</span>), type=<span class="string">&quot;polygon&quot;</span>)</span><br><span class="line"><span class="comment"># 盒须图</span></span><br><span class="line">plotTracks(DataTrack(twoGroups, name = <span class="string">&quot;boxplot&quot;</span>), type=<span class="string">&quot;boxplot&quot;</span>)</span><br><span class="line"><span class="comment"># 各个值的彩色图像</span></span><br><span class="line">plotTracks(DataTrack(twoGroups, name = <span class="string">&quot;heatmap&quot;</span>), type=<span class="string">&quot;heatmap&quot;</span>)</span><br></pre></td></tr></table></figure>
<p><br><img src="https://s4.ax1x.com/2022/01/04/TOOTSK.png" alt="G01-09"></p>
<p><strong>DataTrack 图示例：</strong></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 将箱线图与平均线和数据网格 (g) 结合起来：</span></span><br><span class="line">plotTracks(dTrack, type = <span class="built_in">c</span>(<span class="string">&quot;boxplot&quot;</span>, <span class="string">&quot;a&quot;</span>, <span class="string">&quot;g&quot;</span>))</span><br><span class="line"><span class="comment"># 热图并显示样品名称</span></span><br><span class="line">plotTracks(dTrack, type = <span class="built_in">c</span>(<span class="string">&quot;heatmap&quot;</span>), showSampleNames = <span class="literal">TRUE</span>,</span><br><span class="line">           cex.sampleNames = <span class="number">0.6</span>)</span><br></pre></td></tr></table></figure>
<p><br><img src="https://s4.ax1x.com/2022/01/04/TOXFmQ.png" alt="G-17-18"></p>
<h2 id="数据分组">数据分组</h2>
<p>可以使用因子变量将单个样本组合在一起。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">plotTracks(dTrack, groups = <span class="built_in">rep</span>(<span class="built_in">c</span>(<span class="string">&quot;control&quot;</span>, <span class="string">&quot;treated&quot;</span>), each = <span class="number">3</span>),</span><br><span class="line">           type = <span class="built_in">c</span>(<span class="string">&quot;a&quot;</span>, <span class="string">&quot;p&quot;</span>), legend=<span class="literal">TRUE</span>)</span><br><span class="line"><span class="comment"># 箱形图</span></span><br><span class="line">plotTracks(dTrack, groups = <span class="built_in">rep</span>(<span class="built_in">c</span>(<span class="string">&quot;control&quot;</span>, <span class="string">&quot;treated&quot;</span>), each = <span class="number">3</span>),</span><br><span class="line">           type = <span class="string">&quot;boxplot&quot;</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 聚合组, 聚合可以是平均值、中值、极值、总和、最小值和最大值</span></span><br><span class="line">plotTracks(dTrack, groups = <span class="built_in">rep</span>(<span class="built_in">c</span>(<span class="string">&quot;control&quot;</span>, <span class="string">&quot;treated&quot;</span>),each = <span class="number">3</span>),</span><br><span class="line">           type = <span class="built_in">c</span>(<span class="string">&quot;b&quot;</span>), aggregateGroups = <span class="literal">TRUE</span>,</span><br><span class="line">           aggregation = <span class="string">&quot;max&quot;</span>)</span><br></pre></td></tr></table></figure>
<p><br><img src="https://s4.ax1x.com/2022/01/04/TOXVkn.png" alt="G-19-20-21"></p>
<h3 id="从文件构建-DataTrack-对象">从文件构建 DataTrack 对象</h3>
<p><em><strong>DataTrack</strong></em> 类支持最常见的文件类型，如 <em>wig、bigWig、bedGraph</em> 和 <em>bam</em> 文件</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">bgFile &lt;- system.file(<span class="string">&quot;extdata/test.bedGraph&quot;</span>, package = <span class="string">&quot;Gviz&quot;</span>)</span><br><span class="line">dTrack2 &lt;- DataTrack(<span class="built_in">range</span> = bgFile, genome = <span class="string">&quot;hg19&quot;</span>,</span><br><span class="line">                     type = <span class="string">&quot;l&quot;</span>, chromosome = <span class="string">&quot;chr19&quot;</span>, name = <span class="string">&quot;bedGraph&quot;</span>)</span><br><span class="line">plotTracks(dTrack2)</span><br></pre></td></tr></table></figure>
<blockquote>
<p>请注意， Gviz 包使用 <strong>rtracklayer 包</strong>中的功能进行大多数文件导入操作</p>
</blockquote>
<p>Gviz 包中文件支持来自索引文件的流式传输。<br>在绘图操作期间只需要加载数据的相关部分，因此底层数据文件可能非常大而不会降低性能或导致太大的内存占用。</p>
<p>我们将在此使用随包提供的小 bam 文件来举例说明此功能。</p>
<p>bam 文件包含序列与共同参考的比对。DataTrack 中此类数据的最自然表示是仅查看给定位置的对齐覆盖率，并将其编码在单个 <em>elementMetadata</em> 列中。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">bamFile &lt;- system.file(<span class="string">&quot;extdata/test.bam&quot;</span>, package = <span class="string">&quot;Gviz&quot;</span>)</span><br><span class="line">dTrack4 &lt;- DataTrack(<span class="built_in">range</span> = bamFile, genome = <span class="string">&quot;hg19&quot;</span>,</span><br><span class="line">                     type = <span class="string">&quot;l&quot;</span>, name = <span class="string">&quot;Coverage&quot;</span>, window = -<span class="number">1</span>, chromosome = <span class="string">&quot;chr1&quot;</span>)</span><br><span class="line">plotTracks(dTrack4, from = <span class="number">189990000</span>, to = <span class="number">190000000</span>)</span><br></pre></td></tr></table></figure>
<p><br><img src="https://s4.ax1x.com/2022/01/04/TOX9l8.png" alt="G-22"></p>
<h2 id="AnnotationTrack">AnnotationTrack</h2>
<p>本质上，它们由一个或多个基因组范围组成，如果需要，可以将这些范围分组为复合注释元素。</p>
<p>必要的构建块是范围坐标、染色体和基因组标识符。</p>
<p>信息可以以单独的函数参数的形式传递给函数，如 IRanges、GRanges 或 data.frame 对象。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">aTrack &lt;- AnnotationTrack(start = <span class="built_in">c</span>(<span class="number">10</span>, <span class="number">40</span>, <span class="number">120</span>),</span><br><span class="line">                          width = <span class="number">15</span>, chromosome = <span class="string">&quot;chrX&quot;</span>, strand = <span class="built_in">c</span>(<span class="string">&quot;+&quot;</span>,<span class="string">&quot;*&quot;</span>, <span class="string">&quot;-&quot;</span>),</span><br><span class="line">                          id = <span class="built_in">c</span>(<span class="string">&quot;Huey&quot;</span>, <span class="string">&quot;Dewey&quot;</span>, <span class="string">&quot;Louie&quot;</span>),</span><br><span class="line">                          genome = <span class="string">&quot;hg19&quot;</span>, name = <span class="string">&quot;foo&quot;</span>)</span><br><span class="line">plotTracks(aTrack)</span><br></pre></td></tr></table></figure>
<p><br><img src="https://s4.ax1x.com/2022/01/04/TOXC6S.png" alt="G-23"></p>
<h3 id="从文件构建-AnnotationTrack-对象">从文件构建 AnnotationTrack 对象</h3>
<p>默认的导入函数从 bam 文件中读取所有序列比对的坐标。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#Annotation track</span></span><br><span class="line">aTrack2 &lt;- AnnotationTrack(<span class="built_in">range</span> = bamFile, genome = <span class="string">&quot;hg19&quot;</span>,</span><br><span class="line">                           name = <span class="string">&quot;Reads&quot;</span>, chromosome = <span class="string">&quot;chr1&quot;</span>)</span><br><span class="line">plotTracks(aTrack2, from = <span class="number">189995000</span>, to = <span class="number">190000000</span>)</span><br></pre></td></tr></table></figure>
<p><br><img src="https://s4.ax1x.com/2022/01/04/TOXkwj.png" alt="G-24"></p>
<p>我们现在可以同时绘制 bam 文件的 DataTrack 和 AnnotationTrack 表示，以证明基础数据确实相同。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">plotTracks(<span class="built_in">list</span>(dTrack4, aTrack2), from = <span class="number">189990000</span>,</span><br><span class="line">           to = <span class="number">190000000</span>)</span><br></pre></td></tr></table></figure>
<p><br><img src="https://s4.ax1x.com/2022/01/04/TOXATs.png" alt="G-25"></p>
<h2 id="GeneRegionTrack">GeneRegionTrack</h2>
<p><strong>GeneRegionTrack</strong> 原则上与 <strong>AnnotationTrack</strong> 非常相似。</p>
<p>唯一的区别是它们更以基因/转录本为中心。我们需要传递track中每个注释特征的开始和结束位置（或宽度），并为每个项目提供外显子、转录本和基因标识符，这些标识符将用于创建转录本分组。</p>
<blockquote>
<p>一个有点特殊的情况是直接从 <em>TranscriptDb</em> 之一构建 <em>GeneRegionTrack</em>，下面将更详细地介绍该选项。</p>
</blockquote>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">data(geneModels)</span><br><span class="line">grtrack &lt;- GeneRegionTrack(geneModels, genome = gen,</span><br><span class="line">                           chromosome = chr, name = <span class="string">&quot;foo&quot;</span>, </span><br><span class="line">                           transcriptAnnotation = <span class="string">&quot;symbol&quot;</span>)</span><br></pre></td></tr></table></figure>
<h3 id="从-fTranscriptDbsx-构建-GeneRegionTrack-对象">从 <em>fTranscriptDbsx</em> 构建 <em>GeneRegionTrack</em> 对象</h3>
<p><strong>GenomicFeatures 包</strong>提供了一个框架，可以从在线资源下载基因模型信息并将其本地存储在 <em>SQLite</em> 数据库中。</p>
<p>从 <em>TranscriptDb</em> 构建 <em>GeneRegionTracks</em> 的一个好处是我们获得了关于转录本编码和非编码区域的额外信息，即 5’ 和 3’ UTR 以及 CDS 区域的坐标。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">library(GenomicFeatures)</span><br><span class="line">samplefile &lt;- system.file(<span class="string">&quot;extdata&quot;</span>, <span class="string">&quot;UCSC_knownGene_sample.sqlite&quot;</span>,</span><br><span class="line">    package = <span class="string">&quot;GenomicFeatures&quot;</span>)</span><br><span class="line">txdb &lt;- loadDb(samplefile)</span><br><span class="line">txTr &lt;- GeneRegionTrack(txdb, chromosome = <span class="string">&quot;chr6&quot;</span>, start = <span class="number">300000</span>, end = <span class="number">350000</span>)</span><br><span class="line"><span class="comment">#feature(txTr)</span></span><br><span class="line">plotTracks(txTr)</span><br></pre></td></tr></table></figure>
<h2 id="BiomartGeneRegionTrack"><em><strong>BiomartGeneRegionTrack</strong></em></h2>
<p><em><strong>BiomartGeneRegionTrack</strong></em> ，提供到 <em>ENSEMBL Biomart</em> 服务的直接接口。</p>
<p>我们只需输入基因组、染色体以及该染色体上的开始和结束位置，构造函数 <em>BiomartGeneRegionTrack</em> 将自动联系 <em>ENSEMBL</em>，获取必要的信息并动态构建基因模型。</p>
<blockquote>
<p>需要互联网连接才能工作，并且根据使用情况和网络流量，联系 <em>Biomart</em> 可能需要大量时间。</p>
</blockquote>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">biomTrack &lt;- BiomartGeneRegionTrack(genome = <span class="string">&quot;hg19&quot;</span>,</span><br><span class="line">                                    chromosome = chr, start = <span class="number">20000000</span>, end = <span class="number">21000000</span>,</span><br><span class="line">                                    name = <span class="string">&quot;ENSEMBL&quot;</span>)</span><br><span class="line">plotTracks(biomTrack, col.line = <span class="literal">NULL</span>, col = <span class="literal">NULL</span>)</span><br></pre></td></tr></table></figure>
<p><br><img src="https://s4.ax1x.com/2022/01/05/TX8VED.png" alt="G-26"></p>
<h2 id="Sequence-Track">Sequence Track</h2>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">library(BSgenome.Hsapiens.UCSC.hg19)</span><br><span class="line">sTrack &lt;- SequenceTrack(Hsapiens)</span><br><span class="line"><span class="comment">#sequence track : add 5&#x27;-&gt;3&#x27;</span></span><br><span class="line">plotTracks(sTrack, chromosome = <span class="number">1</span>, from = <span class="number">20000</span>,to = <span class="number">20050</span>,</span><br><span class="line">           add53=<span class="literal">TRUE</span>)</span><br><span class="line"><span class="comment">#The complement</span></span><br><span class="line">plotTracks(sTrack, chromosome = <span class="number">1</span>, from = <span class="number">20000</span>,to = <span class="number">20050</span>,</span><br><span class="line">           add53=<span class="literal">TRUE</span>, complement = <span class="literal">TRUE</span>)</span><br></pre></td></tr></table></figure>
<p><br><img src="https://s4.ax1x.com/2022/01/05/TX8ZUe.png" alt="G-27-28"></p>
<h2 id="AlignmentsTrack">AlignmentsTrack</h2>
<p>比对序列 (<em>Aligned sequences</em>) 的图，通常来自下一代测序实验，可能非常有用，例如在目视检查称为 SNP 的有效性时。这些比对通常存储在 BAM 文件中。</p>
<h3 id="RNA-seq">RNA-seq</h3>
<p>在本演示中，让我们使用一个小 BAM 文件，其中配对的 NGS reads 已映射到人类 hg19 基因组的提取物。</p>
<p>数据来自 RNA-Seq 实验，比对是使用 STAR 进行的，允许存在 gaps。</p>
<p>还从 <em>Biomart</em> 下载了该区域的一些基因注释数据。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">afrom=<span class="number">2960000</span></span><br><span class="line">ato=<span class="number">3160000</span></span><br><span class="line"><span class="comment">#bam file</span></span><br><span class="line">alTrack &lt;- AlignmentsTrack(system.file(package = <span class="string">&quot;Gviz&quot;</span>, <span class="string">&quot;extdata&quot;</span>, <span class="string">&quot;gapped.bam&quot;</span>), isPaired = <span class="literal">TRUE</span>)</span><br><span class="line">bmt &lt;- BiomartGeneRegionTrack(genome = <span class="string">&quot;hg19&quot;</span>, chromosome = <span class="string">&quot;chr12&quot;</span>,</span><br><span class="line">                              start = afrom, end = ato, filter = <span class="built_in">list</span>(with_ox_refseq_mrna = <span class="literal">TRUE</span>),</span><br><span class="line">                              stacking = <span class="string">&quot;dense&quot;</span>)</span><br><span class="line">plotTracks(<span class="built_in">c</span>(bmt, alTrack), from = afrom, to = ato, chromosome = <span class="string">&quot;chr12&quot;</span>)</span><br></pre></td></tr></table></figure>
<p><br><img src="https://s4.ax1x.com/2022/01/05/TX8e4H.png" alt="G-29"></p>
<blockquote>
<blockquote>
<p><strong>上图展示了 track 的总体布局</strong><br><br>
<strong>在顶部</strong>，我们有一个以柱状图形式显示reads覆盖率信息的面板，<strong>在该面板下方</strong>，是单个 reads 的堆积视图。<br>只有一定空间可用于绘图，此处无法显示整个堆积深度。其余由标题面板中的白色向下箭头表明。<br>我们可以通过使用<code>max.height</code>、<code>min.height</code>或<code>stackHeight</code>显示参数来解决这个问题，这些参数都控制堆叠 reads 的高度或垂直间距。或者，我们可以通过设置 <em>coverageHeight</em> 或 <em>minCoverageHeight</em> 参数来减小覆盖区域的大小。</p>
</blockquote>
</blockquote>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">plotTracks(<span class="built_in">c</span>(bmt, alTrack), from = afrom, to = ato,</span><br><span class="line">           chromosome = <span class="string">&quot;chr12&quot;</span>, min.height = <span class="number">0</span>, coverageHeight = <span class="number">0.08</span>,</span><br><span class="line">           minCoverageHeight = <span class="number">0</span>)</span><br></pre></td></tr></table></figure>
<p><br><img src="https://s4.ax1x.com/2022/01/05/TX8nCd.png" alt="G-30"></p>
<p>到目前为止，堆积 (pile-ups) 并不是特别有用，我们可以通过相应地设置类型显示参数来关闭它们。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">plotTracks(<span class="built_in">c</span>(alTrack, bmt), from = afrom, to = ato, chromosome = <span class="string">&quot;chr12&quot;</span>, type = <span class="string">&quot;coverage&quot;</span>)</span><br></pre></td></tr></table></figure>
<p><br><img src="https://s4.ax1x.com/2022/01/05/TX8AHO.png" alt="G-31"></p>
<p>进一步放大以查看堆积部分的详细信息：</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">plotTracks(<span class="built_in">c</span>(bmt, alTrack), from = afrom + <span class="number">12700</span>,</span><br><span class="line">           to = afrom + <span class="number">15200</span>, chromosome = <span class="string">&quot;chr12&quot;</span>)</span><br></pre></td></tr></table></figure>
<p><br><img src="https://s4.ax1x.com/2022/01/05/TX8u8A.png" alt="G-32"></p>
<blockquote>
<p>单个reads的方向由<strong>箭头指示</strong><br>read对由<strong>亮灰色线连接</strong><br>其中的 <em>gaps</em> 由连接的<strong>深灰色线显示</strong><br>在支持透明性的设备上，我们还可以看到一些 reads 实际上是重叠的。</p>
</blockquote>
<p>如前所述，我们可以通过在构造函数中设置 <strong>isPaired 参数</strong>来控制数据应该被视为双端数据还是单端数据。</p>
<p><strong>下面是我们在单端模式下查看同一文件中的数据</strong></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">alTrack &lt;- AlignmentsTrack(system.file(package = <span class="string">&quot;Gviz&quot;</span>,</span><br><span class="line">                                       <span class="string">&quot;extdata&quot;</span>, <span class="string">&quot;gapped.bam&quot;</span>), isPaired = <span class="literal">FALSE</span>)</span><br><span class="line">plotTracks(<span class="built_in">c</span>(bmt, alTrack), from = afrom + <span class="number">12700</span>,</span><br><span class="line">           to = afrom + <span class="number">15200</span>, chromosome = <span class="string">&quot;chr12&quot;</span>)</span><br></pre></td></tr></table></figure>
<p><br><img src="https://s4.ax1x.com/2022/01/05/TX8KgI.png" alt="G-33"></p>
<h3 id="DNA-seq">DNA-seq</h3>
<p>为了更好地展示 <em>AlignmentsTrack</em> 的序列变异特征，我们将加载不同的数据集，这次来自全基因组 <em>DNA-Seq SNP calling</em>。</p>
<p>同样，参考基因组是 hg19，并且使用 Bowtie2 进行了比对。</p>
<p>我们需要指定 <em>AlignmentsTrack</em> 关于参考基因组（<em>sequenceTrack</em>）。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">afrom &lt;- 44945200</span><br><span class="line">ato &lt;- 44947200</span><br><span class="line">alTrack &lt;- AlignmentsTrack(system.file(package = <span class="string">&quot;Gviz&quot;</span>,<span class="string">&quot;extdata&quot;</span>, <span class="string">&quot;snps.bam&quot;</span>), isPaired = <span class="literal">TRUE</span>)</span><br><span class="line">plotTracks(<span class="built_in">c</span>(alTrack, sTrack), chromosome = <span class="string">&quot;chr21&quot;</span>, from = afrom,to = ato)</span><br></pre></td></tr></table></figure>
<p><br><img src="https://s4.ax1x.com/2022/01/05/TX8Mvt.png" alt="G-34"></p>
<blockquote>
<p>不匹配的碱基以叠加直方图的形式显示在堆积部分和覆盖图中的单个read上。</p>
</blockquote>
<p>当放大到明显的杂合 SNP 位置之一时，可以揭示更多细节。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 放大</span></span><br><span class="line">plotTracks(<span class="built_in">c</span>(alTrack, sTrack), chromosome = <span class="string">&quot;chr21&quot;</span>, from = <span class="number">44946590</span>, to = <span class="number">44946660</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 显示单个字母</span></span><br><span class="line">plotTracks(<span class="built_in">c</span>(alTrack, sTrack), chromosome = <span class="string">&quot;chr21&quot;</span>,</span><br><span class="line">           from = <span class="number">44946590</span>, to = <span class="number">44946660</span>, cex = <span class="number">0.5</span>, min.height = <span class="number">8</span>)</span><br></pre></td></tr></table></figure>
<p><br><img src="https://s4.ax1x.com/2022/01/05/TX8lKP.png" alt="G-35-36"></p>
<h1 id="Track突出显示和叠加">Track突出显示和叠加</h1>
<h2 id="突出显示">突出显示</h2>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#highlight</span></span><br><span class="line">ht &lt;- HighlightTrack(trackList = <span class="built_in">list</span>(atrack, grtrack,dtrack),</span><br><span class="line">                     start = <span class="built_in">c</span>(<span class="number">26705000</span>, <span class="number">26720000</span>), width = <span class="number">7000</span>, chromosome = <span class="number">7</span>)</span><br><span class="line">plotTracks(<span class="built_in">list</span>(itrack, gtrack, ht), from = lim[<span class="number">1</span>], to = lim[<span class="number">2</span>])</span><br></pre></td></tr></table></figure>
<p><br><img src="https://s4.ax1x.com/2022/01/05/TX81Df.png" alt="G-37"></p>
<h2 id="叠加">叠加</h2>
<p>对于某些应用程序，在绘图的同一区域上叠加多个track 是有意义的。</p>
<p>为了举例，我们将生成第二个 <em>DataTrack</em> 对象并将其与第二章中的现有对象组合。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#create data</span></span><br><span class="line">dat &lt;- runif(<span class="number">100</span>, <span class="built_in">min</span> = -<span class="number">2</span>, <span class="built_in">max</span> = <span class="number">22</span>)</span><br><span class="line">dtrack2 &lt;- DataTrack(data = dat, start = coords[-<span class="built_in">length</span>(coords)],</span><br><span class="line">                     end = coords[-<span class="number">1</span>], chromosome = chr, genome = gen,</span><br><span class="line">                     name = <span class="string">&quot;Uniform2&quot;</span>, groups = factor(<span class="string">&quot;sample 2&quot;</span>,levels = <span class="built_in">c</span>(<span class="string">&quot;sample 1&quot;</span>, <span class="string">&quot;sample 2&quot;</span>)),</span><br><span class="line">                     legend = <span class="literal">TRUE</span>)</span><br><span class="line">displayPars(dtrack) &lt;- <span class="built_in">list</span>(groups = factor(<span class="string">&quot;sample 1&quot;</span>,levels = <span class="built_in">c</span>(<span class="string">&quot;sample 1&quot;</span>, <span class="string">&quot;sample 2&quot;</span>)), legend = <span class="literal">TRUE</span>)</span><br><span class="line">ot &lt;- OverlayTrack(trackList = <span class="built_in">list</span>(dtrack2, dtrack))</span><br><span class="line">ylims &lt;- extendrange(<span class="built_in">range</span>(<span class="built_in">c</span>(values(dtrack), values(dtrack2))))</span><br><span class="line">plotTracks(<span class="built_in">list</span>(itrack, gtrack, ot), from = lim[<span class="number">1</span>], to = lim[<span class="number">2</span>], ylim = ylims, type = <span class="built_in">c</span>(<span class="string">&quot;smooth&quot;</span>, <span class="string">&quot;p&quot;</span>))</span><br></pre></td></tr></table></figure>
<p><br><img src="https://s4.ax1x.com/2022/01/05/TX83b8.png" alt="G-38"></p>
<p>在支持它的设备上，透明度混合 (<em>Alpha blending</em>) 是一个有用的工具，可以从 track 覆盖中梳理出更多信息，至少在比较少量样本时是这样。</p>
<p>由此产生的透明度有效地消除了过度绘图的问题。以下示例仅在此小插图构建在具有透明度混合支持的系统上时才有效。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">displayPars(dtrack) &lt;- <span class="built_in">list</span>(alpha.title = <span class="number">1</span>, alpha = <span class="number">0.5</span>)</span><br><span class="line">displayPars(dtrack2) &lt;- <span class="built_in">list</span>(alpha.title = <span class="number">1</span>, alpha = <span class="number">0.5</span>)</span><br><span class="line">ot &lt;- OverlayTrack(trackList = <span class="built_in">list</span>(dtrack, dtrack2))</span><br><span class="line">plotTracks(<span class="built_in">list</span>(itrack, gtrack, ot), from = lim[<span class="number">1</span>],</span><br><span class="line">           to = lim[<span class="number">2</span>], ylim = ylims, type = <span class="built_in">c</span>(<span class="string">&quot;hist&quot;</span>), window = <span class="number">30</span>)</span><br></pre></td></tr></table></figure>
<p><br><img src="https://s4.ax1x.com/2022/01/05/TX8GVS.png" alt="G-39"></p>
<h3 id="参考">参考</h3>
<p><em>Gviz <a target="_blank" rel="noopener external nofollow noreferrer" href="http://www.bioconductor.org/packages/release/bioc/html/Gviz.html">http://www.bioconductor.org/packages/release/bioc/html/Gviz.html</a></em></p>
<p><em><a target="_blank" rel="noopener external nofollow noreferrer" href="https://github.com/ivanek/Gviz">https://github.com/ivanek/Gviz</a></em></p>
<p><em>Hahne, Florian, and Robert Ivanek. 2016. “Statistical Genomics: Methods and Protocols.” In, edited by Ewy Mathé and Sean Davis, 335–51. New York, NY: Springer New York. <a target="_blank" rel="noopener external nofollow noreferrer" href="https://doi.org/10.1007/978-1-4939-3578-9_16">https://doi.org/10.1007/978-1-4939-3578-9_16</a>.</em></p>
<p><em>Katz, Yarden, Eric T. Wang, Jacob Silterra, Schraga Schwartz, Bang Wong, Helga Thorvaldsdóttir, James T. Robinson, Jill P. Mesirov, Edoardo M. Airoldi, and Christopher B. Burge. 2015. “Quantitative Visualization of Alternative Exon Expression from Rna-Seq Data.” Bioinformatics. <a target="_blank" rel="noopener external nofollow noreferrer" href="https://doi.org/10.1093/bioinformatics/btv034">https://doi.org/10.1093/bioinformatics/btv034</a>.</em></p>
</article><div class="post-copyright"><div class="post-copyright__author"><span class="post-copyright-meta">文章作者: </span><span class="post-copyright-info"><a href="mailto:lixingzee@gmail.com" rel="external nofollow noreferrer">李星泽</a></span></div><div class="post-copyright__type"><span class="post-copyright-meta">文章链接: </span><span class="post-copyright-info"><a href="http://lxz9.com/2022/01/05/Gviz/">http://lxz9.com/2022/01/05/Gviz/</a></span></div><div class="post-copyright__notice"><span class="post-copyright-meta">版权声明: </span><span class="post-copyright-info">本博客所有文章除特别声明外，均采用 <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" rel="external nofollow noreferrer" target="_blank">CC BY-NC-SA 4.0</a> 许可协议。转载请注明来自 <a href="http://lxz9.com" target="_blank">生信技术</a>！</span></div></div><div class="tag_share"><div class="post-meta__tag-list"><a class="post-meta__tags" href="/tags/Gviz/">Gviz</a></div><div class="post_share"><div class="social-share" data-image="https://s4.ax1x.com/2022/01/05/TvKUv6.png" data-sites="facebook,twitter,wechat,weibo,qq"></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/social-share.js/dist/css/share.min.css" media="print" onload="this.media='all'"><script src="https://cdn.jsdelivr.net/npm/social-share.js/dist/js/social-share.min.js" defer></script></div></div><div class="post-reward"><div class="reward-button"><i class="fas fa-qrcode"></i> 打赏<div class="reward-main"><ul class="reward-all"><li class="reward-item"><a href="https://z3.ax1x.com/2021/01/19/sgZ5y6.jpg" rel="external nofollow noreferrer" target="_blank"><img class="post-qr-code-img" src="https://z3.ax1x.com/2021/01/19/sgZ5y6.jpg" alt="微信"/></a><div class="post-qr-code-desc">微信</div></li><li class="reward-item"><a href="https://z3.ax1x.com/2021/01/19/sgZIOK.jpg" rel="external nofollow noreferrer" target="_blank"><img class="post-qr-code-img" src="https://z3.ax1x.com/2021/01/19/sgZIOK.jpg" alt="支付宝"/></a><div class="post-qr-code-desc">支付宝</div></li></ul></div></div></div><nav class="pagination-post" id="pagination"><div class="prev-post pull-left"><a href="/2022/01/13/Phylogenomics/"><img class="prev-cover" src="https://s4.ax1x.com/2022/01/13/7Mt8UJ.png" onerror="onerror=null;src='/img/404.jpg'" alt="cover of previous post"><div class="pagination-info"><div class="label">上一篇</div><div class="prev_info">系统发育基因组学(Phylogenomics)的介绍以及实操</div></div></a></div><div class="next-post pull-right"><a href="/2021/12/30/Scatterplot3d/"><img class="next-cover" src="https://s4.ax1x.com/2021/12/31/TfUfy9.jpg" onerror="onerror=null;src='/img/404.jpg'" alt="cover of next post"><div class="pagination-info"><div class="label">下一篇</div><div class="next_info">R数据可视化｜使用Scatterplot3d包制作3D散点图</div></div></a></div></nav><hr/><div id="post-comment"><div class="comment-head"><div class="comment-headline"><i class="fas fa-comments fa-fw"></i><span> 评论</span></div></div><div class="comment-wrap"><div><div class="vcomment" id="vcomment"></div></div></div></div></div><div class="aside-content" id="aside-content"><div class="card-widget card-info"><div class="card-info-avatar is-center"><img class="avatar-img" src="https://z3.ax1x.com/2021/03/10/6JSyoq.png" onerror="this.onerror=null;this.src='/img/friend_404.gif'" alt="avatar"/><div class="author-info__name">李星泽</div><div class="author-info__description"></div></div><div class="card-info-data"><div class="card-info-data-item is-center"><a href="/archives/"><div class="headline">文章</div><div class="length-num">42</div></a></div><div class="card-info-data-item is-center"><a href="/tags/"><div class="headline">标签</div><div class="length-num">51</div></a></div><div class="card-info-data-item is-center"><a href="/categories/"><div class="headline">分类</div><div class="length-num">14</div></a></div></div><a class="button--animated" id="card-info-btn" target="_blank" rel="noopener external nofollow noreferrer" href="https://github.com/xxxxxx"><i class="fab fa-github"></i><span>Follow Me</span></a><div class="card-info-social-icons is-center"><a class="social-icon" href="https://blog.csdn.net/m0_49960764" rel="external nofollow noreferrer" target="_blank" title="CSDN"><i class="iconfont icon-csdn"></i></a><a class="social-icon" href="https://z3.ax1x.com/2021/11/25/oAI0h9.jpg" rel="external nofollow noreferrer" target="_blank" title="微信公众号"><i class="iconfont icon-weixin"></i></a><a class="social-icon" href="https://space.bilibili.com/113291922?from=search&amp;seid=3526734520747213504&amp;spm_id_from=333.337.0.0" rel="external nofollow noreferrer" target="_blank" title="bilibili"><i class="iconfont icon-bilibili-copy"></i></a><a class="social-icon" href="mailto:lixingzee@gmail.com" rel="external nofollow noreferrer" target="_blank" title="Email"><i class="fas fa-envelope"></i></a></div></div><div class="card-widget card-announcement"><div class="item-headline"><i class="fas fa-bullhorn card-announcement-animation"></i><span>公告</span></div><div class="announcement_content">如果你有更多想了解的内容，欢迎关注我的个人公众号：<b>生信技术</b> <br> <img src="https://s4.ax1x.com/2021/12/29/TcEUvq.gif"> <br> <br></div><timing></timing></div><div class="sticky_layout"><div class="card-widget" id="card-toc"><div class="item-headline"><i class="fas fa-stream"></i><span>目录</span></div><div class="toc-content"><ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#Gviz%E5%8C%85%E4%BB%8B%E7%BB%8D"><span class="toc-number">1.</span> <span class="toc-text">Gviz包介绍</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#%E7%BB%98%E5%88%B6-AnnotationTrack-%E5%9B%BE"><span class="toc-number">1.1.</span> <span class="toc-text">绘制 AnnotationTrack 图</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E6%B7%BB%E5%8A%A0%E5%9F%BA%E5%9B%A0%E7%BB%84%E5%9D%90%E6%A0%87"><span class="toc-number">1.2.</span> <span class="toc-text">添加基因组坐标</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E6%B7%BB%E5%8A%A0%E6%9F%93%E8%89%B2%E4%BD%93%E8%A1%A8%E6%84%8F%E6%96%87%E5%AD%97"><span class="toc-number">1.3.</span> <span class="toc-text">添加染色体表意文字</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E6%B7%BB%E5%8A%A0%E5%9F%BA%E5%9B%A0%E6%A8%A1%E5%9E%8B"><span class="toc-number">1.4.</span> <span class="toc-text">添加基因模型</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E6%94%BE%E5%A4%A7%E7%BB%98%E5%9B%BE"><span class="toc-number">1.5.</span> <span class="toc-text">放大绘图</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E6%B7%BB%E5%8A%A0%E5%BA%8F%E5%88%97track%E5%B9%B6%E6%94%BE%E5%A4%A7%E6%9F%A5%E7%9C%8B%E5%BA%8F%E5%88%97"><span class="toc-number">1.6.</span> <span class="toc-text">添加序列track并放大查看序列</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E6%B7%BB%E5%8A%A0data-track"><span class="toc-number">1.7.</span> <span class="toc-text">添加data track</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E7%BB%98%E5%9B%BE%E5%8F%82%E6%95%B0"><span class="toc-number">2.</span> <span class="toc-text">绘图参数</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#%E8%AE%BE%E7%BD%AE%E5%8F%82%E6%95%B0"><span class="toc-number">2.1.</span> <span class="toc-text">设置参数</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E7%BB%98%E5%9B%BE%E6%96%B9%E5%90%91"><span class="toc-number">2.2.</span> <span class="toc-text">绘图方向</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#Track-classes"><span class="toc-number">3.</span> <span class="toc-text">Track classes</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#GenomeAxisTrack-%E7%9A%84%E6%98%BE%E7%A4%BA%E5%8F%82%E6%95%B0"><span class="toc-number">3.1.</span> <span class="toc-text">GenomeAxisTrack 的显示参数</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#IdeogramTrack"><span class="toc-number">3.2.</span> <span class="toc-text">IdeogramTrack</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#DataTrack"><span class="toc-number">3.3.</span> <span class="toc-text">DataTrack</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E4%B8%8D%E5%90%8C%E5%9B%BE%E7%89%87%E7%B1%BB%E5%9E%8B"><span class="toc-number">3.4.</span> <span class="toc-text">不同图片类型</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E6%95%B0%E6%8D%AE%E5%88%86%E7%BB%84"><span class="toc-number">3.5.</span> <span class="toc-text">数据分组</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#%E4%BB%8E%E6%96%87%E4%BB%B6%E6%9E%84%E5%BB%BA-DataTrack-%E5%AF%B9%E8%B1%A1"><span class="toc-number">3.5.1.</span> <span class="toc-text">从文件构建 DataTrack 对象</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#AnnotationTrack"><span class="toc-number">3.6.</span> <span class="toc-text">AnnotationTrack</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#%E4%BB%8E%E6%96%87%E4%BB%B6%E6%9E%84%E5%BB%BA-AnnotationTrack-%E5%AF%B9%E8%B1%A1"><span class="toc-number">3.6.1.</span> <span class="toc-text">从文件构建 AnnotationTrack 对象</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#GeneRegionTrack"><span class="toc-number">3.7.</span> <span class="toc-text">GeneRegionTrack</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#%E4%BB%8E-fTranscriptDbsx-%E6%9E%84%E5%BB%BA-GeneRegionTrack-%E5%AF%B9%E8%B1%A1"><span class="toc-number">3.7.1.</span> <span class="toc-text">从 fTranscriptDbsx 构建 GeneRegionTrack 对象</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#BiomartGeneRegionTrack"><span class="toc-number">3.8.</span> <span class="toc-text">BiomartGeneRegionTrack</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Sequence-Track"><span class="toc-number">3.9.</span> <span class="toc-text">Sequence Track</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#AlignmentsTrack"><span class="toc-number">3.10.</span> <span class="toc-text">AlignmentsTrack</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#RNA-seq"><span class="toc-number">3.10.1.</span> <span class="toc-text">RNA-seq</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#DNA-seq"><span class="toc-number">3.10.2.</span> <span class="toc-text">DNA-seq</span></a></li></ol></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#Track%E7%AA%81%E5%87%BA%E6%98%BE%E7%A4%BA%E5%92%8C%E5%8F%A0%E5%8A%A0"><span class="toc-number">4.</span> <span class="toc-text">Track突出显示和叠加</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#%E7%AA%81%E5%87%BA%E6%98%BE%E7%A4%BA"><span class="toc-number">4.1.</span> <span class="toc-text">突出显示</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E5%8F%A0%E5%8A%A0"><span class="toc-number">4.2.</span> <span class="toc-text">叠加</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#%E5%8F%82%E8%80%83"><span class="toc-number">4.2.1.</span> <span class="toc-text">参考</span></a></li></ol></li></ol></li></ol></div></div><div class="card-widget card-recent-post"><div class="item-headline"><i class="fas fa-history"></i><span>最新文章</span></div><div class="aside-list"><div class="aside-list-item"><a class="thumbnail" href="/2022/03/30/git/" title="Git 用法总结"><img src="https://s1.ax1x.com/2022/03/31/qWDeiD.gif" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="Git 用法总结"/></a><div class="content"><a class="title" href="/2022/03/30/git/" title="Git 用法总结">Git 用法总结</a><time datetime="2022-03-29T16:00:00.000Z" title="发表于 2022-03-30 00:00:00">2022-03-30</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2022/03/26/TypeHints/" title="Python 类型提示和静态类型检查"><img src="https://s1.ax1x.com/2022/03/28/qsCkB6.jpg" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="Python 类型提示和静态类型检查"/></a><div class="content"><a class="title" href="/2022/03/26/TypeHints/" title="Python 类型提示和静态类型检查">Python 类型提示和静态类型检查</a><time datetime="2022-03-25T16:00:00.000Z" title="发表于 2022-03-26 00:00:00">2022-03-26</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2022/03/25/mypy/" title="使用 Mypy 进行 Python 静态类型检查"><img src="https://s1.ax1x.com/2022/03/28/qsi4AK.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="使用 Mypy 进行 Python 静态类型检查"/></a><div class="content"><a class="title" href="/2022/03/25/mypy/" title="使用 Mypy 进行 Python 静态类型检查">使用 Mypy 进行 Python 静态类型检查</a><time datetime="2022-03-24T16:00:00.000Z" title="发表于 2022-03-25 00:00:00">2022-03-25</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2022/03/21/poetry/" title="使用 Python Poetry 进行依赖管理"><img src="https://s1.ax1x.com/2022/03/20/qe8TVP.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="使用 Python Poetry 进行依赖管理"/></a><div class="content"><a class="title" href="/2022/03/21/poetry/" title="使用 Python Poetry 进行依赖管理">使用 Python Poetry 进行依赖管理</a><time datetime="2022-03-20T16:00:00.000Z" title="发表于 2022-03-21 00:00:00">2022-03-21</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2022/03/20/f-string/" title="Python f-string字符串格式化的介绍"><img src="https://s1.ax1x.com/2022/03/19/qVGfIO.jpg" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="Python f-string字符串格式化的介绍"/></a><div class="content"><a class="title" href="/2022/03/20/f-string/" title="Python f-string字符串格式化的介绍">Python f-string字符串格式化的介绍</a><time datetime="2022-03-19T16:00:00.000Z" title="发表于 2022-03-20 00:00:00">2022-03-20</time></div></div></div></div></div></div></main><footer id="footer"><div id="footer-wrap"><div class="copyright">&copy;2020 - 2022&nbsp;<i style="color:#FF6A6A;animation: announ_animation 0.8s linear infinite;"class="fas fa-heart"></i> By 李星泽</div></div><div class="container" id="jsi-flying-fish-container"><script src="/js/fish.js"></script></div><style>   @media only screen and (max-width: 767px){
   #sidebar_search_box input[type=text]{width:calc(100% - 24px)}
}</style></footer></div><div id="rightside"><div id="rightside-config-hide"><button id="readmode" type="button" title="阅读模式"><i class="fas fa-book-open"></i></button><button id="darkmode" type="button" title="浅色和深色模式转换"><i class="fas fa-adjust"></i></button><button id="hide-aside-btn" type="button" title="单栏和双栏切换"><i class="fas fa-arrows-alt-h"></i></button></div><div id="rightside-config-show"><button id="rightside_config" type="button" title="设置"><i class="fas fa-cog fa-spin"></i></button><button class="close" id="mobile-toc-button" type="button" title="目录"><i class="fas fa-list-ul"></i></button><a id="to_comment" href="#post-comment" title="直达评论"><i class="fas fa-comments"></i></a><button id="go-up" type="button" title="回到顶部"><i class="fas fa-arrow-up"></i></button></div></div><div id="local-search"><div class="search-dialog"><div class="search-dialog__title" id="local-search-title">本地搜索</div><div id="local-input-panel"><div id="local-search-input"><div class="local-search-box"><input class="local-search-box--input" placeholder="搜索文章" type="text"/></div></div></div><hr/><div id="local-search-results"></div><span class="search-close-button"><i class="fas fa-times"></i></span></div><div id="search-mask"></div></div><div><script src="/js/utils.js"></script><script src="/js/main.js"></script><script src="https://cdn.jsdelivr.net/npm/medium-zoom/dist/medium-zoom.min.js"></script><script src="/js/search/local-search.js"></script><div class="js-pjax"><script>function loadValine () {
  function initValine () {
    let initData = {
      el: '#vcomment',
      appId: '',
      appKey: '',
      placeholder: 'Please leave your footprints',
      avatar: 'monsterid',
      meta: 'nick,mail,link'.split(','),
      pageSize: '10',
      lang: 'en',
      recordIP: false,
      serverURLs: '',
      emojiCDN: '',
      emojiMaps: "",
      enableQQ: false,
      path: window.location.pathname,
    }

    if (true) { 
      initData.requiredFields= ('nick,mail'.split(','))
    }
    
    if (false) {
      const otherData = false
      initData = Object.assign(initData, otherData)
    }
    
    const valine = new Valine(initData)
  }

  if (typeof Valine === 'function') initValine() 
  else getScript('https://cdn.jsdelivr.net/npm/valine/dist/Valine.min.js').then(initValine)
}

if ('Valine' === 'Valine' || !true) {
  if (true) btf.loadComment(document.getElementById('vcomment'),loadValine)
  else setTimeout(loadValine, 0)
} else {
  function loadOtherComment () {
    loadValine()
  }
}</script><script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script></div><script src="/js/jquery.min.js"></script><script src="/js/fish.js"></script><script src="/js/timing.js"></script><script async src="https://cdn.jsdelivr.net/npm/aplayer"></script><script async src="https://cdn.jsdelivr.net/npm/meting/dist/Meting.min.js"></script><div class="aplayer no-destroy" data-id="000PeZCQ1i4XVs" data-server="tencent" data-type="artist" data-fixed="true" data-mini="true" data-listFolded="false" data-order="random" data-preload="none" data-autoplay="flase data-lrctype="3" " muted></div><script defer="defer" id="fluttering_ribbon" mobile="false" src="https://cdn.jsdelivr.net/npm/butterfly-extsrc@1/dist/canvas-fluttering-ribbon.min.js"></script><script id="canvas_nest" defer="defer" color="0,0,255" opacity="0.7" zIndex="-1" count="99" mobile="false" src="https://cdn.jsdelivr.net/npm/butterfly-extsrc@1/dist/canvas-nest.min.js"></script><script>(function(i,s,o,g,r,a,m){i["DaoVoiceObject"]=r;i[r]=i[r]||function(){(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;a.charset="utf-8";m.parentNode.insertBefore(a,m)})(window,document,"script",('https:' == document.location.protocol ? 'https:' : 'http:') + "//widget.daovoice.io/widget/f84bbafb.js","daovoice")
</script><script>var isChatBtn = false
daovoice('init', {
  app_id: 'f84bbafb',},{
  launcher: { 
     disableLauncherIcon: isChatBtn // 悬浮 ICON 是否显示
  },
});
daovoice('update');

if (isChatBtn) {
  var chatBtnFn = () => {
    var chatBtn = document.getElementById("chat_btn")
    chatBtn.addEventListener("click", function(){
      daovoice('show')
    });
  }
  chatBtnFn()
} else {
  if (false) {
    function chatBtnHide () {
      daovoice('update', {},{
        launcher: { 
        disableLauncherIcon: true // 悬浮 ICON 是否显示
        },
      });
    }
    function chatBtnShow () {
      daovoice('update', {},{
        launcher: { 
        disableLauncherIcon: false // 悬浮 ICON 是否显示
        },
      });
    }
  }
}</script><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/aplayer/dist/APlayer.min.css" media="print" onload="this.media='all'"><script src="https://cdn.jsdelivr.net/npm/aplayer/dist/APlayer.min.js"></script><script src="https://cdn.jsdelivr.net/gh/metowolf/MetingJS@1.2/dist/Meting.min.js"></script><script src="https://cdn.jsdelivr.net/npm/pjax/pjax.min.js"></script><script>let pjaxSelectors = [
  'title',
  '#config_change',
  '#body-wrap',
  '#rightside-config-hide',
  '#rightside-config-show',
  '.js-pjax'
]

if (false) {
  pjaxSelectors.unshift('meta[property="og:image"]', 'meta[property="og:title"]', 'meta[property="og:url"]')
}

var pjax = new Pjax({
  elements: 'a:not([target="_blank"])',
  selectors: pjaxSelectors,
  cacheBust: false,
  analytics: true,
  scrollRestoration: false
})

document.addEventListener('pjax:complete', function () {
  window.refreshFn()

  document.querySelectorAll('script[data-pjax]').forEach(item => {
    const newScript = document.createElement('script')
    const content = item.text || item.textContent || item.innerHTML || ""
    Array.from(item.attributes).forEach(attr => newScript.setAttribute(attr.name, attr.value))
    newScript.appendChild(document.createTextNode(content))
    item.parentNode.replaceChild(newScript, item)
  })

  GLOBAL_CONFIG.islazyload && window.lazyLoadInstance.update()

  typeof chatBtnFn === 'function' && chatBtnFn()
  typeof panguInit === 'function' && panguInit()

  if (typeof gtag === 'function') {
    gtag('config', 'G-QRG4NL4207', {'page_path': window.location.pathname});
  }

  typeof loadMeting === 'function' && document.getElementsByClassName('aplayer').length && loadMeting()

  // Analytics
  if (false) {
    MtaH5.pgv()
  }

  // prismjs
  typeof Prism === 'object' && Prism.highlightAll()

  typeof preloader === 'object' && preloader.endLoading()
})


document.addEventListener('pjax:send', function () {
  typeof preloader === 'object' && preloader.initLoading()
  
  if (window.aplayers) {
    for (let i = 0; i < window.aplayers.length; i++) {
      if (!window.aplayers[i].options.fixed) {
        window.aplayers[i].destroy()
      }
    }
  }

  typeof typed === 'object' && typed.destroy()

  //reset readmode
  const $bodyClassList = document.body.classList
  $bodyClassList.contains('read-mode') && $bodyClassList.remove('read-mode')

})</script><script>(function(){
  const bp = document.createElement('script');
  const curProtocol = window.location.protocol.split(':')[0];
  if (curProtocol === 'https') {
    bp.src = 'https://zz.bdstatic.com/linksubmit/push.js';
  }
  else{
    bp.src = 'http://push.zhanzhang.baidu.com/push.js';
  }
  bp.dataset.pjax = ''
  const s = document.getElementsByTagName("script")[0];
  s.parentNode.insertBefore(bp, s);
})()</script></div></body></html>